```{r data-set-creation, eval=FALSE}
## Data acquired from GDC
tcga_ssm <- fread("../data/simple_somatic_mutation.open.tsv.gz", header = T, sep = "\t")
## Somatic mutations focused in analysis
selected_mut <- c("3_prime_UTR_variant", "5_prime_UTR_premature_start_codon_gain_variant", "5_prime_UTR_variant", "frameshift_variant", "missense_variant", "splice_acceptor_variant", "splice_donor_variant", "stop_lost", "start_lost", "stop_gained")
## Filtering tcga_ssm for our selected mutations
tcga_ssm <- tcga_ssm[tcga_ssm$consequence_type %in% selected_mut, ]
## Selecting column that we are interested in
tcga_ssm <- data.frame(chr = paste0("chr",tcga_ssm$chromosome),
                   start = tcga_ssm$chromosome_start,
                   end = tcga_ssm$chromosome_end,
                   ref_allele = tcga_ssm$reference_genome_allele,
                   alt_allele = tcga_ssm$mutated_to_allele,
                   type = tcga_ssm$consequence_type,
                   mut_type = tcga_ssm$mutation_type,
                   ensembl_id = tcga_ssm$gene_affected,
                   donor_id = tcga_ssm$icgc_donor_id,
                   sample_id = tcga_ssm$submitted_sample_id,
                   stringsAsFactors = F)
## Removing duplicate rows
tcga_ssm <- tcga_ssm %>% distinct()
## Recategorizing the mutation type
tcga_ssm[which(tcga_ssm$type == "frameshift_variant"), "type"] <- tcga_ssm[which(tcga_ssm$type == "frameshift_variant"), "mut_type"] 
tcga_ssm <- tcga_ssm %>% mutate(type = case_when(type == "insertion of <=200bp" ~ "frameshins_variant", ## relabel values
                                                 type == "deletion of <=200bp" ~ "frameshdel_variant",
                                                 type == "splice_acceptor_variant" ~ "splice_variant",
                                                 type == "splice_donor_variant" ~ "splice_variant",
                                                 type == "start_lost" ~ "nonsense_variant",
                                                 type == "stop_gained" ~ "nonsense_variant",
                                                 type == "5_prime_UTR_premature_start_codon_gain_variant" ~ "UTR_5_prime_variant",
                                                 type == "5_prime_UTR_variant" ~ "UTR_5_prime_variant",
                                                 type == "3_prime_UTR_variant" ~ "UTR_3_prime_variant",
                                                 type == "missense_variant" ~ "missense_variant",
                                                 type == "stop_lost" ~ "nonstop_variant")) %>% 
  dplyr::select(-mut_type) ## rm mut_type col


## Preparing sample metadata ##

complete_sample <- tcga_ssm %>% .[,"sample_id"] %>% unique() %>% .[grep(pattern= "TCGA-", x = .)] # complete TCGA tumor samples

tcga_sample <- fread("../data/sample.tsv.gz", header = T, sep = "\t") ## loading in sample metadata
tcga_sample <- tcga_sample %>% dplyr::select(project_code, submitted_sample_id, submitted_specimen_id, icgc_donor_id, submitted_donor_id, study) %>% ## selecting the relevant columns
  .[.$submitted_sample_id %in% complete_sample,] %>% ## Only include samples in complete_sample
  mutate(type = substring(.$submitted_sample_id, first = 14, last = 15)) %>% ## Getting "sample type" info from TCGA barcode
  filter(type == "01") ## filtering 01 samples (primary solid tumor)
tcga_dup_id <- tcga_sample %>% .[.$icgc_donor_id %in% .[which(duplicated(.$submitted_donor_id)),]$icgc_donor_id,] %>% ## what are the dup donors
  filter(study == "PCAWG") ## selecting dup samples that are from PCAWG
tcga_sample <- tcga_sample %>% .[!.$icgc_donor_id %in% tcga_dup_id$icgc_donor_id, ] %>% ## remove the dups from tcga_samples
  rbind(., tcga_dup_id) ## rbind the fixed dups

tcga_donor <- fread("../data/donor.tsv.gz", header = T, sep = "\t") ## loading in sample survival data
tcga_donor <- tcga_donor %>% dplyr::select(icgc_donor_id, project_code, submitted_donor_id, donor_sex, donor_vital_status, disease_status_last_followup, donor_age_at_diagnosis, donor_age_at_enrollment, donor_age_at_last_followup, donor_survival_time, donor_interval_of_last_followup) ## selecting columns that we are interested in

sample_sheet <- left_join(tcga_sample, tcga_donor, by = c("submitted_donor_id", "project_code", "icgc_donor_id")) ## combining the tcga sample and donor data

## Preparing gene metadata ##

# genelist <- unique(as.character(tcga_ssm$ensembl_id)) ## complete genelist
# ensembl_2_entrez_full_list <- bitr(genelist, 
#                                    fromType = "ENSEMBL", 
#                                    toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
load("../data/ensembl_2_entrez_full_list.Rda")
tcga_ssm <- tcga_ssm[tcga_ssm$ensembl_id %in% ensembl_2_entrez_full_list$ENSEMBL,] ## Only include genes in the gdc data that is found in the ensembl_2_entrez_full_lust

## Now we only have 19067 genes
complete_ensembl <- as.data.frame(unique(as.character(tcga_ssm$ensembl_id))) ## ensembl gene list
names(complete_ensembl) <- "ensembl_id"

selected_mut <- unique(as.character(tcga_ssm$type))
## For every mutation type.. select them.. and create a tsv file 
for(selected_mut in selected_mut) {
  print(selected_mut)
  df <-  tcga_ssm %>% filter(type == selected_mut) %>% ## filter data by selected_mut
    dplyr::select(sample_id, ensembl_id) %>% ## select column sample_id and ensembl_id
    mutate(mutation = 1) %>% ## create a new column with 1's
    distinct() %>% ## remove duplicated rows
    spread(., key = "sample_id", value = "mutation") %>% ## long to wide format
    remove_rownames() %>% ## move ensembl_id column to rownames
    column_to_rownames(var = "ensembl_id") %>% 
    replace(is.na(.), values = 0) %>% ## convert NA's to 0's
    .[, names(.) %in% sample_sheet$submitted_sample_id] ## rm any sample_id not included in complete_sample
  missing_gene <- as.data.frame(matrix(data = 0, nrow = length(complete_ensembl$ensembl_id[!complete_ensembl$ensembl_id %in% rownames(df)]), ncol = length(df)))
  rownames(missing_gene) <- complete_ensembl$ensembl_id[!complete_ensembl$ensembl_id %in% rownames(df)] ## insert missing ensembl_id into row names
  names(missing_gene) <- names(df) ## insert sample_id names into col names
  df <- rbind(df, missing_gene) ## rbind df and missing_gene
  rm(missing_gene) ## rm data
  gc()
  df <- df[match(complete_ensembl$ensembl_id, rownames(df)),] ## match df rownames with genelist
  missing_sample <- as.data.frame(matrix(data = 0, nrow = nrow(df), ncol = length(sample_sheet$submitted_sample_id[!sample_sheet$submitted_sample_id %in% names(df)])))
  rownames(missing_sample) <- rownames(df) ## insert ensembl_id into row names
  names(missing_sample) <- sample_sheet$submitted_sample_id[!sample_sheet$submitted_sample_id %in% names(df)] ## insert missing sample_id names into col names
  df <- cbind(df, missing_sample) ## rbind df and missing_gene
  rm(missing_sample) ## rm data
  gc()
  df <- df[, match(sample_sheet$submitted_sample_id, names(df))]
  assign(paste0(selected_mut), value = df) ## create df in r environment
  rm(df)
  gc()
  }

# ensembl <- useEnsembl(biomart="ensembl", dataset = "hsapiens_gene_ensembl")
 # genelist <- rownames(missense_variant)
 # gene_data <- getBM(
 #   attributes = c(
 #     'ensembl_gene_id',
 #     'external_gene_name',
 #     'gene_biotype',
 #     'chromosome_name', 
 #     'start_position',
 #     'end_position', 
 #     'strand'),
 #   filters = 'ensembl_gene_id',
 #   values = genelist,
 #   mart = ensembl)
 # save(gene_data, file = "gene_data.Rda")
load("../data/gene_data.Rda")
###################### Converting ensembl id to gene symbol

## Saniety check to make sure all the genes in the mutation data is found in the gene_data. If not, 
## then only include the genes in gene_data
missense_variant <- missense_variant[rownames(missense_variant) %in% gene_data$ensembl_gene_id,]
UTR_3_prime_variant <- UTR_3_prime_variant[rownames(UTR_3_prime_variant) %in% gene_data$ensembl_gene_id,]
UTR_5_prime_variant <- UTR_5_prime_variant[rownames(UTR_5_prime_variant) %in% gene_data$ensembl_gene_id,]
frameshdel_variant <- frameshdel_variant[rownames(frameshdel_variant) %in% gene_data$ensembl_gene_id,]
frameshins_variant <- frameshins_variant[rownames(frameshins_variant) %in% gene_data$ensembl_gene_id,]
splice_variant <- splice_variant[rownames(splice_variant) %in% gene_data$ensembl_gene_id,]
nonstop_variant <- nonstop_variant[rownames(nonstop_variant) %in% gene_data$ensembl_gene_id,]
nonsense_variant <- nonsense_variant[rownames(nonsense_variant) %in% gene_data$ensembl_gene_id,]


## Matching df to the ensembl_id orders in gene_data
match_ensembl <- function(x) {
  x <- x[match(gene_data$ensembl_gene_id, rownames(x)),] 
}

missense_variant <- match_ensembl(missense_variant)
UTR_3_prime_variant <- match_ensembl(UTR_3_prime_variant)
UTR_5_prime_variant <- match_ensembl(UTR_5_prime_variant)
frameshdel_variant <- match_ensembl(frameshdel_variant)
frameshins_variant <- match_ensembl(frameshins_variant)
splice_variant <- match_ensembl(splice_variant)
nonstop_variant <- match_ensembl(nonstop_variant)
nonsense_variant <- match_ensembl(nonsense_variant)

## Convert gene_data into GRanges for Summarized Experiment
genes <- GRanges(
  seqnames = paste("chr", gene_data$chromosome_name, sep=""),
  ranges = IRanges(
    start = gene_data$start_position,
    end = gene_data$end_position),
  strand = gene_data$strand,
  mcols = list(
    ensembl_gene_id = gene_data$ensembl_gene_name,
    hgnc_symbol = gene_data$hgnc_symbol))
## Create Summarized Experiment
gdc_tumor_coding_complete <- SummarizedExperiment(
  assays = list(
    missense   = missense_variant,
    splice     = splice_variant,
    frameshdel = frameshdel_variant,
    frameshins = frameshins_variant,
    utr3       = UTR_3_prime_variant,
    utr5       = UTR_5_prime_variant,
    nonsense   = nonsense_variant,
    nonstop    = nonstop_variant),
  rowRanges = genes,
  colData = sample_sheet)
save(gdc_tumor_coding_complete, file = "../data/gdc_tumor_coding_complete.Rdata")
rm(list = ls(all.names = T))
gc()
```

```{r reactome-pathway-creation, eval=FALSE}
load("../data/gene_data.Rda")
## Preparing level_0 to level_1 pathway conversion table. level_0 is the master pathway. level_1 is the parent pathway
main2first <- fread(file = "../data/reactome_main2first_level.txt", header = T, sep = "\t")

main2first <- main2first %>% .[,2:3] %>%
  'colnames<-' (c("first_level", "main_level")) ## rename colnames

## Filtering out master pathways from first2second_level
first2second <- fread(file = "../data/reactome_first2second_level.txt", header = F, sep = "\t")
first2second <- first2second %>% 'colnames<-' (c("first_level", "second_level")) %>% ## rename colnames
  .[grep(pattern = "-HSA-", .$first_level)] %>% ## keep human pathways
  .[!.$first_level %in% unique(main2first$main_level),] %>% ## remove main level pathways from first level
  .[!.$first_level %in% unique(.$second_level),] ## remove second level pathways from first level

ensembl2reactome <- fread(file = "../data/Ensembl2Reactome_All_Levels.txt", header = F, sep = "\t") ## ensembl to pathway list acquired from reactome database
ensembl2reactome <- ensembl2reactome %>% .[,c(1:2,4)] %>% ## select columns ensembl, pathway_id, pathway_name columns
  'colnames<-' (c("ENSEMBL", "PATH_ID", "PATH_NAME")) %>% ## rename colnames
  .[grep(pattern = "-HSA-", .$PATH_ID),] %>% ## keep human pathways
  .[.$PATH_ID %in% first2second$second_level,] %>% ## include only second level pathways
  left_join(., main2first, by = c("PATH_ID" = "first_level")) %>% distinct() %>% 
  left_join(., ensembl2reactome[,c(2,4)], by = c("main_level" = "V2")) %>% distinct()

## Filtering pathways not relevant to our analysis by keywords
filtered_ensembl2reactome <- ensembl2reactome %>% filter(!V4 %in% c("Disease", "Muscle contraction", "Reproduction")) ## Pathways in disease, muscle contraction and reproduction parent pathway were removed

toMatch <- c("disease", "defect", "cancer", "oncogenic", "disorder", "infection", "mutant", "bacteria", "listeria", "loss") ## pathways with the following key terms were removed.

filtered_ensembl2reactome <- filtered_ensembl2reactome %>% mutate(l_pathname = tolower(.$PATH_NAME)) %>% ## lower casing the pathway names because some pathways have disease vs Disease
  .[!grep(paste(toMatch,collapse="|"), .$l_pathname),] ## Removing pathways with the key terms in toMatch 

filtered_ensembl2reactome <- filtered_ensembl2reactome %>% .[which(!.$PATH_ID %in% c("R-HSA-212436", "R-HSA-388396")),] ## Removing Generic Transcription Pathway and GPCR downstream signalling as they are broad pathways

gene2level2_lookup <- filtered_ensembl2reactome %>% dplyr::select(ENSEMBL, PATH_ID) %>% ## remove PATH_NAME col
  .[.$ENSEMBL %in% gene_data$ensembl_gene_id,] %>% ## only include genes that are in the gene_data 
  mutate(member = 1) %>% ## add a column for ensembl to pathway membership
  distinct() %>% ## keep unique rows
  spread(data = ., key = "PATH_ID", value = "member") %>% ## long to wide format
  column_to_rownames(var = "ENSEMBL") %>% ## ensembl_id as rownames
  replace(is.na(.), values = 0) ## replace NA's with 0's
#save(gene2level2_lookup, file = "../data/gene2_level2_lookup.Rda")
## Pathway names to level 2 lookup table
pathname2level2 <- filtered_ensembl2reactome %>% dplyr::select(PATH_ID, PATH_NAME) %>% ## remove ENSEMBL col
  .[.$PATH_ID %in% names(gene2level2_lookup),] %>% ## only include pathways in gene2level2_lookup
  distinct(.) ## keep unique rows
#save(pathname2level2, file = "../data/pathname2level2.Rda")
rm(list = ls(all.names = T))
gc()
```

```{r filter-tissue-specific-low-expressed-genes, eval=FALSE}
gtf_file <- "../data/gencode.v30.annotation.gtf" ## loading gene information downloaded from gencode
gencode_gtf <- rtracklayer::import(gtf_file) ## import data
gene_info <- as.data.frame(mcols(gencode_gtf)[, c("type", "gene_id", "transcript_id", "gene_type", "transcript_type", "gene_name")]) ## selecting relevant columns
## We only want transcript only 
gene_info <- gene_info[gene_info$type == "transcript",]
## We only want protein_coding only
gene_info <- gene_info[gene_info$gene_type == "protein_coding",]
gene_info <- gene_info[gene_info$transcript_type == "protein_coding",]
## Filtering for transcript_id, gene_id, gene_name columns
g2t <- gene_info %>% dplyr::filter(type == "transcript") %>% dplyr::select(transcript_id, gene_id, gene_name)
## Condense df to only having unique data. 
good.g2t <- as_tibble(unique(g2t))

## df with transcript_id and entrez id                      
gencode_entrez <- readr::read_tsv("../data/gencode.v30.metadata.EntrezGene.gz", col_names = FALSE)
colnames(gencode_entrez) <- c("transcript_id", "entrez_id")
## Joining data that are joined together by transcript_id
entrez_g2t <- left_join(good.g2t, gencode_entrez)
head(entrez_g2t[1:10,])
## Remove transcript_id column
entrez_g2t$transcript_id <- NULL
## Remove . and the numbers following from the ensembl id.
entrez_g2t$gene_id <- gsub("\\..*", "", entrez_g2t$gene_id)
## Removing data that does not have entrez id. 83091 out of 83647 had NA values
table(is.na(entrez_g2t$entrez_id))
entrez_g2t <- entrez_g2t[!is.na(entrez_g2t$entrez_id),]
entrez_g2t <- unique(entrez_g2t[, c("gene_id", "gene_name", "entrez_id")])
## Are there any duplicates in our df? Yes. Only one dupped ensembl id.
table(duplicated(entrez_g2t$gene_id))
## These are pseudogenes... We will remove them all.
dups_gene_id <- entrez_g2t[which(duplicated(entrez_g2t$gene_id)),]
## We're removing the rows that are associated with this pseudogene
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285607")),]
## It's gone
which(entrez_g2t$gene_id == "ENSG00000285607")
## Are there any duplicate gene names? Yes! 14 of them
table(duplicated(entrez_g2t$gene_name))
dups_gene_name <- entrez_g2t[which(duplicated(entrez_g2t$gene_name)),]
entrez_g2t[entrez_g2t$gene_name == "TBCE",]
## ENSG00000285053  is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285053")),]
entrez_g2t[entrez_g2t$gene_name == "PDE11A",]
## ENSG00000284741 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284741")),]
entrez_g2t[entrez_g2t$gene_name == "ATXN7",]
## ENSG00000285258 is not in the database anywhere
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285258")),]
entrez_g2t[entrez_g2t$gene_name == "MATR3",]
## ENSG00000280987 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000280987")),]
entrez_g2t[entrez_g2t$gene_name == "SOD2",]
## ENSG00000285441 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285441")),]
entrez_g2t[entrez_g2t$gene_name == "ABCF2",]
## ENSG00000285292 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285292")),]
entrez_g2t[entrez_g2t$gene_name == "PINX1",]
## ENSG00000258724 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000258724")),]
entrez_g2t[entrez_g2t$gene_name == "HSPA14",]
## ENSG00000284024 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284024")),]
entrez_g2t[entrez_g2t$gene_name == "IGF2",]
## ENSG00000284779 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284779")),]
entrez_g2t[entrez_g2t$gene_name == "ATF7",]
## ENSG00000267281 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000267281")),]
entrez_g2t[entrez_g2t$gene_name == "DIABLO",]
## ENSG00000284934 is a homolog
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284934")),]
entrez_g2t[entrez_g2t$gene_name == "GGT1",]
## ENSG00000286070 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286070")),]
entrez_g2t[entrez_g2t$gene_name == "SCO2",]
## ENSG00000130489 no longer exist
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000130489")),]
entrez_g2t[entrez_g2t$gene_name == "TMSB15B",]
## ENSG00000269226 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000269226")),]

## Are there any dupped entrez id? 
dups_entrez <- entrez_g2t[which(duplicated(entrez_g2t$entrez_id)),]

entrez_g2t[entrez_g2t$entrez_id == "55486",]
## ENSG00000283765 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000283765")),]
entrez_g2t[entrez_g2t$entrez_id == "55300",]
## ENSG00000281028 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000281028")),]
entrez_g2t[entrez_g2t$entrez_id == "653082",]
## ENSG00000286094 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286094")),]
entrez_g2t[entrez_g2t$entrez_id == "8622",]
## ENSG00000284762 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284762")),]
entrez_g2t[entrez_g2t$entrez_id == "221468",]
## ENSG00000286105 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286105")),]
entrez_g2t[entrez_g2t$entrez_id == "1124",]
## ENSG00000285162 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285162")),]
entrez_g2t[entrez_g2t$entrez_id == "102723849",]
## ENSG00000283765 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000273520")),]
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000262461")),]
entrez_g2t[entrez_g2t$entrez_id == "883",]
## ENSG00000286112 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286112")),]
entrez_g2t[entrez_g2t$entrez_id == "22891",]
## ENSG00000261130 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285551")),]
entrez_g2t[entrez_g2t$entrez_id == "54816",]
## ENSG00000285253 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285253")),]
entrez_g2t[entrez_g2t$entrez_id == "23370",]
## ENSG00000268861 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000268861")),]
entrez_g2t[entrez_g2t$entrez_id == "284391",]
## ENSG00000286098 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286098")),]
entrez_g2t[entrez_g2t$entrez_id == "440519",]
## ENSG00000283201 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000283201")),]
entrez_g2t[entrez_g2t$entrez_id == "149951",]
## ENSG00000285382 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285382")),]
entrez_g2t[entrez_g2t$entrez_id == "388820",]
## ENSG00000268861 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000278961")),]
entrez_g2t[entrez_g2t$entrez_id == "8471",]
## ENSG00000286098 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000260548")),]

## manual gene removal complete

load("../data/TCGA.RNA.Rda") ## loading in mRNA expression data from UCSC xena browser


## Put entrez ids as rownames and shortening the TCGA ids since the last three portions of the TCGA barcode is coded to explain the test performed and etc. 
tumor_samples <- gdc_tumor_coding_complete %>% colData(.) %>% 
  as.data.frame(.) %>%
  dplyr::select(project_code, submitted_specimen_id) %>%
  mutate(submitted_specimen_id = substring(text = .$submitted_specimen_id, first = 1, last = 15))

TCGA.RNA <- TCGA.RNA %>% dplyr::select(-gene_symbol) %>% ## Don't need gene symbol info
  remove_rownames(.) %>% column_to_rownames(var = "gene_id") %>%  ## entrez id to rownames
  .[rownames(.) %in% entrez_g2t$entrez_id,] %>% ## Filter TCGA data to only include genes found in gencode (entrez_g2t)
  'colnames<-' (substring(text = names(.), first = 1, last = 15)) %>% ## only need the first 15 character of barcode
  .[, names(.) %in% tumor_samples$submitted_specimen_id] ## only include tumor samples in our analysis

g2e_list <- entrez_g2t[entrez_g2t$entrez_id %in% rownames(TCGA.RNA),] ## getting the list of genes in entrez_g2t thats in TCGA.RNA

express_no_express_lookup <- list()
for(i in unique(tumor_samples$project_code)) {
  df <- which(tumor_samples$project_code == i) ## which rows are the spec cancer types
  df <- tumor_samples[df,] ## select only those rows
  gdc <- TCGA.RNA[, names(TCGA.RNA) %in% df$submitted_specimen_id] ## filter the norm_gdc_gene_exp dataset with those tumor samples
  mean_gdc <- as.data.frame(apply(gdc, MARGIN = 1, function(x) {
    mean(x)
  })) ## take the mean for each gene across the tumor samples
  names(mean_gdc) <- i
  express_no_express_lookup[[i]] <- mean_gdc ## store data in list
  rm(df, gdc, mean_gdc)
  gc
}
express_no_express_lookup <- do.call(cbind, express_no_express_lookup) ## cbind list
express_no_express_lookup[is.na(express_no_express_lookup)] <- 0 ## NA values are 0

gdc_ensembl_id <- gdc_tumor_coding_complete %>% rowRanges(.) %>% as.data.frame(.) %>% ## getting the ensembl_ids in our gdc dataset
  rownames(.)
g2e_list <- g2e_list[g2e_list$gene_id %in% gdc_ensembl_id,] ## subsetting the g2e_list to only have ensembl_ids in our gdc dataset


express_no_express_lookup <- express_no_express_lookup[rownames(express_no_express_lookup) %in% g2e_list$entrez_id, ] ## Saniety check to only include genes in express lookup that are in g2e_list
express_no_express_lookup[express_no_express_lookup <= 10] <- 0 ## Set threshold to <=10 for not expressed
express_no_express_lookup[express_no_express_lookup >10] <- 1 ## >10 is expressed

## convert entrez id to ensembl_id
express_no_express_lookup <- express_no_express_lookup %>% .[match(g2e_list$entrez_id, rownames(.)),] %>% ## match the entrez_id express_no_express_lookup with g2e_list
  'rownames<-' (g2e_list$gene_id) ## replace entrez_id with 

## Booleanize mutation count for each tumor sample
missense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$missense) ## Load in each mut type data
splice <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$splice)
frameshdel <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshdel)
frameshins <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshins)
utr3 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr3)
utr5 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr5)
nonstop <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonstop)
nonsense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonsense)

gdc_mut_count_patient <- missense + splice + frameshdel + frameshins + utr3 + utr5 + nonstop + nonsense ## add up all the mutation count
rm(missense, splice, frameshdel, frameshins, utr3, utr5, nonstop, nonsense)
gc()
gdc_mut_count_patient <- gdc_mut_count_patient[rownames(gdc_mut_count_patient) %in% rownames(express_no_express_lookup),] ## filter data with genes in the express_no_express_lookup
gdc_mut_count_patient[gdc_mut_count_patient > 0] <- 1 ## booleanize data, anything greater than 0 is equal to 1

gdc_gene_exp_adj <- list()
for(i in unique(tumor_samples$project_code)) {
  df <- which(tumor_samples$project_code == i) ## which rows are my spec cancer types
  df <- tumor_samples[df,] ## select only those rows
  gdc <- gdc_mut_count_patient[, names(gdc_mut_count_patient) %in% rownames(df)] ## filter the norm_gdc_gene_exp dataset with those tumor samples
  exp_data <- as.data.frame(express_no_express_lookup[, i]) ## select correct lookup column for the cancer type
  rownames(exp_data) <- rownames(express_no_express_lookup) ## Insert rownames to exp_data
  gdc <- gdc[match(rownames(exp_data), rownames(gdc)),] ## Match ensembl_id of exp_data with gdc data
  gdc_adj <- as.data.frame(apply(gdc, MARGIN = 2, function(x) {
    exp_data * x
  })) ## Function to multiply lookup to the data for the specific cancer type
  names(gdc_adj) <- names(gdc) ## Rename gdc_adj with tumor samples names
  gdc_gene_exp_adj[[i]] <- gdc_adj ## Store data to list
  rm(df, gdc, exp_data, gdc_adj)
  gc()
}
## Combining list into a dataframe.
gdc_gene_exp_adj <- do.call(cbind, gdc_gene_exp_adj) ## cbind list
names(gdc_gene_exp_adj) <- gsub("^.*\\.", "", names(gdc_gene_exp_adj)) ## tumor sample names have cancer type name in it. Removing it using gsub
rm(gdc_mut_count_patient, express_no_express_lookup, TCGA.RNA, gdc_tumor_coding_complete, gencode_gtf, dups_entrez, dups_gene_id, dups_gene_name, g2t, good.g2t, gene_info, g2e_list, entrez_g2t, gencode_entrez)
gc()
save(gdc_gene_exp_adj, file = "gdc_gene_exp_adj.Rda")
```

```{r genes-to-pathways-transformation, eval=FALSE}
load("../data/gdc_gene_exp_adj.Rda")
load("../data/gene2_level2_lookup.Rda")
sample_list <- as.character(names(gdc_gene_exp_adj)) ## Create sample list
complete_path <- as.character(names(gene2level2_lookup)) ## Get names of pathways

df_gene <- as.data.frame(t(gdc_gene_exp_adj)) ## Transpose data so samples are rows and genes are columns
df_gene <- df_gene > 0 ## if 1 == TRUE if 0 == FALSE
df_gene <- as.data.frame(df_gene) ## turn matrix back to df

df_gene <- df_gene %>% rownames_to_column(var = "sample") %>% 
  gather(key = "gene", value = "mutation", -sample) %>% ## Wide to long format
  .[.$mutation,] %>% ## Keep TRUE's (which are 1's); Remove FALSE's (0's)
  dplyr::select(-mutation) ## Remove column named mutation

path_lookup <- as.data.frame(t(gene2level2_lookup)) ## Transpose data so paths are rows and genes are columns
path_lookup <- path_lookup > 0 ## if 1 == TRUE if 0 == FALSE
path_lookup <- as.data.frame(path_lookup) ## turn matrix back to df

path_lookup <- path_lookup %>% rownames_to_column(var = "path") %>%
  gather(key = "gene", value = "path_member", -path) %>% ## Wide to long format
  .[.$path_member,] %>% ## Keep TRUE's (which are 1's); Remove FALSE's (0's)
  dplyr::select(-path_member)

df_lookup <- df_gene %>% left_join(., path_lookup, by = "gene") %>% ## Match df_gene and lookup dataframes
  na.omit(.) ## Omit NA's that were generated due to no matches

## getting the genes involved in our analysis
genes_in_pathanalysis <- unique(df_lookup$gene)
write.table(genes_in_pathanalysis, file = "genes_in_pathanalysis.tsv", quote = FALSE, sep = "\t", row.names =  FALSE) ## save data

df_lookup <- as.data.frame(table(df_lookup$sample, df_lookup$path)) ## Generate 1's for the matches

df_lookup <- df_lookup %>% spread(key = "Var1", value = "Freq") %>% ## Long to wide format
  remove_rownames() %>%
  column_to_rownames(var = "Var2") ## path_id to rownames

## Insert missing tumor samples due to no mutations
df_missing <- sample_list[!(sample_list %in% names(df_lookup))] 
mx <- as.data.frame(matrix(data = 0, nrow = nrow(df_lookup), ncol = length(df_missing))) ## Create mx to store data
names(mx) <- df_missing ## Missing tumor sample names in col names
rownames(mx) <- rownames(df_lookup) ## Path names in rownames
df_lookup <- cbind(df_lookup, mx) ## cbind data

## Insert missing pathway due to no mutations
path_missing <- complete_path[!c(complete_path %in% rownames(df_lookup))]
mx <- as.data.frame(matrix(data = 0, nrow = length(path_missing), ncol = ncol(df_lookup))) ## Create mx to store data
names(mx) <- names(df_lookup) ## Insert tumor sample names
rownames(mx) <- path_missing ## Insert missing path names
df_lookup <- rbind(df_lookup, mx) ## cbind data

pathway_order <- rownames(df_lookup) ## need to perserve the order of pathway for the next operation
gdc_reactome_path_analysis <- df_lookup %>% mutate_if(is.numeric, ~1 * (. > 0)) %>% as.data.frame() %>% ## Mutate anything >0 to 1's (Booleanize data)
  'rownames<-' (pathway_order) %>% ## rename rows with pathway order names
  rownames_to_column(var = "PATH_ID") %>% ## move rownames to column
  left_join(., pathname2level2, by = "PATH_ID") %>% ## leftjoin pathname with the data
  dplyr::select(-PATH_ID) %>% ## get rid of the path id 
  remove_rownames() %>% column_to_rownames(var = "PATH_NAME") ## move pathnameto rownames
rm(df_missing, path_missing, mx, df_gene, sample_list, complete_path, pathway_order, df_lookup)
gc()
save(gdc_reactome_path_analysis, file = "gdc_reactome_path_analysis.Rda")
```

```{r tumor-clusters, eval = FALSE}
###################
## Figure 1 data ##
###################
load("../data/gdc_reactome_path_analysis.Rda")
load("../data/gdc_tumor_coding_complete.Rdata")
load("../data/gene2_level2_lookup.Rda")

## Column distance metric for pheatmap
path_daisy <- daisy(t(gdc_reactome_path_analysis), type = list(symm  = c(1:col(gdc_reactome_path_analysis)))) 
path_daisy <- as.dist(as.matrix(path_daisy))
## Row distance metric for pheatmap
path_daisy_row <- daisy(gdc_reactome_path_analysis, type = list(symm  = c(1:nrow(gdc_reactome_path_analysis))))
path_daisy_row <- as.dist(as.matrix(path_daisy_row))

pathway_dendro <- pheatmap(gdc_reactome_path_analysis,
         show_rownames = F,
         show_colnames = F,
         cellheight = 1,
         cellwidth = 0.5,
         clustering_distance_cols = path_daisy,
         clustering_distance_rows = path_daisy_row,
         cluster_cols = T,
         cluster_rows = T,
         cutree_cols = 19,
         annotation_legend = F,
         legend_breaks = c(0, 0.25, 0.75, 1),
         legend_labels = c("","Not Mut", "Mut", ""),
         color = c("grey", "firebrick3"),
         clustering_method = "ward.D2", ## ward.D2 clustering
         legend = T)
dev.off()
pathway_dendro <- pathway_dendro$tree_row$order ## getting pathway row ordering (numerical values)
df <- rownames(gdc_reactome_path_analysis) ## getting pathway name from the pathway analysis
pathway_dendro <- df[pathway_dendro] ## Reorder the pathway names based on the ordering from pheatmap

tumor_samples <- gdc_tumor_coding_complete %>% colData(.) %>%  ## get sample_id data from gdc
  as.data.frame(.) %>%
  dplyr::select(project_code, submitted_sample_id, submitted_specimen_id) %>%  ## selecting relevant colums
  mutate(project_code = gsub(pattern = "\\-.*", replacement = "", x = .$project_code)) ## getting rid of the -US at the end of each cancer type

cancer_type <- tumor_samples %>% .$project_code %>% unique %>% .[order(.)] ## get unique cancer_type

cancer_dendro <- list() ## getting cancer type specific tumor sample clustering info

for(n in cancer_type) {
  cancer_info <- tumor_samples %>% rownames_to_column(var = "sample_id") %>% ## move sample_id to column
    filter(project_code == n) %>% ## filter for specific cancer type
    .[, "sample_id"] ## retain only sample_id
  cancer_path <- gdc_reactome_path_analysis %>% .[, names(.) %in% cancer_info] ## only including filtered sample_id
  cancer_heatmap <- pheatmap(cancer_path[pathway_dendro,], ## using pathway row order 
                          show_rownames = F,
                          show_colnames = F,
                          cellheight = 1,
                          cellwidth = 0.5,
                          cluster_cols = T,
                          cluster_rows = F,
                          annotation_legend = F,
                          color = c("grey", "firebrick3"),
                          clustering_method = "single", ## ordering is single linkage
                          legend = T)
  cancer_order <- cancer_heatmap$tree_col$order ## cancer order
  cancer_path <- names(cancer_path) ## getting the sample_id original order 
  cancer_dendro[[n]] <- cancer_path[cancer_order] ## reorder cancer by heatmap order
  rm(cancer_info, cancer_path)
  gc()
}
dev.off()

for(n in cancer_type[1]) { ## getting the order of clustering by cancer type by alphabetical order
  print(n)
  cancer_order <- as.data.frame(do.call(cbind, cancer_dendro[n])) ## read in first list
  names(cancer_order) <- "sample_id"
  cancer_order$type <- n
  for(p in cancer_type[-1]) {
    print(p)
    df1 <- as.data.frame(do.call(cbind, cancer_dendro[p])) ## read in rest of list in loop
    names(df1) <- "sample_id"
    df1$type <- p
    cancer_order <- rbind(cancer_order, df1) ## combine the two df and repeat the loop until end of cancer_type
    rm(df1)
    gc()
  }
}

cancer_dendro <- cancer_order %>% .[,"sample_id"] %>% as.character(.) ## sample_id ordering

cancer_disrupt_heatmap <- gdc_reactome_path_analysis %>% rownames_to_column(var = "path_name") %>%  ## move rownames to column
  gather(., key = "sample_id", value = "membership", -path_name) ## wide to long format
cancer_disrupt_heatmap$membership <- as.character(cancer_disrupt_heatmap$membership) ## turn boolean values into categories

## set levels based on clustering algorithm above
cancer_disrupt_heatmap$path_name <- factor(cancer_disrupt_heatmap$path_name, levels = c(rev(pathway_dendro)))
cancer_disrupt_heatmap$sample_id <- factor(cancer_disrupt_heatmap$sample_id, levels = c(cancer_dendro))

cancer_disrupt_heatmap <- left_join(cancer_disrupt_heatmap, cancer_order, by = "sample_id") ## add in cancer type information for faceting
save(cancer_disrupt_heatmap, file = "cancer_disrupt_heatmap.Rda") 

########################
## Figure 2A + B data ##
########################
gdc_path <- gdc_reactome_path_analysis %>%
  t() %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample") %>% ## move path_names to column
  .[match(tumor_samples$submitted_sample_id, .$sample), ]

gdc_path_mca <- left_join(gdc_path, tumor_samples[, c("project_code", "submitted_sample_id")],
  by = c("sample" = "submitted_sample_id")) %>%
  column_to_rownames(var = "sample")

gdc_path_sample_names <- rownames(gdc_path_mca) ## Getting sample id order
gdc_path_tumor_types <- gdc_path_mca$project_code ## Getting cancer type order

gdc_path_mca$project_code <- NULL ## Remove project_code column from df

## create a categorical variable from the boolean variable
pathway_list <- make_clean_names(colnames(gdc_path_mca))
pathway_count <- length(pathway_list)
gdc_path_mca <- as.data.frame(lapply(gdc_path_mca, function(x) {
  factor(x, levels = c(0, 1), labels = c("no", "yes")) ## converting boolean values to categorical 'yes' or 'no'
}))
colnames(gdc_path_mca) <- pathway_list
gdc_path_mca$project_code <- gdc_path_tumor_types
rownames(gdc_path_mca) <- gdc_path_sample_names

## perform mca
mca_res <- MCA(X = gdc_path_mca[, pathway_list], graph = FALSE)

expl_var <- mca_res$eig[, 2]
expl_var_perm <- matrix(NA, ncol = length(mca_res$eig[, 2]), nrow = 100) ## creating a matrix for permutation data

set.seed(4242)
for (k in seq(100)) { ## Perform 100 permutation MCA analysis
  gdc_perm <- as.data.frame(lapply(gdc_path_mca[, pathway_list], sample)) ## permute data
  MCA_perm <- MCA(X = gdc_perm, graph = FALSE) ## perform MCA analysis 
  expl_var_perm[k, ] <- MCA_perm$eig[, 2] ## Store data in mx. 
}

randomize <- tibble(
  eigenvalue = 1:length(expl_var),
  expl_var = expl_var, ## observed explained variance
  perm = colMeans(expl_var_perm), ## mean permuted explained variance
  pvalue = rowSums(t(expl_var_perm) >= expl_var) / 100
)

## Determining the optimal mca dimension
optimal_mca_dimensions <- randomize %>%
  filter(pvalue < 0.05) %>% ## filtering for pvalue < 0.05
  filter(eigenvalue == max(eigenvalue)) %>% ## filtering for the maximum eigenvalue
  pull(eigenvalue) 

mca_res <- MCA(
  X = gdc_path_mca[, pathway_list],
  ncp = optimal_mca_dimensions, graph = FALSE
)

save(randomize, mca_res, file = "figure1B_data.rda")

####################
## Figure 2C data ##
####################
n_iter <- 40
max_n_pt <- 200
rng <- RNGseq(max_n_pt * n_iter, 42)
r <- rng[[3649]]

rngtools::setRNG(r)
gdc.umap <- umap(mca_res$ind$coord, 
                 n_components = 3, ## reducing mca 36 dimension into 3 dimension
                 n_neighbors = round(sqrt(nrow(mca_res$ind$coord))), ## 87; which is the sqrt of sample size 7,607
                 n_epochs = 1000, ## number of times to find clusters
                 n_threads = 1)
gdc_umap_coors <- as.data.frame(gdc.umap) %>%
  rename(plot_x = V1, plot_y = V2, plot_z = V3) # extract x, y, z coordinates from umap
rownames(gdc_umap_coors) <- rownames(gdc_path_mca) 

####################
## Figure 2D data ## Optimization of HDBscan
####################
n_iter <- 40
max_n_pt <- 200
score_mca <- matrix(as.numeric(NA), nrow = max_n_pt - 2, ncol = n_iter)

n_tumors <- nrow(mca_res$ind$coord)

cl <- makeCluster(8)
registerDoParallel(cl)

for (i in 3:max_n_pt) {
  print(paste0("MIN POINTS = ", i))
  score_iter <- foreach(1:n_iter,
    .combine = c,
    .export = c("mca_res"),
    .packages = c("dbscan", "uwot"),
    r = rng[(i - 1) * n_iter + 1:n_iter]
  ) %dopar% {
    rngtools::setRNG(r)
    umap_init_mca <- umap(mca_res$ind$coord, 
                          min_dist = 1e-6, 
                          n_components = 3, 
                          n_neighbors = round(sqrt(nrow(mca_res$ind$coord))), 
                          n_epochs = 1000, 
                          n_threads = 1)
    res <- hdbscan(umap_init_mca, minPts = i)
    score_iter_temp <- sum(res$membership_prob < 0.05) / n_tumors
    return(score_iter_temp)
  }
  score_mca[i - 2, ] <- score_iter
}

score <- rowMeans(score_mca)
score_df <- tibble(minPts = 3:max_n_pt, score = score)

sample_df <- score_mca %>%
  as.data.frame() %>%
  as_tibble() %>%
  add_column(minPts = 3:max_n_pt, .before = 1) %>%
  pivot_longer(-minPts) %>%
  group_by(minPts)

optimal_minPts <- score_df %>%
  filter(score == min(score)) %>%
  pull(minPts)

save(score_df, sample_df, optimal_minPts, file = "optimal_minpts_score.rda")

####################
## Figure 2E data ## HDBscan umap to find clusters
####################
rngtools::setRNG(r)

cluster_umap <- umap(mca_res$ind$coord,
  min_dist = 1e-6,
  n_components = 3,
  n_neighbors = round(sqrt(nrow(mca_res$ind$coord))), 
  n_epochs = 1000,
  n_threads = 1
)
colnames(cluster_umap) <- c("cluster_x", "cluster_y", "cluster_z")
hdb_cluster_assignment <- hdbscan(cluster_umap, minPts = optimal_minPts)
umap_3d_coors <- gdc_umap_coors %>%
  bind_cols(as.data.frame(cluster_umap)) %>%
  mutate(
    n_pathways_mut = rowSums(gdc_path[, -1]),
    clust = as.factor(hdb_cluster_assignment$cluster),
    member_prob = hdb_cluster_assignment$membership_prob,
    sample_id = rownames(gdc_umap_coors)
  )
## clust 0 is unclassified cluster                       

####################
## Figure 2F data ## kNN clustering of unclassified clusters
####################
## reclassifying clust 0 using our classified cluster as training test and unclassified as test set
umap_3d_coors$clust_knn <- umap_3d_coors$clust
umap_3d_coors$clust_knn[umap_3d_coors$clust == 0] <- class::knn(
  train = umap_3d_coors[which(umap_3d_coors$clust != 0), c("cluster_x", "cluster_y", "cluster_z")],
  test = umap_3d_coors[which(umap_3d_coors$clust == 0), c("cluster_x", "cluster_y", "cluster_z")],
  cl = umap_3d_coors$clust[umap_3d_coors$clust != 0], k = 87
)

###################
## Figure 4 data ## Mutation count
###################
missense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$missense) ## reading in mutation data from SE
splice <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$splice)
frameshdel <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshdel)
frameshins <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshins)
utr3 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr3)
utr5 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr5)
nonstop <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonstop)
nonsense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonsense)

mutation_count <- missense + splice + frameshdel + frameshins + utr3 + utr5 + nonstop + nonsense ## add up all the mutation count
rm(missense, splice, frameshdel, frameshins, utr3, utr5, nonstop, nonsense)
gc()
mutation_count <-mutation_count %>% .[rownames(.) %in% rownames(gene2level2_lookup),] %>% ## filter data with genes in our pathway lookup table
  colSums() %>% as.data.frame() %>% ## sum the mutation count
  'colnames<-' (c("mut_count")) %>% ## rename colname
  rownames_to_column(var = "sample_id") ## move rownames to column

umap_3d_coors <- left_join(umap_3d_coors, mutation_count, by = "sample_id") 
rm(mutation_count)

####################
## Figure 3A data ## Cancer type projections on UMAP
####################
umap_3d_coors <- left_join(umap_3d_coors, tumor_samples[,c("project_code", "submitted_sample_id")], by = c("sample_id" = "submitted_sample_id"))

####################
## Figure 3B data ## Subtype projections on UMAP
####################
## getting subtype information on our cancer type
subtypes <- PanCancerAtlas_subtypes() 
tumor_subtypes <- substring(text = tumor_samples$submitted_specimen_id, first = 1, last = 12) ## sample_id for subsetting subtypes
subtypes <- subtypes %>% dplyr::select(pan.samplesID, Subtype_Selected) %>% ## select the relevant columns
  mutate(Subtype_Selected = gsub(pattern = ".*\\.", replacement = "", x = .$Subtype_Selected)) %>% ## remove the cancer acronym at the beginning
  mutate(pan.samplesID = substring(text = .$pan.samplesID, first = 1, last = 12)) %>% ## interested in 
  distinct() %>% ## only unique columns
  .[.$pan.samplesID %in% tumor_subtypes, ] %>% ## include sample_ids that are in our analysis
  filter(Subtype_Selected != "Normal") %>%
  .[!.$pan.samplesID %in% .[which(duplicated(.$pan.samplesID)),]$pan.samplesID, ] ## remove dups

#save(subtypes, file = "subtypes.Rda")
load("../data/subtypes.Rda")

umap_3d_coors <- umap_3d_coors %>%
  mutate(pan.samplesID = substring(text = .$sample_id, first = 1, last = 12)) %>%
  left_join(., subtypes, by = "pan.samplesID") %>% ## left_join subtype data to umap
  dplyr::select(-pan.samplesID)

##########################
## Figure 3D and E data ## Staging and metastasis projection onto UMAP
##########################
## getting tumor_metastasis, staging
tcga_clinical_tumor_staging <- list()
for(n in cancer_type) {
  print(n)
  cancername <- paste0("TCGA-",n)
  query <- GDCquery(project = cancername, ## downloading clinical data for each cancer type using TCGAbiolink
                  data.category = "Clinical",
                  data.type = "Clinical Supplement",
                  data.format = "BCR Biotab",
                  file.type = "patient")
  GDCdownload(query)
  clinical <- GDCprepare(query)[[1]]
  if("ajcc_pathologic_tumor_stage" %in% colnames(clinical) & "ajcc_tumor_pathologic_pt" %in% colnames(clinical) == T) {
    cat("Yep, ajcc info in there!\n") ## If all relevant info is there. Do this.
    clinical <- clinical[-c(1:2), c("bcr_patient_barcode","ajcc_pathologic_tumor_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")]
  }
  if(!"ajcc_pathologic_tumor_stage" %in% colnames(clinical) & "clinical_stage" %in% colnames(clinical) & "ajcc_tumor_pathologic_pt" %in% colnames(clinical)== T) { 
    cat("Nope, ajcc stage not there, but pathologic scores is!\n") ## If ajcc_pathologic_tumor_stage is not there. Do this.
    clinical <- clinical[-c(1:2), c("bcr_patient_barcode", "clinical_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")] ## Use clinical stage in place of ajcc pathologic tumor stage
    names(clinical) <- c("bcr_patient_barcode","ajcc_pathologic_tumor_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")
  }
  if(!"ajcc_pathologic_tumor_stage" %in% colnames(clinical) & "clinical_stage" %in% colnames(clinical) & !"ajcc_tumor_pathologic_pt" %in% colnames(clinical) & "pathologic_T" %in% colnames(clinical) == T) {
    cat("Nope, ajcc info not there!\n") ##If ajcc pathologic tumor stage andpathologic pt is not there. Do this
    clinical <- clinical[-c(1:2), c("bcr_patient_barcode", "clinical_stage", "pathologic_T", "pathologic_M")]
    names(clinical) <- c("bcr_patient_barcode","ajcc_pathologic_tumor_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm") ## Use clinical stage in place of ajcc pathologic tumor stage, and pathologic T in place of ajcc tumor pathologic pt.
  }
  ## If ajcc pathologic tumor stage and clinical stage is not present, skip to next
  if(!"ajcc_pathologic_tumor_stage" %in% colnames(clinical) & !"clinical_stage" %in% colnames(clinical) == T) next
  ## If ajcc_pathologic tumor stage is present and not ajcc_tumor_pathologic pt, skip to next
  if("ajcc_pathologic_tumor_stage" %in% colnames(clinical) & !"ajcc_tumor_pathologic_pt" %in% colnames(clinical) == T) next
  tcga_clinical_tumor_staging[[n]] <- clinical ## store data in list
}

tcga_clinical_tumor_staging <- do.call(rbind, tcga_clinical_tumor_staging) ## rbind the list together
tcga_clinical_tumor_staging <- tcga_clinical_tumor_staging[tcga_clinical_tumor_staging$bcr_patient_barcode %in% tumor_subtypes,] ## We have 6633 out of 7607 tumor data
missing_sample <- as.data.frame(matrix(data = "NA", nrow = length(tumor_subtypes[!tumor_subtypes %in% tcga_clinical_tumor_staging$bcr_patient_barcode]), ncol = length(tcga_clinical_tumor_staging) -1 )) ## creating df of NA's for missing samples 

tcga_clinical_tumor_staging <- missing_sample %>% 'colnames<-' (names(tcga_clinical_tumor_staging[-1])) %>% ## colnames based on tcga_clinical_tumor_staging
  mutate(bcr_patient_barcode = tumor_subtypes[!tumor_subtypes %in% tcga_clinical_tumor_staging$bcr_patient_barcode]) %>%
  .[,c(4,1,2,3)] %>% ## reorder columns for rbinding
  rbind(tcga_clinical_tumor_staging, .) ## rbind

save(tcga_clinical_tumor_staging, file = "tcga_clinical_tumor_staging.Rda")

umap_3d_coors <- umap_3d_coors %>%
  mutate(bcr_patient_barcode = substring(text = .$sample_id, first = 1, last = 12)) %>%
  left_join(., tcga_clinical_tumor_staging, by = "bcr_patient_barcode") %>% ## left_join clinical data to umap
  dplyr::select(-bcr_patient_barcode)

## Convert metastasis inputs into set categorical values
M0 <- umap_3d_coors$ajcc_metastasis_pathologic_pm %in% c("cM0 (i+)", "M0")
M1 <- umap_3d_coors$ajcc_metastasis_pathologic_pm %in% c("M1", "M1a", "M1b", "M1c")
NA.value <- umap_3d_coors$ajcc_metastasis_pathologic_pm %in% c("[Discrepancy]", "[Not Applicable]", "[Not Available]", "MX")
  
umap_3d_coors$ajcc_metastasis_pathologic_pm[M0] <- "M0"
umap_3d_coors$ajcc_metastasis_pathologic_pm[M1] <- "M1"
umap_3d_coors$ajcc_metastasis_pathologic_pm[NA.value] <- "NA"

## Convert stage inputs into set categorical values
stage1 <- umap_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage I", "Stage IA", "Stage IA1", "Stage IA2", "Stage IB", "Stage IB1", "Stage IB2", "Stage IC")
stage2 <- umap_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage II", "Stage IIA", "Stage IIA1", "Stage IIA2", "Stage IIB", "Stage IIC")
stage3 <- umap_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage III", "Stage IIIA", "Stage IIIB", "Stage IIIC", "Stage IIIC1", "Stage IIIC2")
stage4 <- umap_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage IV", "Stage IVA", "Stage IVB", "Stage IVC")
unknown <- umap_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage X", "[Discrepancy]", "[Not Applicable]", "I/II NOS", "NA", "[Not Available]")

umap_3d_coors$ajcc_pathologic_tumor_stage[stage1] <- "Stage I"
umap_3d_coors$ajcc_pathologic_tumor_stage[stage2] <- "Stage II"
umap_3d_coors$ajcc_pathologic_tumor_stage[stage3] <- "Stage III"
umap_3d_coors$ajcc_pathologic_tumor_stage[stage4] <- "Stage IV"
umap_3d_coors$ajcc_pathologic_tumor_stage[unknown] <- "Not Available"

write.table(x = umap_3d_coors, file = "umap_3d_coors.tsv", quote = FALSE, sep = "\t", row.names = FALSE) ## save the data
```

```{r fig-1a-1l, warning=FALSE, fig.height=12, fig.width=11, message = FALSE}
#fig.height=15, fig.width=16,
load("../data/cancer_disrupt_heatmap.Rda")
umap_cancertype <- fread("../data/umap_3d_coors.tsv")

kelly.colours <- c("gray95", "gray13", "gold2", "plum4", "darkorange1",
                   "lightskyblue2", "firebrick", "burlywood3", "gray51", "springgreen4",
                   "lightpink2", "deepskyblue4", "lightsalmon2", "mediumpurple4",
                   "orange", "maroon", "yellow3", "brown4", "yellow4", "sienna4", "chocolate", "gray19")

fig1A <- ggplot(data = cancer_disrupt_heatmap, aes(y=sample_id, x=path_name, fill = membership)) + 
  geom_tile(size = 0.6) +
  scale_fill_manual(values = c("grey", "firebrick3")) +
  facet_grid(type ~ ., scales = "free", space = "free", switch = "y") +
scale_y_discrete(position = "left") +
  ylab("") +
  xlab("377 Reactome Pathways") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        strip.text.y = element_text(size = 12, hjust = 0),
        strip.text.y.left = element_text(angle = 0),
        strip.background = element_blank(),
        legend.position = "none") 

## fixing the text cutoff in heatmap
 fig1Ag <- ggplotGrob(fig1A)
 
 for(i in which(grepl("strip-l", fig1Ag$layout$name))){
   fig1Ag$grobs[[i]]$layout$clip <- "off"
 }
 
 fig1A <- as.ggplot(fig1Ag)
 rm(fig1Ag)
 
##figure 1b
load("../data/figure1B_data.rda")
optimal_mca_dimensions <- randomize %>%
  filter(pvalue < 0.05) %>%
  filter(eigenvalue == max(eigenvalue)) %>%
  pull(eigenvalue)

permut_p <- randomize %>%
  filter(eigenvalue < 50) %>%
  ggplot(aes(x = eigenvalue, y = expl_var / 100)) +
  geom_line(aes(colour = "#998ec3")) +
  geom_line(aes(y = perm / 100, colour = "#f1a340")) +
  scale_color_identity(guide = guide_legend(title = ""), 
                       labels = c("Eigenvalues", "Permuted Eigenvalues")) +
  scale_y_continuous(labels = scales::percent) +
  xlab("MCA Eigenvalues") +
  ylab("explained variance") +
  theme_classic() +
  theme(axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        legend.position = c(0.8, 0.8))

pvalue_p <- randomize %>%
  filter(eigenvalue < 50) %>%
  ggplot(aes(x = eigenvalue, y = pvalue)) +
  geom_line() +
  geom_point() +
  geom_vline(xintercept = optimal_mca_dimensions, color = "red") +
  ggtitle(label = "", subtitle = paste0("Significant dimensions = ", optimal_mca_dimensions)) +
  xlab("MCA Eigenvalues") +
  ylab("P-value") +
  theme_classic() +
  theme(axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))

fig1B <- permut_p /
  pvalue_p 

fig1C <- ggplot(umap_cancertype %>% arrange(n_pathways_mut), 
                aes(y=plot_x, x=plot_y)) +
  geom_point(size = 0.25) +
  xlab("UMAP 1") +
  ylab("UMAP 2") +
  coord_fixed() +
  theme_classic() +
  scale_color_viridis_c() +
  labs(tag = "C") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank())

load("../data/optimal_minpts_score.rda")
fig1D <- ggplot(score_df, aes(x = minPts, y = score)) +
  geom_point(data = sample_df, color = "grey", aes(y = value), size = 0.25) +
  geom_point(size = 0.25) + 
  geom_line() +
  geom_vline(xintercept = optimal_minPts, color = "#ff0000") +
  scale_y_log10() +
  xlab("Minimum Cluster Size") +
  ggtitle(paste0("Optimal Minimum Cluster Size = ", optimal_minPts)) +
  theme_classic() +
  theme(axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))

## umap hdb_scan
fig1E <- ggplot(subset(umap_cancertype, clust !=0), aes(y = plot_x, x = plot_y, color = as.factor(clust))) +
  geom_point(data = umap_cancertype, aes(y = plot_x, x = plot_y), colour = "grey", size = 0.25) +
  geom_point(size = 0.25) + scale_color_manual(values = kelly.colours[-c(1:2)]) +
  guides(color = F) +
  theme_classic() +
  xlab("UMAP 1") +
  ylab("UMAP 2") +
  coord_fixed() +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank())

## umap knn_clust 
fig1F <- ggplot(subset(umap_cancertype), aes(y = plot_x, x = plot_y, color = as.factor(clust_knn))) +
  geom_point(size = 0.25) + scale_color_manual(values = kelly.colours[-c(1:2)]) +
  guides(color = F) +
  theme_classic() +
  xlab("UMAP 1") +
  ylab("UMAP 2") +
  coord_fixed() +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank())

## mutation burden plot
mut_order <- list() ## store data in list 
for(n in 1:max(as.numeric(as.character(umap_cancertype$clust_knn)))) { ## loop through all the clusters
  message(n)
  mut_order[[n]] <- umap_cancertype %>% filter(clust_knn == n) %>%  ## filter for specific class
    dplyr::select(sample_id, plot_x, plot_y, clust_knn, mut_count) %>% ## select relevant columns
    arrange(mut_count) ## arrange mutation in increasing order
}
mut_order <- mut_order %>% do.call(rbind, .) %>% as.data.frame() ## rbind list

mut_order$sample_id <- factor(mut_order$sample_id, levels = mut_order$sample_id)
mut_order$clust_knn <- factor(mut_order$clust_knn, levels = c(1:10))

## Need the breaks positions for the x-axis
previous_class <- 0 ## starting position 1
position_of_previous_class <- 0 ## starting position 2
xaxis_breaks <- list() ## list for for loop
for(n in 1) { ## loop through 1 to 12 clusters
  message(n)
  class_spec <- round(mut_order %>% filter(clust_knn == n) %>% nrow(.) / 2, digits = 0) ## calculate halfway mark of class 
  xaxis_breaks[[n]] <- class_spec + previous_class + position_of_previous_class ## adding up all variables
  previous_class <- class_spec ## storing class halfway mark
  for(n in 2:max(as.numeric(as.character(mut_order$clust_knn)))) {
    message(n)
    position_of_previous_class <- xaxis_breaks[[n-1]] ## getting the previous class position
    class_spec <- round(mut_order %>% filter(clust_knn == n) %>% nrow(.) / 2, digits = 0) ## calculate halfway mark of new class
    xaxis_breaks[[n]] <- class_spec + previous_class + position_of_previous_class ## adding up all variables
    previous_class <- class_spec ## storing class halfway mark
  }
}
xaxis_breaks <- as.data.frame(do.call(rbind, xaxis_breaks)) ## rbind list
xaxis_breaks <- mut_order[rownames(mut_order) %in% xaxis_breaks$V1,] ## find the tumor samples in these positions

total_n <- as.data.frame(table(mut_order$clust_knn)) ## getting total n in cluster

mut_order$log10mut <- log10(mut_order$mut_count + 1) ## mutation count log transformation + pseudo count

mut_median <- mut_order %>% 
  group_by(clust_knn) %>% 
  dplyr::summarize(mut_med = median(log10mut)) ## calculate median for each class

## Need to add 50 pseudo samples front and end of data so they don't touch the y-axis
mut_order <- mut_order[,c("sample_id", "clust_knn", "log10mut")] ## selecting relevant column

begin_NA <- data.frame(sample_id = paste0("sample", 1:100), 
                       clust_knn = min(as.numeric(mut_order$clust_knn)), ## put lowest cluster to prevent a new facet
                       log10mut = NA) ## put NA values so they don't get plotted
end_NA <- data.frame(sample_id = paste0("sample", 101:200), ## repeat the step for the ends
                       clust_knn = max(as.numeric(mut_order$clust_knn)),
                       log10mut = NA)
mut_order <- rbind(begin_NA, mut_order, end_NA) ## rbind the data in a specfic order

mut_order$sample_id <- factor(mut_order$sample_id, levels = mut_order$sample_id) ## factor samples
mut_order$clust_knn <- factor(mut_order$clust_knn, levels = c(1:10)) ## factor clusters

fig1G <- ggplot(mut_order, aes(x = sample_id, y = log10mut, color = clust_knn)) + 
  geom_point(size = 0.25) + 
  ylab("Log10(Mutation Count)") + 
  xlab("") +
  theme_classic() +
  facet_wrap(~ factor(clust_knn, levels = c(1:10)), ncol = 10, scales = "free_x") +
  geom_hline(data= mut_median, aes(yintercept=mut_med)) +
  scale_y_continuous(sec.axis = sec_axis(~.)) + ## dup y-axis
  scale_x_discrete(breaks = c(as.character(mut_order$sample_id[1]), as.character(xaxis_breaks$sample_id)), labels = c("n =" ,total_n$Freq)) +
  theme(axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks.x = element_blank(),
        axis.line.x = element_line(colour = "black", size = 0.5),
        axis.line.x.top = element_line(colour = "black", size = 0.5),
        panel.spacing = unit(1, "lines"),
        strip.background = element_rect(size = 0.5),
        strip.placement = "outside",
        strip.text.x = element_text(size = 10),
        legend.position = "none") +
  scale_color_manual(values = kelly.colours[-c(1:2)])

## code to keep only the bottom border line of the facet
fig1G <- ggplotGrob(fig1G)
lg <- linesGrob(x=unit(c(0,1),"npc"), y=unit(c(0,0),"npc"), 
                   gp=gpar(col="black", lwd=4))

for (k in grep("strip-t",fig1G$layout$name)) {
  fig1G$grobs[[k]]$grobs[[1]]$children[[1]] <- lg
}

fig1G <- as.ggplot(fig1G)

## umap by cancer type
fig1H <- ggplot(umap_cancertype[umap_cancertype$project_code %in% c("COAD", "BRCA", "HNSC", "PAAD"),], aes(y = plot_x, x = plot_y, colour = project_code)) +
  geom_point(data = umap_cancertype, aes(y = plot_x, x = plot_y), colour = "grey", size = 0.25) +
  geom_point(size = 0.25) + 
  theme_classic() +
  guides(colour = guide_legend(override.aes = list(size=3))) +
  ylab("UMAP 2") +
  xlab("UMAP 1") +
  coord_fixed() +
  labs(color = "") +
  scale_color_manual(labels = c("BRCA", "COAD", "HNSC", "PAAD"), 
                     values = c('#fb9a99', '#b15928', '#6a3d9a', 'springgreen4')) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

## umap of brca subtype
fig1I <- ggplot(umap_cancertype %>% filter(project_code == "BRCA" & !is.na(Subtype_Selected)), aes(y = plot_x, x = plot_y, colour = Subtype_Selected)) +
  geom_point(data = umap_cancertype, aes(y = plot_x, x = plot_y), colour = "grey", size = 0.25) +
  geom_point(size = 0.25) +
  theme_classic() +
  ylab("UMAP 2") +
  xlab("UMAP 1") +
  coord_fixed() +
  labs(color = "") +
  guides(colour = guide_legend(override.aes = list(size=3))) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

## umap brca somatic mutations
load("../data/gdc_gene_exp_adj.Rda")
load("../data/gene_data.Rda")
umap_cancertype <- fread("../data/umap_3d_coors.tsv")
brca <- gene_data %>% .[.$external_gene_name %in% c("BRCA1", "BRCA2"),] %>% dplyr::select(ensembl_gene_id, external_gene_name) ## Getting BRCA gene information
brca <- gdc_gene_exp_adj %>% .[rownames(.) %in% brca$ensembl_gene_id, ] %>% ## we only want brca gene
  rownames_to_column(var = "ensembl_gene_id") %>%
  left_join(., brca, by = "ensembl_gene_id") %>% 
  remove_rownames()  %>% column_to_rownames(var = "external_gene_name") %>% ## replace ensembl_id with gene_name
  dplyr::select(-ensembl_gene_id) %>% ## get ride of ensembl_id  col
  t() %>% as.data.frame() %>% ## correct orientation for umap_cancertype df
  rownames_to_column(var = "sample_id") %>% ## getting data ready for leftjoin 
  mutate(BRCA1 = case_when(BRCA1 == "1" ~ "BRCA1",
                           BRCA1 == "0" ~ "NA"),
         BRCA2 = case_when(BRCA2 == "1" ~ "BRCA2",
                           BRCA2 == "0" ~ "NA")) %>%
  mutate(mut_type = gsub('NA,|,NA', '', paste(.$BRCA1, .$BRCA2, sep = ","))) %>% ## combine mutation cause some patients can have both
  dplyr::select(-BRCA1, -BRCA2) %>% ## remove col we don't need
  left_join(., umap_cancertype[,c("sample_id", "plot_x", "plot_y", "project_code")], by = "sample_id") ## leftjoin by sample id

fig1J <- ggplot(brca %>% filter(mut_type != "NA"), aes(y = plot_x, x = plot_y, colour = mut_type)) +
  geom_point(data = brca, aes(y = plot_x, x = plot_y), colour = "grey", size = 0.25) +
  geom_point(size = 0.25) + 
  theme_classic() +
  ylab("UMAP 2") +
  xlab("UMAP 1") +
  coord_fixed() +
  labs(color = "") +
  guides(colour = guide_legend(override.aes = list(size=3))) +
  scale_color_manual(labels = c("BRCA1", "BRCA1,BRCA2", "BRCA2"), 
                     values = c('#33a02c', '#e31a1c', '#ff7f00')) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

## umap stage
fig1K <- ggplot(subset(umap_cancertype, ajcc_pathologic_tumor_stage != "Not Available"), aes(y = plot_x, x = plot_y, colour = ajcc_pathologic_tumor_stage)) +
  geom_point(data = umap_cancertype, aes(y = plot_x, x = plot_y), colour = "grey", size = 0.25) +
  geom_point(size = 0.25) + 
  theme_classic() +
  ylab("UMAP 2") +
  xlab("UMAP 1") +
  coord_fixed() +
  labs(color = "") +
  guides(colour = guide_legend(override.aes = list(size=3))) +
  scale_color_manual(labels = c("Stage I", "Stage II", "Stage III", "Stage IV"), 
                     values = c('darkgoldenrod3','dodgerblue3','yellow3','tomato2')) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom") 

## umap metastasis
fig1L <- ggplot(umap_cancertype, aes(y = plot_x, x = plot_y)) +
  geom_point(data = umap_cancertype, aes(y = plot_x, x = plot_y), colour = "grey", size = 0.25) +
  geom_point(data = subset(umap_cancertype, ajcc_metastasis_pathologic_pm == "M0"), aes(y = plot_x, x = plot_y, colour = ajcc_metastasis_pathologic_pm), size = 0.25) + 
  geom_point(data = subset(umap_cancertype, ajcc_metastasis_pathologic_pm == "M1"), aes(y = plot_x, x = plot_y, colour = ajcc_metastasis_pathologic_pm), size = 0.25) + 
  ylab("UMAP 2") +
  xlab("UMAP 1") +
  coord_fixed() +
  labs(color = "") +
  guides(colour = guide_legend(override.aes = list(size=3))) +
  scale_color_manual(labels = c("M0", "M1"), 
                     values = c('#a6cee3','#1f78b4')) +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

empty_plot <- ggplot() + theme_void()

layout <- '
AAAABBBCCCDDD
AAAABBBCCCDDD
AAAABBBEEEFFF
AAAABBBEEEFFF
AAAAGGGGGGGGG
AAAAGGGGGGGGG
AAAAGGGGGGGGG
AAAAHHHIIIJJJ
AAAAHHHIIIJJJ
AAAAKKKLLL###
AAAAKKKLLL###
'

fig1A + fig1B + fig1C + fig1D + fig1E + fig1F + fig1G + fig1H + fig1I + fig1J + fig1K + fig1L + plot_layout(design = layout) + plot_annotation(tag_levels = 'A')

```