### Classification of tumors independent of tissue-of-origin 

One of the biggest hurdles in cancer research is the sparsity of data; ~20,000
protein-coding genes is comparable with the number of tumor samples, even with
multiple mutations per sample. We sought to simplify the problem by employing a
"knowledge-base driven analysis" [@pathways-Atul-Butte], investigating cancer as
a disease of basic cellular and biochemical pathways. We accomplished this by
translating gene-level mutations into pathway level disruptions. Our approach
differs from previously described methods [@PATHNET] in that we chose to focus
on the pathway (defined in the methods) as the unit of disruption instead of the
gene, where individual mutations may be sufficient to alter pathway activity.

To our knowledge, this approach has not previously been attempted despite its
relative simplicity. We limited our analysis to mutations with likely
deleterious effects in genes that are actively expressed in each cancer type,
thus avoiding bias from transcription coupled repair. Our method of filtration
differs from the "rank-and-cut" method of [@PCAWG-MAIN] but represents a
reasonable attempt to account for the same biases. We also restricted our
analysis to biochemical pathways, excluding curated gene sets related to
diseases, syndromes, or classes of proteins with shared catalytic activity or
conserved domains which are potentially problematic [@pathways-Atul-Butte]. We
chose this approach to limit redundancy and exclude biologically unrelated
collections of genes.

The hypothesis that cancer results from dysfunction in basic cellular processes
common to eukaryotic cells was introduced and later expanded on in a pair of
essays by Hanahan and Weinberg [@weinberg2000;
@hallmarks_II]. An alternative hypothesis is that every tumor belongs to one of
a large number of syndromes which are unique to each tissue-of-origin, that
share some mechanisms and treatment strategies. Recent publication of TCGA
consortium papers present a view largely, and surprisingly, consistent with this
latter hypothesis [@hoadleyMultiplatformAnalysis2014;
@hoadleyCellofOriginPatternsDominate2018]. Perhaps owing to the intractable
complexity of genomics, proteomics, and patient metadata in all its forms, the
inescapable conclusion thus far is that tissue-of-origin remains the most
important driver of tumor characteristics at every scale and by every measure.
Our observations contrast with this view, and instead support an interpretation
of publicly available data in which all tumors manifest one of a limited number
of phenotypes resulting from disruptions of basic pathways.

We attempted to account for our clusters in terms of more trivial explanations.
For example, it could be that the clusters are consistent with disease
progression. We were unable to identify any such trend in the number of
mutations, the relative staging or metastasis; and each cluster instead was
associated with unique combinations of pathways. Some cancer types are unevenly
distributed among the clusters, though we could not identify any cancers that
were exclusive to a single cluster. Only cancer types with the fewest samples
were found to be absent from one or more clusters at all. Additionally, we annotated 
the clusters with patient attribute values (_e.g._ age of diagnosis, gender, and race) 
and were unable to identify any trends.

Somewhat surprisingly to us, this finding extends to histological subtypes of
breast, head and neck cancers, leukemias, _etc._ This result implies that
histological subtypes could reflect differences in cell-of-origin, rather than
fundamental differences in cancer phenotype. The four major subtypes of breast
cancer correspond to histological and molecular expression profiles that define
them and how they respond to experimental stimuli [@breast-subtypes2]. It has
been hypothesized that differences in molecular regulators of development in
precursor cell types present in breast epithelium drive histological phenotypes
[@breast-subtypes; @breast-subtypes3]. Consistent with this view we found that
breast tumor samples of the Luminal A subtype were heavily biased toward
membership in clusters 1 and 7, and basal tumor samples were biased toward
cluster 6. However, both subtypes also contained samples in every other cluster
(without exception), and Luminal B and Her2 positive samples are distributed
across clusters. Our interpretation of these data is that inherited
cell-of-origin signatures could predispose certain precursor cells within the
breast epithelium to forming tumors of one cluster or another but are not
determinative. This view is compatible with the previously stated hypothesis but
opens the way for a more granular view of individual tumors.

It would be surprising if we did not observe bias for some cancers and
subtypes amongst our classes, since some treatment regimens have greater
efficacy for patients of a given cancer or histological subtype
[@breast-subtypes2]. Nonetheless, the basis for some tumors being
treatment-refractory in spite of receiving the standard of clinical care for
diagnostic markers remains elusive. Doubtless some of this is due to chance
events, as tumors can metastasize and remain dormant years before they are
detected at distal sites, or resistant clones may have already arisen at
undetectable levels [@hallmarks_II], but our analysis suggests the possibility
of identifying more informative molecular, histological or cellular subtypes
that could form a basis for future stratification of patients into different
precision treatment regimens.

### Tissue specific manifestation of pathway-centric disruptions

Our results illustrate how unique combinations of mutations in pan-cancer driver
genes with tissue-specific pathway disruptions result in common categories when
viewed at the level of the pathway knowledge-base. Top cancer driver genes
(_e.g. PIK3CA_ and _TP53_) are found in most of the clusters, in spite of the
fact that they contribute to many cluster-specific pathways. This can only be
explained by less commonly mutated genes complementing the unique combination of driver
genes in each tumor, and we speculate that many of these less commonly mutated genes are
sensitive to increased mutation frequency in different tissues. For example, the most 
frequently mutated genes (_TP53_, _PIK3CA_ and _MUC16_) in cluster 2 were also frequently 
mutated in several other clusters. Yet, pathway enrichment in cluster 2 was defined 
by less commonly mutated genes in the proteasomal degradation pathway (_e.g. PSMA3_, 
_PSMA1_ or _PSME4_). The ubiquitin proteasome system maintains cellular protein 
homeostasis by regulating protein turnover. Pathway mutations found in cluster 2 
(_e.g._ “B-catenin degradation”, “cellular response to hypoxia”, “regulation of cell cycle” 
and _etc_) covered several aspects of the cancer hallmarks described by Hanahan 
and Weinberg. These hallmarks included “sustaining proliferative signaling”, “evading 
tumor suppressors” and “resisting cell death” [@Bhattacharjee2014]. Cluster 
3 was defined by less commonly mutated genes related to ribosomal subunit. Pathway 
enrichment for cluster 3 included dysregulation in “rRNA processing in the nucleus 
and cytosol” and consequently, pathways involved in eukaryotic translation and 
nonsense mediated decay. Mutations in the ribosomal subunit have been associated 
with selective translation of mRNA to upregulate proliferation [@kampen2019]. 
Pathway enrichment in cluster 3 consist of hallmarks relating to “sustaining proliferative 
signaling”, “evading tumor suppressors”, “resisting cell death”, “deregulating 
cellular energetics”, and “genome instability and mutations”. Cluster 4 was defined 
by less commonly mutated genes related to nucleosome structure that resulted in 
pathway enrichment in DNA methylation and DNA double-strand break repair pathway. 
Cluster 4 was also enriched for “telomere maintenance”, a mechanism contributing to 
cancer progression by counteracting telomere shortening. Mutations in the 
nucleosome as found in cluster 4 have been proposed to hit all attributes of hallmarks of cancer. Cluster 
7 and 8 shared multiple enrichment in signaling pathways related to the hallmarks, 
showing that some clusters share common driver mechanisms: 
“sustaining proliferative signaling”, “evading tumor suppressors”, “resisting cell 
death”, and “inducing angiogenesis”. These clusters were strongly defined by 
mutations in _PI3K_ or _RAS_, respectively, suggesting a shared etiology in the EGFR 
pathway for which these genes represent alternative intracellular signaling mechanisms. Taken together, our clusters highlight 
the similarities and differences at a gene and pathway level between and within 
different cancer types – uncovering the molecular mechanism in cancer. In the future,
we hope to explore the defining features of the clusters by connecting the pathway 
mutations to cancer hallmarks.


### Estimates of survival reveal pathway-dependent differences 

By modeling CDC longevity data as a baseline risk function we showed that each
of cluster exhibits cancer-type specific effects on survival expectancy.
However, considering that within each cancer type there are different clinical
standards of care, and even within classes of drugs the preferred treatment can
very between cancers, it makes sense that we observe tissue-specific cluster
effects. Contrast ovarian _vs._ breast cancer, which are both hormonally driven
cancers, for example. Ovarian cancer has but one main treatment axis, platinum,
whereas breast cancer patients have a variety of treatment regimens based on
molecular subtype and other factors. Unfortunately, given the diversity of drug
classes and treatments, we lack sufficient power to explore these variables in
the TCGA data. It is our hope that future studies will help to distinguish
between treatment-specific effects on survival given different pathway
disruption clusters.

### Implications for the evolution of cancer 

Our findings imply separate processes in the etiology of cancer
that can be broadly thought of as general cancer promoting, cluster-specific
mutations and metastasis. General cancer promoting processes include genome
stability and immortality, as "enabling characteristics" of the cancer phenotype
[@hallmarks_II]. Such pathways are disrupted in most clusters and are frequently
the result of aberrations involving common driver genes such as BRCA1/2, MMR
genes, mitotic checkpoints, cohesion complexes, _etc._ Cluster-specific
evolution must involve the acquisition of disruptions to pathways that may
individually be harmful (_e.g._ highly proliferative cells are more likely to
senesce) but together produce more specialized cancer phenotype and increased
fitness. Importantly, our observations do not imply the order in which these
mutations should accumulate. This could be addressed in a future study by
evolutionary analysis of clonality, drawing inference from variant allele
frequencies as in @PCAWG-EVO. However, since many of the genes in the
non-cluster-specific pathways involve the known driver genes, it is reasonable
to surmise that these mutations promote or enable acquisition of
cluster-specific defects via random mutation and natural selection, thus
producing the clusters we observed. In support of this, the pan-cancer analysis
of whole genomes consortium (PCAWG) found that oncogenic driver mutations are
highly enriched in early arising clones, whereas later arising clones have much
greater diversity in driver mutations [@PCAWG-EVO]. Moreover, driver genes that
are known to be responsible for discrete mutation signatures such as _APOBEC_,
_BRCA1_ and _BRCA2_ produce mutational hotspots reflecting varying selective
pressures in different tissues [@PCAWG-MAIN].

One of the drawbacks of bulk tumor whole genome sequencing data is the problem
of tumor heterogeneity. Consortium samples are likely to contain contamination
from support tissue, stroma, inflammatory cells, immune cells of the innate and
adaptive immune systems, all potentially harboring cancer supporting mutations
[@bhowmick_review]. We think it will be instructive to explore these ideas in
the context of single cell experiments.

### On metastasis as a convergence of phenotypes 

We report that enrichment in metastatic tumors across all clusters yielded
generally lower effect sizes and larger p-values than the cluster specific
analysis, suggesting that signal is diluted when clusters are pooled, and
supporting the view that metastasis has cluster-specific requirements. Since the
number of metastatic samples is relatively low, this part of our analysis is
likely underpowered and subject to expanded analysis with larger cohorts. The
fact that most metastatic enrichment is cluster-specific and has a tendency to
overlap with cluster-specific pathways from non-metastatic tumors of neighboring
clusters suggests that newly acquired mutations result in similar clusters
converging on one or more deadly phenotypes with critical features of end-stage
cancer. Thus, even at relatively low power, our analysis of metastasis uncovered
differences between tumors against the noisy backdrop of tissue specific
profiles. As a major caveat to our pan-cancer analysis, we acknowledge that many
therapies _do_ target highly-specific driver genes, markers, and signaling
pathways ( _e.g. TP53_, _EGFR_ or _HER2_), but understanding the broader context of
the genetic background and pathway vulnerability of tumors containing such
markers may aid in creating smarter combination therapies. We submit that when
we discover the requirements of each cluster with respect to pathway disruptions
and metastasis we may be able to target them therapeutically and prevent further
adaptation.