### Classification of tumors independent of tissue-of-origin
 
One of the biggest hurdles in cancer research is that the search space
comprising ~20,000 protein-coding genes is comparable with the number of tumor
samples, even with multiple mutations per sample. And that is not accounting for
the fact that ~30k non-coding genes and hundreds of thousands of non-coding
regulatory elements exist with oncogenic or tumor suppressor potential
[@deesMuSiC2012;
@tamboreroOncodriveCLUST2013; @lawrenceDiscoverySaturationAnalysis2014;
@kumarStatisticallyIdentifyingTumor2015; @jiangWITER2019;
@zhaoModelingPositive2019; @non-coding-mutations-Zhang]. Hence, we sought to
investigate cancer as a disease of basic cellular and biochemical pathways,
which we accomplished by translating gene-level mutations into pathway level
disruptions. Our approach differs from previously proposed methods [@PATHNET] in
that we chose to focus on pathways as the unit of disruption instead of the
gene, using the conceit that individual mutations may be sufficient to disrupt
entire pathways (as is often the case with drivers). Rather than evaluating the
burden of mutations on each pathway within or across tumors, which can be biased
by the number of genes or the coverage of pathway-linked genes in the genome,
and which we find to be relatively uniformly distributed, we scored each pathway
in binary fashion, allowing one mutation or multiple mutations to count equally
in a sample. To our knowledge, this approach has not previously been attempted
in the literature in spite of its relative simplicity.

We limited our analysis to mutations with likely deleterious effects, including
missense, nonsense, frame-shift stop-gain and splice site mutations in genes
that are actively expressed in each cancer type, thus avoiding bias from
transcription coupled repair. Our method of filtration differs from the
"rank-and-cut" method of [@PCAWG-MAIN] but represents a reasonable attempt to
account for the same biases. We also filtered genes on the basis of pathway
membership in biochemical pathways, excluding curated genesets related to
diseases, syndromes, or classes of proteins ( _e.g._ transcription factors). We
chose this approach to limit redundancy, which is already high in the remaining
pathways for some driver genes, and to exclude biologically unrelated
collections of genes. We do not account for non-coding mutations, amplification
events or structural variants, though we hope to include them in subsequent
studies (see below). Our pathway-centric analysis also allows us to circumvent
the problem that some cancer-promoting mutations likely occur at low enough
frequencies as to be indistinguishable from background. Attempts to capture
these associations better using Bayesian networks have yielded important new
insights into this problem and new genes that would not have been possible with
traditional analysis [@hornNetSigNetworkbasedDiscovery2018], supporting the case
that pathway-centric approaches like ours have the potential to increase
statistical power.

The hypothesis that cancer is the result of dysfunction from a limited number of
basic cellular processes common to all living human cells was first
introduced and later expanded on in a pair of classic essays by Hanahan and
Weinberg [@weinberg200; @hallmarks_II]. In this view, all tumors regardless of
tissue of origin acquire 6 essential phenotypes; sustaining proliferative
signaling, evading growth suppressors, replicative immortality, evading
apoptosis, supportive angiogenesis, tissue invasion and metastasis
[@weinberg2000]. Advancements in the field over the ensuing decade led to the
inclusion of new tumor promoting processes, namely deregulation of cellular
energy cycle, tumor promoting inflammation, evasion of the immune system, genome
instability, and on top of these acquired traits a new appreciation for the
concept of the tumor microenvironment and its contribution to the evolution and
promotion of metastatic tumorigenesis [@hallmarks_II]. 

An alternative hypothesis is that every tumor belongs to one of a large number
of syndromes which are unique to each tissue-of-origin, that share some
mechanisms and treatment strategies. Recent publication of TCGA consortium
papers present a view largely, and surprisingly, consistent with this hypothesis
[@hoadleyMultiplatformAnalysis2014; @hoadleyCellofOriginPatternsDominate2018].
Perhaps owing to the intractable complexity of genomic, proteomic data and
patient metadata in all its forms, the inescapable conclusion thus far is that
tissue-of-origin remains the most important driver of tumor characteristics at
every scale and by every measure. Our observations contrast with this view, and
instead support an interpretation of publicly available data in which all tumors
manifest one of a limited number of metastable phenotypes resulting from
disruptions in basic pathways represented by our classes (**figure 1a-k**). Each
of these classes may represent unique combinations of pathway disruptions with
increased fitness regardless of tissue of origin, which then converge on end
stage cancer by accumulating complementary mutations to the other classes
(**figure 1l** and **Table 2**).

We find that most tumors segregate into one of twelve biologically distinct
clusters, regardless of the cancer type (**figure 1h** and **supplementary
figure S1**). Some cancer types ( _e.g._ pancreatic adenocarcinomas in the RAS
arm clusters 9 and 10, and breast cancers being excluded from the same cluster,
see **supplementary figure S1**) are unevenly distributed, though we could not
identify cancer type that was exclusive to a single tumor class. Somewhat
surprisingly to us, this finding extends to histological subtypes of breast,
head and neck cancers, leukemias, _etc._ (see **Figure 1i** and **Supplemental
Figure S2**). Indeed, it would be surprising if we did not observe bias for some
cancers, as it has long been known that certain treatment regimens are likely to
work for a significant fraction of patients of a given cancer type. Nonetheless,
the basis for some tumors being treatment-refractory in spite of clinical
indications remains elusive. Our analysis suggests the possibility of
identifying more informative molecular, histological or cellular subtypes and
could form a basis for future separation of patients into different precision
treatment regimens.

It could be that the classes we identified are consistent with a signature of
overall disease progression, such that successive, adjacent classes on the UMAP
projection exhibit increasing specificity. We identified no such trend (see
**figure 1k**) and each class instead was associated with unique combinations of
pathways (as in **figure 3**). In our view, these observations suggest that
cancers have available to them a limited number of stable configurations with
respect to pathway disruption which are consistent with increased fitness and
therefore positive selection.

### Incompleteness of the pathway disruption data

Finally, we must remark on the limitations of our work exemplified in two of the
classes, 1 and 8, which for which we did not find distinctive associations with
pathways. Class 1 has a relatively low level of mutated pathways overall,
although it is broadly enriched in many of the same tumor-promoting pathways
common to the other groups. This likely does not reflect a true difference in
stages of tumor evolution, as our data show clearly that this class is as likely
to contain stage IV metastatic tumors as it is to contain those of stage I. We
speculate this class represents a group of tumors with large aberrations in
methylation, copy number or other genome rearrangements which we excluded from
the present analysis. In the future, we hope to incorporate these other data
into a pathway-centric analysis as we have done here for point mutations and
indels. Now that we have established this as a feasible and potentially fruitful
approach to understanding the cancer genome, it may be worth expanded effort in
this area. In addition, Class 8 represents a group of hyper-mutated tumors that
harbor so many mutations that virtually no pathway is unaffected. It seems
likely that a significant fraction of the samples flagged as “mutant” for each
pathway are actually unaffected, consistent with those samples have low
information content. Similar to class 1 we speculate that a future analysis of
other data types, including improved filtering of mutations across all classes,
may resolve these tumors into improved classifications.

### Estimates of survival reveal pathway-dependent differences

By modeling CDC longevity data as a baseline risk function we showed that each
of our pathway disruption classes exhibit cancer-type specific effects on
expected survival (**figure 5**). Contrary to what one might expect at the
outset, we found that models assuming different survival based on class _and_
cancer type fit better than models in which these variables are independent
factors. However, considering that within each cancer type there are different
clinical standards of care, and even within classes of drugs the preferred
treatment can very between cancers, it makes sense in hindsight that they should
all be different. Contrast the situation with ovarian vs breast cancer, which
are both hormonally driven cancers, for example. Ovarian cancer has but one main
treatment, platinum, whereas breast cancer patients have a variety of treatment
regimens based on molecular subtype and other factors. It is easy to see how
standard of care for one cancer and class could produce different outcomes than
the same class of a different cancer type under the circumstances.
Unfortunately, given the diversity drug classes and treatments (including
non-chemical treatments such as surgery and palliative care), we lack sufficient
power in the current study to explore these variables in the GDC data. It is our
hope that more focused future studies will help to distinguish between
treatment-specific effects or survival given different pathway disruption
classes.

### Implications for the evolution of cancer

Our findings imply that there are separate processes in the etiology of cancer
that can be broadly thought of as general cancer promoting, class-specific
mutations and metastasis. General cancer promoting processes must include
factors that relate to genome stability and immortality, as "enabling
characteristics" of the cancer phenotype [@hallmarks_II]. Such pathways are
disrupted in all or most of the defined classes (see **figures 2** and
**supplementary figure S1**) and often are the result of aberrations involving
common driver genes such as BRCA1/2, MMR genes, mitotic checkpoints, Cohesion
complexes, _etc._ (**supplementary figure S3** and **supplemental file 1**).
Class-specific evolution must involve the acquisition of combinatorial
disruptions to pathways that may individually be harmful but together produce
more specialized cancer phenotypes and increased fitness (as in **figure 4** and
**tables 1** and **ST2**). Importantly, our work does not distinguish between
the order in which these mutations should accumulate. However, since many of the
genes in the non-specific pathways involve genome stability, it is reasonable to
hypothesize that such mutations promote the acquisition of class specific
defects, and natural selection produces the classes we observed.

One of the drawbacks of bulk tumor whole genome sequencing data is the problem
of tumor heterogeneity, both in terms of evolving subclones with different or
passively diverging mutation profiles which then serve as a substrate for futher
selection [@PCAWG-MAIN; @PCAWG-EVO]. Moreover consortium samples are likely to
contain contamination from support tissue and inflammatory and immune cells of
the innate and adaptive immune systems, potentially harboring cancer supporting
mutations [@bhowmick_review]. We think it will be instructive to explore these
ideas in the context of single cell experiments.

### On metastasis as a convergence of phenotypes

We report that pathway enrichment across all classes (pooled) yielded generally
lower effect sizes and larger p-values than the cluster specific analysis,
suggesting that signal is diluted when the samples are analyzed this way, and
supporting the view that metastasis has class-specific requirements. It should
be noted, however, that in spite of this lack of global enrichment, some of the
cluster specific pathways reflect known metastatic associations, _e.g._ pathways
associated with asparagine and aspartate metabolism in class 10
[@knott_asparagus]. Since the number of metastatic tumors is relatively low this
part of our analysis is likely underpowered and subject to expanded analysis
with larger cohorts. Nonetheless, the class-specific metastatic pathway
enrichment we observe (in **table 2**) could be a result of the pathways
directly promoting metastatic phenotypes in these classes, or because they
acquire a stronger or more complete oncogenic phenotype, since (again) we don’t
know the order in which they were acquired in individual tumor samples. The fact
that most metastatic enrichment is cluster-specific and has a tendency to
overlap with cluster-specific pathways from non-metastatic tumors of neighboring
clusters favors the latter interpretation, and suggests a mechanism of
complementarity, in which newly acquired mutations result in similar classes
converging on a single deadly phenotype that captures most or all of the
critical features of end-stage cancer. This is consistent with the observation
that tumors increase diversity of driver mutations in later stages [@PCAWG-EVO].
When we discover the requirements of each cluster with respect to metastasis we
can target them therapeutically and prevent further adaptation.