---
title: "A molecular taxonomy of tumors independent of tissue-of-origin"
author: Peter T. Nguyen$^{1,\dag}$, Simon G. Coetzee$^{1,\dag}$, Daniel L. Lakeland$^{2}$, and
  Dennis J. Hazelett$^{1,3,*}$
bibliography: references.bib
output:
  word_document:
    reference_docx: "../text/word-style-reference-ttmanu.docx"
always_allow_html: TRUE
editor_options:
  chunk_output_type: console
---

###### Institutional affiliations: 1. The Center for Bioinformatics and Functional Genomics, Cedars-Sinai Medical Center, Los Angeles, California. 2. Lakeland Applied Sciences LLC, Los Angeles, CA. 3. Samuel Oschin Comprehensive Cancer Institute, Cedars-Sinai Medical Center, Los Angeles, CA. $\dag$ These authors contributed equally. $*$ To whom correspondence should be addressed.

```{r echo=FALSE, eval=FALSE}
  bookdown::pdf_book:
    number_sections: FALSE
    toc: FALSE
    keep_tex: yes

output:
  word_document:
    reference_docx: word-styles-reference-01.docx
  number_sections: FALSE
  toc: FALSE
  
output: 
bookdown::html_document2:
  fig_captions: TRUE
  number_sections: FALSE
  toc: TRUE
  toc_float: TRUE
  toc_depth: 3
  code_folding: hide
  theme: flatly
  smart: true
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, dpi=600, warning = FALSE)
suppressPackageStartupMessages(c(
  library(rstan),
  library(foreach),
  library(sqldf),
  library(tidyverse),
  library(ggthemes),
  library(data.table),
  library(biomaRt),
  library(dplyr),
  library(GenomicRanges),
  library(SummarizedExperiment),
  library(reactome.db),
  library(plyr),
  library(cluster),
  library(pheatmap),
  library(clusterProfiler),
  library(org.Hs.eg.db),
  library(dbscan),
  library(class),
  library(TCGAbiolinks),
  library(patchwork),
  library(ggplotify),
  library(ComplexHeatmap),
  library(knitr),
  library(kableExtra),
  library(UpSetR),
  library(cowplot),
  library(doRNG),
  library(uwot),
  library(janitor),
  library(FactoMineR),
  library(seriation),
  library(rtracklayer),
  library(poibin),
  library(flextable),
  library(plot3D),
  library(ggfortify),
  library(maftools),
  library(survival),
  library(ggalluvial),
  library(grid),
  library(gridBase)
))
```

## Abstract

```{r child='abstract.Rmd'}
```

## Introduction

```{r child='introduction.Rmd'}
```

## Results
 
### A taxonomy of tumors based on disrupted molecular pathways

To study cancer pathways we obtained a set of 7,607 solid tumor samples from The
Cancer Genome Atlas (TCGA) through the [Genomic Data Commons
(GDC)](gdc.cancer.gov) portal. TCGA data were most appropriate for our study
given the relative completeness of the patient metadata, particularly for
survival and staging. We chose to analyze somatic mutations in exome sequencing
because the affected target gene is known unambiguously. Therefore, we
selected all missense, nonsense, frameshift, stop-loss, untranslated region, and
splicing mutations. In order to minimize bias from well-studied diseases and
processes, we selected 377 [Reactome pathways](https://reactome.org/) (see
supplemental table ST1) of interest corresponding to basic cellular processes
and biochemical pathways, excluding gene sets that correspond to miscellaneous
categories (_e.g._ “transcription factors”) or disease associations (_e.g._
“mutated in colon cancer”) and filtered our gene list on membership in these
pathways (total of 8,940 genes; see methods for details).

To avoid bias toward larger pathways with more genes, we counted them as
disrupted if one or more member genes were mutated. We do not attempt to
calculate enrichment for mutations within a pathway. Data binarized by pathway
are likely to be noisy for several reasons. First, point mutations can be
deleterious (attenuating, hypomorphic or antimorphic) or activating (neomorphic
or hypermorphic) in genes, and these can in turn act functionally as either
oncogenes or tumor suppressors. For this study we assume a significant fraction
of these mutations are generically disruptive to normal pathway activity since
it is impossible to know the tumor promoting effects of all mutations, including
rarely studied genes. Second, we know that low- and non-expressed genes
accumulate mutations at a higher rate due to transcription coupled repair
[@kandothMutationalLandscapeSignificance2013;
@kimPancancerAnalysisSomatic2018;
@Pervasive_lesion_segregation; @expression-aware-annotation]. To address this
issue, we identified low expressed genes in each type of cancer and eliminated
them for that cancer type only (see methods for details). Highly expressed genes
could also have high mutation rates owing to transcription induced mutagenesis
[@parkGenomicEvidenceElevated2012]. This phenomenon would likely result in
cell-type-specific biases that might be biologically meaningful for
predisposition to different classes of cancer however, and therefore chose not
to exclude these genes from our analysis. After selecting our pathways and
genes, we then compiled a matrix of the pathways, assigning a Boolean value of 1
to each pathway with one or more genes mutated and 0 for all others (**figure
1**).

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 1
```{r child='../code/figure1v2.Rmd'}
```

**Figure 1: Clustering the molecular pathway profiles of tumor samples with one or more
mutations.** A) Each of 377 selected Reactome pathways (rows) is classified as
disrupted if one or more genes is mutated in the tumor sample (columns). Red
cells denote pathway disruption. Tumors types reflect standardized abbreviations
from the TCGA project. B) Different rotational perspectives of
the same MCA-based UMAP projection in three-dimensional space. Each dot
corresponds to a tumor sample. The same colors indicate the tumor's cluster
identity throughout this manuscript.

```{r, results='asis'}
cat("\\newpage")
```

We investigated this dataset further using multiple correspondence analysis
(MCA)  [@JSSv025i01], and visually summarized the analysis with UMAP (**figure
1B** and see interactive media from [**supplemental file 1**](https://junkdnalab.shinyapps.io/PANCAN_supplemental/)) [@McInnes2018]. 
We used the resulting UMAP graph coordinates to perform
density based clustering with HDBSCAN [@hdbscan], which resulted in
identification of 10 well-defined clusters capturing about 80% of the tumor
samples. To capture the remaining samples into one of these 10 clusters we used
kNN (see **Supplementary Methods** for details on clustering methods).

#### Independence from tissue-of-origin.

Having defined tumors in terms of their pathway disruption profile, we sought to
understand whether different cancer types segregate into one or more predominant
clusters. To our surprise, most cancer types were not heavily biased in one
cluster, and all well-represented cancer types had tumors in every cluster (see
**figure 2A** and full tumor profiles in **supplementary figure S1**, see also
interactive media from **supplementary file 1**), suggesting that our
pathway-disruptions identify clusters of molecular pathology largely independent
of tissue-of-origin. As an example of one type of cancer that does have a biased
pathway profile, pancreatic adenocarcinoma (PAAD) was predominantly found in
cluster 8 (**figure 2A** and **supplementary figure S1** and **supplementary
file 1**). But even PAAD comprises tumors from the nine remaining classes,
meaning that patients with these tumors have potentially different underlying
molecular pathologies.

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 2
```{r child = '../code/figure3.Rmd'}
```

**Figure 2: Pathway-based clustering independent of tissue-of-origin** 
A) Sankey plot showing correspondence between cancer type and cluster identity.
B) Projection of breast cancer subtypes onto the UMAP. C) Projection of BRCA1/2
somatic mutation onto the UMAP. D) Projection of tumor stage onto the UMAP,
regardless of cancer type. E) Projection of metastatic status onto the UMAP.
Abbrevs: M0 = non-metastatic tumors, M1 = metastatic tumors. F) Somatic mutation 
frequencies for each cluster. Vertical axis shows log mutation count, horizontal 
axis is cluster identity. Each dot represents an individual tumor sample, ranked 
lowest to highest by mutation count. The median mutation count in each cluster is 
indicated by the horizontal line.

```{r, results='asis'}
cat("\\newpage")
```

#### Independence from molecular and histological subtype.

Many cancers have molecular or histological subtypes defined based on gene
expression, pathology or other -omics profiles. These subtypes often have
different standards of care owing to different drug sensitivity (or other
factors). If the histological subtypes represent true molecular phenotypes, one
predicts that they should segregate with our pathway-based clusters, therefore
providing support for the clusters as proxies for molecular pathology
sub-typing. To our surprise, we find a similar result to the previous analysis
of cancer types projected onto the UMAP of pathway disruptions. To illustrate
this, we projected annotations for each of the breast cancer subtypes, composed
of Triple-negative/Basal-like, Her2 positive, normal-like, and luminal A and B
subtypes onto the UMAP. These are among the most heavily studied molecular
subtypes in cancer, each with different prognoses and standards of care. We did
not observe any exclusive segregation by pathway for these subtypes (**Figure
3B**). We also projected histological subtype data for the remaining cancers
(see **supplementary figure S2** and online [**supplemental file
1**](https://junkdnalab.shinyapps.io/PANCAN_supplemental/)); we find that the
subtypes, though often biased to one or more clusters, are almost never
exclusive. We interpret these data in aggregate to mean that the clusters do not
correspond to previously identified molecular subtypes within the parent cancer
types.

#### Independence from drivers of genome instability.

There are several well-known familial cancer-causing mutations that have been
studied for differences in basic biology, survival and treatment outcomes. The
functions of these genes are related to risk factors such as genome stability,
proof-reading & DNA damage repair, and telomere length. _BRCA1/2_ genes for
example are key for DNA double-stranded break repair
[@moynihan-1999-brca1;@davies-2001-brca2] and germline mutations in these genes
confer elevated risk for breast, prostate and ovarian cancers. The mechanism of
risk is thought to involve loss of heterozygosity resulting in loss of the
wildtype, functional allele [@brca-loh], so we projected the somatic mutations
for _BRCA1_ and _BRCA2_ genes onto the UMAP. We did not observe segregation of
these mutations into specific clusters (**figure 2C** and see interactive media
from [**supplemental file
1**](https://junkdnalab.shinyapps.io/PANCAN_supplemental/)). We made similar
projections for the mismatch repair (MMR) genes _MSH2_, _MSH6_, _MLH1_, _MLH3_,
_PMS1_ and _PMS2_; _BRIP1_, _RAD51_, _CHEK2_ and _APC_. None of these genes
except for _APC_ exhibited any remarkable specificity with respect to cluster
assignment (**supplementary figure S3**). To look at other risk factors such as
maintenance of DNA methylation levels and telomere length, we projected somatic
mutations of the _TET2_ and _TET3_ genes, plus _TERT_, _TEP1_, and _DKC1_, and
observed similar lack of segregation by cluster (**supplementary figure S3**).

#### Independence of stage, mutation count and mutation profile.

Tumor staging is based on physico-pathological criteria, including tumor
diameter, which can vary greatly in importance between different tissues. Stage
is used clinically as a proxy for advancement toward a more deadly state and
metastasis. Given these criteria, it is possible that more advanced tumors have
common pathway disruption profiles. The UMAP, which features a series of
lobe-like structures on a common backbone of tumor samples (**figure 1B**) could
in principle reflect progression through a series of stages. The backbone starts
with a cluster of tumors (class 1) that has the fewest point mutations and
culminates in a cluster (class 10) which has nearly every pathway disrupted
(**figure 2F**). However, outside of class 10 we don't observe a trend in the
overall mutation burden across the backbone of the UMAP. Nonetheless, to test
the hypothesis that the molecular-pathway disruption clusters represent
advancement through stages, we projected staging data onto the UMAP. Similar to
tissue of origin and other categories of tumor, we did not observe any bias
among the stages to specific clusters (**figure 2D**), suggesting that stage is
not a contributing factor to cluster identity.

Finally, as a measure of tumor advancement, metastasis is the condition in which
certain phenotypic criteria are met: loss of
differentiation, cell-cell contacts, epithelial to mesenchyme transition, immune
system evasion and tissue invasiveness [@hallmarks_II]. To determine whether any
of our clusters correspond to an especially advanced stage of cancer across
tissue types, we projected the metastases onto our UMAP, and surprisingly
we observed even distribution of the samples across clusters (**figure 2E**).
This final observation suggests that our pathway-disruption clustering is
dependent on particular combinations of gene mutations affecting different
pathways that can each give rise to advanced stages of disease and metastasis,
regardless and independent of overall mutational burden.

### Tissue specific genes define cluster membership

In order to identify pathway enrichment across all cancers (pan-cancer), we
created a list of pathway disruptions with percent mutated samples and top genes
(**supplementary table ST1**). As expected, these analyses reveal the broad
importance of many well known pathways that are disrupted in cancer, including
"PIP3 activates Akt signaling" (77% of samples), "MAP1K/MAP3K signaling" (70% of
samples), "Mitotic G2-G2/M phases" (67% of samples), "Cellular senescence" (64%
of samples), "G2/M Checkpoints" (62% of samples), _etc._

To discover what pathways are most important for clustering, we calculated
percent enrichment _within cluster_ relative to all other clusters (see methods)
and ranked pathways from highest to lowest enrichment. We visualized the
enrichment as a heatmap (**figure 3A**). Using this approach, we identified
about fourteen pathways per cluster (enrichment score ≥ 0.3, 95% confidence; see
methods) (**supplementary table ST2**). Clusters 7, 8, and 9 had several
pathways in common. To explore the specific pathways marking each cluster, we
projected disruptions for each of the 377 pathways onto the UMAP
(**supplementary figure S4** and see interactive media from [**supplemental file
1**](https://junkdnalab.shinyapps.io/PANCAN_supplemental/)). Clusters 3 and 5
were distinguished by metabolic pathways including RNA and protein biosynthesis
(**supplementary figure S4**). Similarly, cluster 4 was distinguished by
mutations affecting regulation of DNA and histone methylation ("DNA
methylation", "PRC2 methylates histones and DNA", and "Nucleosome assembly").
Clusters 7-9 have in common mutations in extracellular, intracellular, and
immune-related signaling pathways (see **figure 3B** and **supplementary figure
S4**). Cluster 2 had the highest pathway enrichment levels of the three, having
mutations in hedgehog signaling, "β-catenin degradation", "cellular response to
hypoxia", "regulation of cell cycle" and "apoptosis" among others.

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 3
```{r child = '../code/figure5.Rmd'}
```

**Figure 3: Pan-cancer enrichment of pathway disruptions.** 
A) Heatmap shows the relative enrichment of each pathway (rows) within each
numbered cluster (columns). Effect size is displayed as colors representing
percent enrichment. B) Proportion of mutated samples in the each significant
pathway (columns; union set of pathways with effect size ≥ 0.30 in each
cluster) within each numbered cluster (rows).

```{r, results='asis'}
cat("\\newpage")
```

Prior efforts to extract signatures from pan-cancer datasets met with difficulty
in distinguishing tumor samples from their tissue-specific -omics data
signatures. Given our pathway-disruption based clustering, this raises the
question, are tumor phenotypes driven entirely by common driver genes, or by
“silent” tissue-specific effectors (_i.e._ too few samples to detect above
statistical significance thresholds), or a combination of both? To answer this
question, we compared top pathway genes for each cluster relative to the TCGA
background to find differentially mutated genes. We ranked odds ratios and
selected the top ten enriched and depleted genes (pvalue < 0.01) for each
cluster (**Figure 4**; odds ratios plot). Clusters 7 and 8, which shared
multiple enrichment in signaling pathways, are largely driven by mutations in
_PI3K_ and its orthologs and _Ras_ genes, respectively (compare _PIK3CA_ and
_KRAS_ panels of **Supplementary figure S5** and see interactive media from
[**supplemental file
1**](https://junkdnalab.shinyapps.io/PANCAN_supplemental/)). Interestingly,
cluster 9, which also shared multiple enrichment in signaling pathways with
clusters 7 and 8, is enriched for both _PIK3CA_ and _KRAS_. Clusters 3 and 5,
defined by enrichment in metabolic pathways, had mutations in ribosomal proteins
and nuclear pore complex, respectively. Cluster 4 had mutations in genes
responsible for nucleosome structure. Cluster 2 had mutations in proteasomal
subunit genes involved in protein degradation. We also observed that genes that
were enriched for one cluster are depleted from others (_i.e._ is enriched in
cluster 6, but depleted in cluster 7; _PIK3CA_ is enriched in cluster 7, but
depleted in clusters 3 and 8). Next, we investigated the proportion of samples
per cancer type for the significant genes within a cluster (**Figure 4**;
heatmap). Surprisingly, cluster-specific tumors were not predominated by one or
more highly mutated genes across all cancers. Instead, when observing the
mutation rate for these genes within samples that belong to a cluster, the
mutation rate is heterogeneous across tumors by tissue origin (_e.g._ In cluster
4, CESC was enriched for _H2AFX_, OV was enriched for _HIST1H2BD_, and UCEC was
enriched for _HIST1H2AC_). Even among the top enriched genes within clusters
there is no global pattern, indicating that our clusters are not driven by
individual genes, but rather networks as a whole. Taken together, our data
identify a framework of cancer type-specific mutations associated with specific
clusters.

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 4
```{r child = '../code/figure6.Rmd'}
```

**Figure 4: Gene level analysis reveals tissue-specific class signatures.** 
Odds ratio plot; Column uses a logarithmic axis to represent odds ratio with a
95% confidence interval. Rows: significant genes from each cluster. Each cluster
was compared against the background (all other clusters) to find differentially
mutated genes. Significant genes (pvalue < 0.01) were selected and limited to
the top ten results for each cluster. Heatmap; Columns: cancer type. Rows:
significant genes. The heatmap shows proportion of samples per cluster and
cancer type mutated for each gene. Depleted significant genes (left). Enriched
significant genes (right).

```{r, results='asis'}
cat("\\newpage")
```

### Enrichment of pathways in metastasis is cluster-specific

Since metastatic tumors are distributed across all ten clusters, we first
compared them with non-metastatic tumors following the same logic we used to
investigate cluster specific enrichment. Using all non-metastatic tumors as
background, we found very low levels of enrichment (< 10%) in a handful of
pathways. We reasoned that the individual clusters might be too different to
detect global metastasis enrichment signal given the small sample size (n = 215
metastatic tumor samples).

Therefore, we calculated cluster-specific enrichment in metastatic tumors and
found a total of 31 enriched pathways (significant with enrichment score ≥ 0.3)
across all clusters (**table 1**). A number of enrichments represented pathways
that were already shown to be enriched in non-metastatic samples of other
clusters. For example, “Signaling by _PTK6_” is enriched in non-metastatic
samples of cluster 8 (see **supplementary figure S4**) but not in 7 and 9. This
pathway is enriched in metastatic tumors of clusters 7 and 9 ($p < 10^{-3}$,
**table 1**). This is also true of “Erythropoietin activates _RAS_”, which is
enriched in non-metastatic tumors of cluster 8 (**supplementary figure S4**) and
also in metastatic tumors of clusters 7 and 9. Cluster 4 metastatic samples were
enriched for “Fc epsilon receptor (FCERI) signaling”, a key neutrophil pathway,
which is also specific to clusters 2, 7 and 8 non-metastatic tumors. Thus,
metastatic-enriched pathways from one cluster are often enriched in
non-metastatic tumors of other clusters.

```{r, results='asis'}
cat("\\newpage")
```

#### Table 1
```{r child = '../code/table1.Rmd'}
```

**Table 1: Cluster-specific enriched pathways (effect size ≥ 0.30) in metastasis.**

### Pathway disruption clusters vary in short-term prognosis of survival

```{r child='survival.Rmd'}
```

## Discussion
 
```{r child='discussion.Rmd'}
```

## Materials & Methods

All code and data for producing the analyses and figures herein are included in
this fully reproducible manuscript in R markdown format. R markdown files and
all other models are available from [our
repositories](https://github.com/dennishazelett/TTmanu) on the distributed
version control site, GitHub. Extended descriptions of clustering, models,
formulae, pathway selection, and gene filtering are provided in supplementary
methods.

## Acknowledgements

We wish to thank David Van Valen, Kate Lawrenson, Simon Knott, and Megan
Hitchens for critical reading of this manuscript, and Ivetth Corona for early
discussions and feedback.

## Declarations

The authors have no conflicts of interest to declare. This work was supported by
a grant from the Cedars-Sinai Precision Health Initiative.

## References
