---
title: "A molecular taxonomy of tumors independent of tissue-of-origin"
author: Peter T. Nguyen$^{1,4}$, Simon G. Coetzee$^{1,4}$, Daniel L. Lakeland$^{2}$, and
  Dennis J. Hazelett$^{1,3,5,*}$
bibliography: references.bib
nocite: '@*'
csl: iscience.csl
output:
  word_document:
    reference_docx: "../text/word-style-reference-ttmanu-iscb-iscience-v2.docx"
always_allow_html: TRUE
editor_options:
  chunk_output_type: console
---
###### Institutional affiliations: $^{1}$The Center for Bioinformatics and Functional Genomics, Cedars-Sinai Medical Center, Los Angeles, CA, 90048, USA. $^{2}$Lakeland Applied Sciences LLC, Los Angeles, CA, 91001, USA. $^{3}$Samuel Oschin Comprehensive Cancer Institute, Cedars-Sinai Medical Center, Los Angeles, CA, 90048, USA. $^{4}$These authors contributed equally. $^{5}$Lead Contact $^{*}$Correspondence: Dennis.Hazelett@cshs.org

```{r, results='asis', echo = FALSE}
#cat("\\newpage")
```

```{r echo=FALSE, eval=FALSE}
  bookdown::pdf_book:
    number_sections: FALSE
    toc: FALSE
    keep_tex: yes

output:
  word_document:
    reference_docx: word-styles-reference-01.docx
  number_sections: FALSE
  toc: FALSE
  
output: 
bookdown::html_document2:
  fig_captions: TRUE
  number_sections: FALSE
  toc: TRUE
  toc_float: TRUE
  toc_depth: 3
  code_folding: hide
  theme: flatly
  smart: true
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, dpi=300, warning = FALSE)
suppressPackageStartupMessages(c(
  library(rstan),
  library(foreach),
  library(sqldf),
  library(tidyverse),
  library(ggthemes),
  library(data.table),
  library(biomaRt),
  library(dplyr),
  library(GenomicRanges),
  library(SummarizedExperiment),
  library(reactome.db),
  library(plyr),
  library(cluster),
  library(pheatmap),
  library(clusterProfiler),
  library(org.Hs.eg.db),
  library(dbscan),
  library(class),
  library(TCGAbiolinks),
  library(patchwork),
  library(ggplotify),
  library(ComplexHeatmap),
  library(knitr),
  library(kableExtra),
  library(UpSetR),
  library(cowplot),
  library(doRNG),
  library(uwot),
  library(janitor),
  library(FactoMineR),
  library(seriation),
  library(rtracklayer),
  library(poibin),
  library(flextable),
  library(plot3D),
  library(ggfortify),
  library(maftools),
  library(survival),
  library(ggalluvial),
  library(grid),
  library(gridBase)
))
```

## Summary

```{r child='abstract.Rmd'}
```

## Introduction

```{r child='introduction.Rmd'}
```

## Results
 
### A taxonomy of tumors based on disrupted molecular pathways

To study cancer pathways we obtained a set of 7,607 solid tumors from The Cancer
Genome Atlas (TCGA) through the Genomic Data Commons (GDC)(gdc.cancer.gov)
portal. TCGA data are most appropriate given the relative completeness of the
patient metadata for survival and staging. We chose to analyze exome sequencing
data because the affected target gene is known unambiguously. Therefore, we
selected all missense, nonsense, frameshift, stop-loss, untranslated region, and
splicing mutations. To minimize bias from well-studied diseases and processes,
we selected 377 Reactome pathways (reactome.org) (see **Table S1**) of interest 
corresponding to basic cellular processes and biochemical
pathways, excluding those corresponding to catalytic categories (_e.g._
“transcription factors”) or disease associations (_e.g._ “mutated in colon
cancer”) and filtered our gene list on membership in these pathways (total of
8,940 genes).

To avoid bias toward pathways with more genes, we counted disruptions if one or
more member genes were mutated. We do not attempt to calculate enrichment for
mutations within a pathway. Binarized pathways are likely noisy for a couple
reasons. First, point mutations can be deleterious (attenuating, hypomorphic or
antimorphic) or activating (neomorphic or hypermorphic) in genes, and these can
in turn be oncogenes or tumor suppressors. For this study we assume a
significant fraction of these mutations are generically disruptive to pathway
activity since it is impossible to know the tumor promoting effects of all
mutations, including rarely studied genes. Second, low- and non-expressed genes
accumulate mutations at a higher rate due to transcription coupled repair
[@kandothMutationalLandscapeSignificance2013;
@kimPancancerAnalysisSomatic2018;
@Pervasive_lesion_segregation; @expression-aware-annotation]. To address this
issue, we identified low expressed genes in each type of cancer and eliminated
them for that cancer type only. Highly expressed genes could also have high
mutation rates owing to transcription induced mutagenesis
[@parkGenomicEvidenceElevated2012]. This phenomenon could result in
cell-type-specific biases that might result in predisposition to different
classes of cancer however, and we therefore did not exclude these genes from our
analysis. After selecting pathways and genes, we compiled a matrix of pathways,
assigning a Boolean value of 1 to each pathway with one or more genes mutated
and 0 for all others (**Figure 1A**).

We investigated this dataset using multiple correspondence analysis (MCA)
[@JSSv025i01], and visually summarized the analysis with UMAP (**Figure 1B** and
see interactive media from **File S1** (junkdnalab.shinyapps.io/PANCAN_supplemental/)) [@McInnes2018]. We
used the resulting graph coordinates to perform density based clustering
with HDBSCAN [@hdbscan], which resulted in identification of 10 well-defined
clusters capturing about 80% of the tumor samples. To capture the remaining
samples into one of these 10 clusters we used kNN (see **STAR Methods**
for details on clustering methods).

### Independence from tissue-of-origin.

Having defined tumors in terms of pathway disruption profile, we sought to
understand whether different cancer types segregate into one or more predominant
clusters. To our surprise, most cancer types were _not_ heavily biased in one
cluster, and all well-represented cancer types had tumors in every cluster (see
**Figure 2A** and full tumor profiles in **Figure S1**,
interactive media **File S1**), suggesting that these clusters
indicate molecular pathology largely independent of tissue-of-origin. As an
example of one cancer that does have a biased pathway profile, pancreatic
adenocarcinoma (PAAD) was predominantly found in cluster 8 (**Figure 2A** and
**Figure S1** and **File S1**), but even PAAD
comprises tumors from the nine remaining classes. Thus, patients with these
tumors have potentially different underlying molecular pathologies.

### Independence from molecular and histological subtype.

Many cancers have molecular or histological subtypes defined based on gene
expression, pathology or other -omics profiles. These subtypes often have
different standards of care owing to different drug sensitivity (or other
clinical trial data). If histological subtypes represent true molecular
phenotypes, one predicts they should segregate with pathway-based clusters,
supporting the clusters as proxies for molecular pathology sub-typing. To our
surprise, we find a similar result to the previous analysis of cancer types. To
illustrate this, we projected annotations for each of the breast cancer
subtypes, composed of Triple-negative/Basal-like, Her2 positive, normal-like,
and luminal A and B subtypes onto the UMAP. These are among the most heavily
studied molecular subtypes in cancer, each with different prognoses and
standards of care. We did not observe exclusive segregation by cluster for these
subtypes (**Figure 2B**). We also projected histological subtype data for the
remaining cancers (see **Figure S2** and see interactive media from 
**File S1**); we find that
the subtypes, though often biased to one or more clusters, are almost never
exclusive. We interpret these data to mean that the clusters do not correspond
to previously identified molecular subtypes within the parent cancer types.

### Independence from drivers of genome instability.

There are several familial cancer-causing mutations studied for differences in
basic biology, survival and treatment outcomes. The functions of these genes are
related to risk factors such as genome stability, proof-reading & DNA damage
repair, and telomere length. _BRCA1/2_ genes are key for DNA double-stranded
break repair [@moynihan-1999-brca1;@davies-2001-brca2] and germline mutations in
these genes confer elevated risk for breast, prostate and ovarian cancers. The
mechanism of risk is thought to involve loss of heterozygosity, resulting in
loss of the wildtype, functional allele [@brca-loh], so we projected somatic
mutations for _BRCA1_ and _BRCA2_ genes onto the UMAP. We did not observe
segregation of these mutations into specific clusters (**Figure 2C** and see
interactive media from **File S1**). We also projected
mismatch repair (MMR) genes _MSH2_, _MSH6_, _MLH1_, _MLH3_, _PMS1_ and _PMS2_,
_BRIP1_, _RAD51_, _CHEK2_ and _APC_. None of these genes except for _APC_
exhibited remarkable specificity with respect to cluster assignment
(**Figure S3**). To look at other risk factors such as maintenance
of DNA methylation levels and telomere length, we projected somatic mutations of
the _TET2_ and _TET3_ genes, plus _TERT_, _TEP1_, and _DKC1_, and observed
similar lack of cluster bias (**Figure S3**).

### Independence of stage, mutation count and mutation profile.

Tumor staging is based on pathological criteria, including tumor diameter, which
can vary greatly in importance between different tissues. Stage is used
clinically as a proxy for advancement toward a more deadly state and metastasis.
Thus, it is possible that more advanced tumors have common pathway disruption
profiles. The UMAP, which features a series of lobe-like structures on a common
backbone of tumor samples could reflect progression through a series of stages.
The backbone starts with a cluster of tumors (class 1) that has the fewest point
mutations and culminates in a cluster (class 10) which has nearly every pathway
disrupted (**Figure 2F**). However, we don't observe a trend in the overall
mutation burden across the backbone of the UMAP. Nonetheless, to test the
hypothesis that the molecular-pathway disruption clusters represent advancement
through stages, we projected staging data onto the UMAP. Similar to tissue of
origin and other categories of tumor, we do not observe any bias among the
stages to specific clusters (**Figure 2D**), suggesting that stage does not
contribute to cluster identity.

Finally, as a measure of tumor advancement, metastasis is the condition in which
certain phenotypic criteria are met: loss of differentiation, cell-cell
contacts, epithelial to mesenchyme transition, immune system evasion and tissue
invasiveness [@hallmarks_II]. To determine whether any clusters correspond to an
especially advanced stage of cancer across tissue types, we projected the
metastases onto our UMAP, and surprisingly we observed even distribution of the
samples across clusters (**Figure 2E**). This final observation suggests that
our pathway-disruption clustering is dependent on particular combinations of
gene mutations affecting different pathways that can each give rise to advanced
stages of disease and metastasis, regardless and independent of overall
mutational burden.

### Tissue specific genes define cluster membership

In order to identify pathway enrichment across all cancers, we created a list of
pathway disruptions with percent mutated samples and top genes (**Table S1**). 
As expected, these analyses reveal the broad importance of many
well known pathways that are disrupted in cancer, including "PIP3 activates Akt
signaling" (77% of samples), "MAP1K/MAP3K signaling" (70% of samples), "Mitotic
G2-G2/M phases" (67% of samples), "Cellular senescence" (64% of samples), "G2/M
Checkpoints" (62% of samples), _etc._

To discover what pathways are most important for clustering, we calculated
percent enrichment _within cluster_ relative to all other clusters and ranked
pathways from highest to lowest enrichment. We visualized enrichment as a
heatmap (**Figure 3A**). Using this approach, we identified about fourteen
pathways per cluster (enrichment score ≥ 0.3, 95% confidence; see methods)
(**Table S2**). Clusters 7, 8, and 9 had several pathways in
common. To explore the specific pathways marking each cluster, we projected
disruptions for each of the 377 pathways onto the UMAP (**Table S2** 
and see interactive media from **File S1**). Clusters 3 and 5
were distinguished by metabolic pathways including RNA and protein biosynthesis
(**Table S2**). Similarly, cluster 4 was distinguished by
mutations affecting regulation of DNA and histone methylation ("DNA
methylation", "PRC2 methylates histones and DNA", and "Nucleosome assembly").
Clusters 7-9 have in common mutations in extracellular, intracellular, and
immune-related signaling pathways (see **Figure 3B** and **Table S2**). 
Cluster 2 had the highest pathway enrichment levels of the three, having
mutations in hedgehog signaling, "β-catenin degradation", "cellular response to
hypoxia", "regulation of cell cycle" and "apoptosis" among others.

Prior efforts to extract signatures from pan-cancer datasets met with difficulty
in distinguishing tumor samples from tissue-specific -omics data signatures.
Given our pathway-disruption based clustering, this raises the question, are
tumor phenotypes driven by common driver genes, “silent” tissue-specific
effectors (_i.e._ too few samples to detect above statistical significance
thresholds), or a combination of both? To answer this question, we compared top
pathway genes for each cluster relative to TCGA background to find
differentially mutated genes. We ranked odds ratios and selected the top ten
enriched and depleted genes (pvalue < 0.01) for each cluster (**Figure 4**; odds
ratios plot). Clusters 7 and 8, which shared multiple enrichment in signaling
pathways, are largely driven by mutations in _PI3K_ and its orthologs and _Ras_
genes, respectively (compare _PIK3CA_ and _KRAS_ panels of **Figure S4** and see 
interactive media from **File S1**. Interestingly,
cluster 9, which also shared multiple enrichment in signaling pathways with
clusters 7 and 8, is enriched for both _PIK3CA_ and _KRAS_. Clusters 3 and 5,
defined by enrichment in metabolic pathways, had mutations in ribosomal proteins
and nuclear pore complex, respectively. Cluster 4 had mutations nucleosome
structural or subunit genes. Cluster 2 had mutations in proteasomal subunits
involved in protein degradation. We also observed that genes enriched for one
cluster are depleted from others (_i.e._ _TP53_ is enriched in cluster 6, but depleted
in cluster 7; _PIK3CA_ is enriched in cluster 7, but depleted in clusters 3 and
8). Next, we investigated the proportion of samples per cancer type for the
significant genes within a cluster (**Figure 4**; heatmap). Surprisingly,
clusters were not predominated by one or more highly mutated genes across all
cancers. Instead, when observing the mutation rate for these genes within
samples that belong to a cluster, the mutation rate is heterogeneous across
tumors by tissue origin (_e.g._ In cluster 4, CESC was enriched for _H2AFX_, OV
was enriched for _HIST1H2BD_, and UCEC was enriched for _HIST1H2AC_). Even among
the top enriched genes within clusters there is no global pattern, indicating
that our clusters are not driven by individual genes, but rather networks as a
whole. Taken together, our data identify a framework of cancer type-specific
mutations associated with specific clusters.

### Enrichment of pathways in metastasis is cluster-specific

Since metastatic tumors are distributed across all ten clusters, we first
compared them with non-metastatic tumors following the logic we used to
investigate cluster specific enrichment. Using non-metastatic tumors as
background, we found very low levels of enrichment (< 10%) in a handful of
pathways. We reasoned that the individual clusters might be too different to
detect global metastasis enrichment signal given the small sample size (n = 215
metastatic tumor samples).

Therefore, we calculated cluster-specific enrichment in metastatic tumors and
found a total of 31 enriched pathways (significant with enrichment score ≥ 0.3)
across all clusters (**Table 1**). A number of enrichments represented pathways
that were already shown to be enriched in non-metastatic samples of other
clusters. For example, “Signaling by _PTK6_” is enriched in non-metastatic
samples of cluster 8 (**Table S2**) but not in 7 and 9. This
pathway is enriched in metastatic tumors of clusters 7 and 9 ($p < 10^{-3}$,
**Table 1**). This is also true of “Erythropoietin activates _RAS_”, which is
enriched in non-metastatic tumors of cluster 8 (**Table S2**) and
also in metastatic tumors of clusters 7 and 9. Cluster 4 metastases were
enriched for “Fc epsilon receptor (FCERI) signaling”, a key neutrophil pathway,
which is also specific to clusters 2, 7 and 8 non-metastatic tumors. Thus,
metastases pathways from one cluster are often enriched in non-metastases of
other clusters.

### Pathway disruption clusters vary in short-term prognosis of survival

```{r child='survival.Rmd'}
```

## Discussion
 
```{r child='discussion.Rmd'}
```

## Limitations of the Study

We must remark on the limitations of our work exemplified in 
clusters 1 and 10, for which we did not find many distinctive associations with
pathways. Cluster 1 had a relatively low proportion of mutated pathways,
although it is broadly enriched in many of the same tumor-promoting pathways
common to other clusters. In addition, our data show clearly that this cluster
is as likely to contain stage IV metastatic tumors as it is to contain those of
stage I. This cluster likely represents a group of tumors with aberrations in
methylation, copy number, or other structural variants. Consistent with this,
kidney chromophobe and thyroid cancers have high proportions of structural
variations _vs._ other variant types [@PCAWG-MAIN] and are heavily skewed to
cluster 1 membership. Likewise, cluster 10 represents a group of hyper-mutated
tumors that harbor so many mutations that virtually no pathway is unaffected. It
seems likely that a significant fraction of the “mutant” samples for each
pathway are burdened with excess passenger mutations. This could be addressed
with more sophisticated filtering of likely passenger mutations [_e.g._
@spliceAI; @deep-learning-clinical-variants]. In the future, we hope to
incorporate these other data into a comprehensive pathway-centric analysis as we
have done here for point mutations and indels.

The outcome of our model comparison with other published classification models
in predicting tdeath showed kNN performing slightly better. However, for the
purpose of predicting lifetime within this dataset, all models are practically
equivalent. One possible explanation for this is the cluster identity does not
provide much new information from tissue-of-origin. However another possibility
is that because treatment is not specialized to cluster, the treatments are not
taking advantage of potential differences between clusters which might
dramatically change survival. Most of the cluster-specific rate ($A_{cl}$)
peaked around 1 for all models – leaving the remaining variables ($A_{0}$,
$A_{r}$, and $k_{tis}$), which are tissue-specific, to compute tdeath. While our
study focused on somatic mutations, the incorporation of other types of omics
data may help differentiate the classes within cancer type. Additionally, more
work in solidifying the pathway model into cancer hallmarks could provide
biologically meaningful distinction of the clusters. Finally, if treatments can
be specialized to biological information encoded in cluster, perhaps this could
drive dramatic improvements in survival among some clusters in the future.

## Acknowledgements

We wish to thank David Van Valen, Kate Lawrenson, Simon Knott, and Megan
Hitchens, for critical comments on this manuscript, and Tim Triche Jr for
suggested improvements to the preprint. We also thank Ivetth Corona for early
discussions and feedback. Research reported in this publication was supported 
in part by Graduate Research Education at Cedars-Sinai (P.N.), by a Cedars-Sinai 
Precision Health award (D.H.), by institutional commitments to D.H., and by the
Office of the Director, National Institutes of Health under Award Number 
OT2OD026675 (D.H.).

## Author Contributions
P.N., S.G. and D.L. conducted the experiments. D.H. designed the experiments and wrote the paper.

## Declaration of Interests

The authors declare no competing interests.

## Figure and Table Legend
**Figure 1. Clustering pathways of tumor samples.** A) Each of 377 selected
Reactome pathways (rows) is classified as disrupted if one or more genes is
mutated in the tumor sample (columns). Red cells denote pathway disruption.
Tumors types reflect standardized abbreviations from the TCGA project. B)
Different rotational perspectives of the same MCA-based UMAP projection in
three-dimensional space. Each dot corresponds to a tumor sample. The same colors
indicate the tumor's cluster identity throughout this manuscript.

**Figure 2. Pathway-based clustering independent of tissue-of-origin** 
A) Sankey plot of correspondence between cancer type and cluster identity.
B) Projection of breast cancer subtypes onto the UMAP. See also Figure S1. C) Projection of BRCA1/2
somatic mutation onto the UMAP. D) Projection of tumor stage onto the UMAP,
regardless of cancer type. E) Projection of metastatic status onto the UMAP.
Abbrevs: M0 = non-metastatic tumors, M1 = metastatic tumors. F) Somatic mutation 
frequencies for each cluster. Vertical axis shows log mutation count, horizontal 
axis is cluster identity. Each dot represents an individual tumor sample, ranked 
lowest to highest by mutation count. The median mutation count in each cluster is 
indicated by the horizontal line.

**Figure 3. Pan-cancer enrichment of pathway disruptions.** A) Heatmap shows
relative enrichment of each pathway (rows) within numbered cluster (columns).
Color represents effect size as percent enrichment. B) Proportion of mutated
samples in the each significant pathway (columns; union set of pathways with
effect size ≥ 0.30 in each cluster) within cluster (rows).

**Figure 4. Gene level analysis reveals tissue-specific class signatures.** 
Odds ratio plot; Column uses a logarithmic axis to represent odds ratio with a
95% confidence interval. Rows: significant genes from each cluster. Each cluster
was compared against the background (all other clusters) to find differentially
mutated genes. Significant genes (pvalue < 0.01) were selected and limited to
the top ten results for each cluster. Heatmap; Columns: cancer type. Rows:
significant genes. The heatmap shows proportion of samples per cluster and
cancer type mutated for each gene. Depleted significant genes (left). Enriched
significant genes (right).

**Figure 5. Comparison of different cancers, simulated and actual results.**
Comparison of survival in six simulated cohorts. A) A group consisting of
randomly selected 30 year old women, unaffected (light blue) _vs._ cancer 
using the typical values from the posterior samples for a BRCA patient (yellow) 
and an artificial 10 years added to the effective age for a BRCA patient 
(dark blue), another group consisting of uniformly selected 30-70 year-old
women at random, unaffected (green) and cancer without (pink) and with (dark orange) 
the artificial 10 years. B) Posterior probability density of the effective age 
baseline ($A_{0}$) by cancer type. Red dashed line represents
age for unaffected individual, C) Posterior probability density of effective 
age rate ($A_{r}$) by cancer type. Red 
dashed line represents effective age rate for unaffected individual, D) 
Posterior probability density of the $k_{tis}$ multiplier. Values greater 
than 1 correspond to decreased age-independent life expectancy. Red dashed 
line represents $k_{tis}$ for unaffected individual. E,F,G,H) Comparison 
of our model to the actual survival for the patients in our dataset. We 
compare random group with correct age distribution for the given cancer 
(turqoise) to the model predicted survival for the given average effective 
age (blue) and the actual survival for the patients in our dataset (orange). 
Differences with actual data are likely due to a mixture of cluster 
identities which we exclude from these simulations for simplicity.

**Figure 6. Cluster specific $A_{cl}$ values for each cancer type.** The overall
effective age value is the product of $A_{0} x A_{cl} + A_{r} x (A^1 – 20)$. 
The cluster-specific A represents relative aggressiveness of each cluster within 
cancer type. Alpha transparency (α) is set to reflect varying confidence in the posterior
distribution when the number of observations is N<10, α = 0.1; N<25, α = 0.5; N
≥ 25 α = 1.0.

**Table 1: Cluster-specific enriched pathways (effect size ≥ 0.30) in metastasis.**

## STAR Methods

### KEY RESOURCES TABLE

### RESOURCES AVAILABILITY
### Lead contact
Requests for further information and resources should be directed to the Lead Contact, 
Dennis J. Hazelett (Dennis.Hazelett@cshs.org).

### Materials availbility
This study did not generate any new material.

### Data and code availability
* This paper analyzes existing, publicly available data. These datasets are listed 
in the key resources table. Data generated in this study have been deposited at Zenodo 
and are publicly available as of the date of publication. DOI is listed in the key 
resources table.
* All code for producing the analyses and figures herein are included in this fully 
reproducible manuscript in R markdown format. R markdown files and all other models are 
available from our repository on the distributed version control site, Github, listed 
in the key resources table.
* Any additional information required to reanalyze the data reported in this paper is 
available from the lead contact upon request.

### METHOD DETAILS

### Selection of pathways

To understand the molecular mechanism of cancer at a pathway level, we used
Reactome (https://reactome.org/), a knowledge-based pathway database. The
mapping files of ENSEMBL genes to pathways, pathway hierarchy relationships, and
complexes to top pathways were downloaded from
https://reactome.org/download-data. Using these data, we imposed pathway
criteria to define basic cellular processes and biochemical pathways: (1)
human-derived pathways ( _“HSA"_ ) (2) limited to grandchild node for each
parent pathway (_e.g._ ‘Beta-catenin independent WNT signaling’ in ‘Signal
Transduction’) (3) exclusion of pathways in the parent pathway: "Disease",
"Muscle contraction", and "Reproduction" or pathway names that include any of
the following keywords: "disease", "defect", "cancer", "oncogenic", "disorder",
"mutant", "loss", "infection", "bacteria", or "listeria". While some of the
excluded pathways have been shown to play an important role in cancer, they are
highly specialized (_e.g._ "PI3K/AKT Signaling in Cancer"). Additionally, for
most of the excluded pathways, a neutral version pathway of the pathway exists
(_e.g._ "PIP3 activates Akt signaling"). Finally, we mapped the 18,577
Ensembl IDs from the TCGA dataset to the highly selected Reactome pathways. This
operation produced a lookup table that consisted of 377 pathways mapped to 8,940
genes.

### Filtering genes

We filtered likely erroneous mutations due to transcription coupled repair. Our
approach was to determine the status of each gene (_i.e._ expressed or not
expressed) in each tissue type in order to exclude lowly expressed
genes. To do this, we obtained the TCGA RNA sequencing data adjusted for batch
effect dataset (https://pancanatlas.xenahubs.net). Using the data, we removed
the genes and tumor samples that were not included in our analysis, grouped the
tumor samples by tissue type and computed the mean expression value for each
gene. A minimum threshold of 10 transcripts per million was set for expressed
genes based on an inflection point observed when plotting the mean expression
values of genes ranked by expression in each tissue type. Genes that did not
meet this threshold were considered not expressed. This operation produced a
lookup table for gene expression status in each tumor sample for 18,127 genes.

### Clustering

In order to classify tumors using this dataset, we used multiple correspondence
analysis (MCA) [@JSSv025i01]. First, we determined the number of dimensions
containing useful information by selecting the eigenvalue with the most
explanatory power, using the average of 100 permutations of the data as baseline
(**Figure S6A**). We then chose the maximum eigenvalue for which the p-value
remained ≤ 0.05 (see cutoff in **Figure S6B**). Then we performed a UMAP
analysis [@McInnes2018], both in order to summarize the MCA graphically, and as
a preprocessing step to boost the performance of density based clustering. The
resulting map was notable for its lobed structure, with several reproducible
projections regardless of random seed setting. A representative version of this
3D UMAP is shown in **Figure S6C**, rotated to enhance the visibility of the
major features. Following this spatial mapping we attempted to define groupings
of similar tumors within the spatial map using HDBSCAN, which performs
hierarchical clustering and provides metrics of cluster stability and
probabilities of cluster membership for each node [@hdbscan]. However, HDBSCAN
is sensitive to several parameters; key for our analysis are the minimum number
of tumor samples in a cluster that capture the maximum number of tumor samples,
measured by probability of membership of ≥ 5% in at least one cluster. Thus, we
created a score metric as the fraction of classified tumors with max probability
< 5% in one cluster and chose a cluster size of 92 to minimize the score
function (**Figure S6D**). HDBSCAN with these settings resulted in ten distinct
high-density clusters which we then projected onto the UMAP (**Figure S6E**).
This classified 6,038 out of 7,607 tumors but still resulted in a significant
fraction of unclassified tumor samples. Since we ultimately wish to be able to
classify any tumor using this scheme, we performed k Nearest Neighbors (kNN)
analysis, which computes a similarity metric to every tumor in the set and then
lets the _k_ most similar tumors "vote" as to the identity of the query tumor
sample based on their cluster labels. We set _k_ to be the square root of the
number of tumor samples (87). Using this method, we assigned cluster membership
to the remaining tumors (**Figure S6F** and see interactive media from **File S1** 
(junkdnalab.shinyapps.io/PANCAN_supplemental/)).

### Survival

We used Bayesian inference to explore the relative impact of cancer diagnosis on
survival at the time of diagnosis. Our model assumes that at birth each person
has a small initial probability per unit time to die, and that this probability
grows exponentially in time at some baseline rate that matches the observed CDC
data. We assume that at diagnosis this information will allow us to treat
patients as if they were a different age, and with further time passing at a
different rate, a self-similarity based model. The effective age at diagnosis is
based on a baseline age for that tissue/class combination, plus a perturbation
for each year over 20 years old at diagnosis. To fit the model, we constrain the
baseline risks using the available CDC life tables (**Figure S7**). We also
constrain the cancer specific parameters using the longevity data within the
cancer tumor dataset. Our model estimates the effect of cancer type and
class-specific cancer effects independently, resulting in a cancer and class
specific estimate of the effective age.

Most TCGA cases are diagnosed late in life, so we fit a model that emphasizes
accuracy in the right tail of the distribution. We modeled baseline longevity
using a risk rate function. The risk rate of death at a time $t$ days after the
$A$ birthday is given as:

$$R(t) = R_{20}\exp(a(A-20) + D(t) k a (\frac{t}{365}))$$ 

Time is split into two components. $A$ represents the "effective age" of the
patient at diagnosis, whereas $t$ starts at 0 upon diagnosis and represents the
days since diagnosis, a field recorded in the dataset. $D(t)$
is an indicator for whether diagnosis has occurred or not (before diagnosis
$D(t) = 0$, after $D(t) = 1$). The model assumes that the risk of death is
$R_{20}$ at age 20 years and increases exponentially with a constant rate $a$ up
to the age at diagnosis. Risk of death assumes a new rate $ka$ thereafter.

Given this risk per unit time, the probability of death at time $T+dT$ is the
probability to survive to time $T$ which is $(1-P(T))$ and then die in the
remaining interval which is $R(T)dT$

$$dP = (1-P(T)) R(T) dT$$

leading to the differential equation for the cumulative probability of death
$P(T)$

$$\frac{dP}{dT} = (1-P(T))R(T)$$

The solution of this ODE is:

$$P(T) = 1 - \exp(-\int_0^T R(t) dt)$$

And the density of deaths per unit time is the derivative with respect to T:

$$p(T) = R(T) \exp(-\int_0^T R(t)dt)$$

However, to model the risk of cancer effectively we found that it was necessary
not to use the actual age $A'$ but instead the "effective age" for that
tissue/class type. Our effective age is calculated as:

$$A = A_{0} A_{cl} + A_r (A'-20)$$ Where $A_0$ is the tissue specific effective
age parameter (expressed in years), $A_{cl}$ is a dimensionless multiplier for
the given class, and $A_r$ is the dimensionless age rate which determines how
actual age $A'$ affects effective age A using a linear perturbation. For a
non-cancer patient, $A_0=20$, $A_{cl}=1$, and $A_r=1$ are the relevant
benchmarks.

The $R(T)$ function has two important parameters, the $a$ value which represents
the background risk rates, and the $k$ value, which is a function of cancer
tissue type.

$$k = k_{tis}$$

The product $A_0 A_{cl}$ is obviously symmetric between the two factors. To
disambiguate the meaning of the two, the prior for $A_0$ has high probability
range from about 40 to 100 years, whereas the prior distribution of $A_{cl}$ has
peak probability at 1 and a relatively narrow width, due to the fact that it is
a tissue specific multiplicative modifying factor. To understand the roles of
these priors refer to (**Figure S8**).

Using this model, we split the data by Male/Female sex. Sex is known to be a
risk factor for death, with males dying at slightly higher rates for all ages of
interest in our dataset.

As an aside, we did not explicitly account for stage at diagnosis in our model
for the following reasons. In the tumor sample data, stage at diagnosis is
confounded with cancer type because some cancers are screened aggressively
(_e.g._ colorectal and prostate cancers) while others are diagnosed typically
after they become problematic for the patient's lifestyle (_e.g._ ovarian and
pancreatic cancers). Secondly, such a model specification would suffer from
added noise because the staging data are not well standardized across cancer
types, have different criteria, and because it is unclear what the relationship
between stage and advancement of disease is (for example some sub-stage 4 tumors
are metastatic). To compound this latter issue, our tumor set is vastly
under-powered given the uneven representation of stage across cancer types.

### QUANTIFICATION AND STATISTICAL ANALYSIS

### Class and stage-specific Enrichment Calculations

We calculated the enrichment of pathways in one set of tumors as the relative
fraction of tumors (with estimated uncertainty) with a mutated gene in the
pathway to all tumors, inclusive of the category of interest. To do this
operation, we use the beta distribution to permute a posterior distribution on
the fraction of tumors with a pathway mutated for each category based on the
observed set, and compared this to the posterior obtained from the full set of
tumors (all tumors) as the distribution of differences between all permuted
samples. We considered a pathway enriched if the 95% range of credible
differences thus obtained excludes 0 and the mean credible difference was
greater than or equal to 30% enrichment, which excludes a large number of small
differences that are not likely to be biologically relevant.

## Supplemental Figure and Table Legends

**Figure S1. Distribution of cancer type on UMAP clusters, Related to Figure 2.** Each panel shows the identity of 
individual tumors by one of 23 cancer types projected onto the cluster UMAP identified 
in **Figure 1B of the main text**. The colors indicate the tumor's cluster identity.

**Figure S2. Histological subtypes, Related to Figure 2.** Each panel shows the histological subtypes of one of the 23 
cancers projected onto the tumors of the UMAP from **Figure 1B of the main text**. The colors 
indicate the tumor's histological subtype identity.

**Figure S3. Distribution of mismatch repair genes on UMAP clusters, Related to Figure 2.** Each panel shows one of 15 
genes involved in genome stability projected on the UMAP from **Figure 1B of the main text**. 
The colors indicate the tumor's cluster identity.

**Figure S4. Distribution of oncogenes/tumor suppressors on UMAP clusters, Related to Figure 2.** Each panel shows 
one of the twelve oncogenes/tumor suppressors projected onto the tumors of the UMAP from 
**Figure 1B of the main text**. The colors indicate the tumor's cluster identity.

**Figure S5. Performance comparison of different classification model - knn, COCA, iCluster, 
tissue-of-origin (tis), Related to Figure 5.** A) Empirical cumulative distribution function comparing time of death 
prediction of all classification models with actual time of death (dark green). B) Error 
distribution plot of predicted and actual time of death.

**Figure S6. Method for tumor classification using pathway aberration data, Related to Figure 1.** 
A) MCA analysis was performed to determine the eigenvalue with the most
explanatory power from the categorical data. Average of 100 permutations of the
data was used as a baseline. B) The p-value of each eigenvalue was plotted and
the max eigenvalue with a pvalue < 0.05 was selected. C) The resulting MCA was
projected down to 3 dimensions for the purpose of density based clustering (only
the first 2 dimensions are shown). Each dot represents a tumor D) HDBSCAN was
used to perform hierarchical density based clustering tuned to minimize the
fraction of unassigned tumors. E) HDBSCAN clustered tumors were visualized on
the UMAP projection; tumors are colored by cluster membership (see legend). This
results in some tumors remaining unclassified (grey points). F) kNN was used to
assign membership to the remaining tumors.

**Figure S7. Comparison between CDC data and ODE model fit for
males and females of European descent, Related to Figure 5.** This comparison does not distinguish on
ethnicity for simplicity in visualization.

**Figure S8. Priors for the tissue and class coefficients, Related to Figure 5 and 6.** A) The tissue
coefficient is chosen to admit ages between 40 and 100 years old as a baseline
age. B) Within a given tissue, the class parameter multiplies the tissue value and
is given a peak density at 1, and a credible range from near 0.6 to 1.4,
reflecting our expectation of a moderate multiplicative perturbation of
variation across the classes.

**Table S1. Ranked pan-cancer reactome pathway mutation, Related to Figure 1.** 
Pathways are ranked from highest to lowest percent mutated. Gene 1-3 are the top 3 
pathway genes mutated.

**Table S2. Cluster-specific enriched pathways (effect size $\ge$ 0.30), Related to Figure 3.**

**File S1. An interactive media of the results, Related to Figure 1-6.**

## References

<div id="refs"></div>
