---
title: "A molecular taxonomy of tumors independent of tissue-of-origin"
author: Peter T. Nguyen$^{1}$, Simon G. Coetzee$^{1}$, Daniel L. Lakeland$^{2}$, and
  Dennis J. Hazelett$^{1,3}$
bibliography: references.bib
output:
  word_document:
    reference_docx: "../text/word-style-reference-ttmanu.docx"
always_allow_html: TRUE
editor_options:
  chunk_output_type: console
---

###### Institutional affiliations: 1. The Center for Bioinformatics and Functional Genomics, Cedars-Sinai Medical Center, Los Angeles, California. 2. Lakeland Applied Sciences LLC, Los Angeles, CA. 3. Samuel Oschin Comprehensive Cancer Institute, Cedars-Sinai Medical Center, Los Angeles, CA.

```{r echo=FALSE, eval=FALSE}
  bookdown::pdf_book:
    number_sections: FALSE
    toc: FALSE
    keep_tex: yes

output:
  word_document:
    reference_docx: word-styles-reference-01.docx
  number_sections: FALSE
  toc: FALSE
  
output: 
bookdown::html_document2:
  fig_captions: TRUE
  number_sections: FALSE
  toc: TRUE
  toc_float: TRUE
  toc_depth: 3
  code_folding: hide
  theme: flatly
  smart: true
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, dpi=600, warning = FALSE)
suppressPackageStartupMessages(c(
  library(rstan),
  library(foreach),
  library(sqldf),
  library(tidyverse),
  library(ggthemes),
  library(data.table),
  library(biomaRt),
  library(dplyr),
  library(GenomicRanges),
  library(SummarizedExperiment),
  library(reactome.db),
  library(plyr),
  library(cluster),
  library(pheatmap),
  library(clusterProfiler),
  library(org.Hs.eg.db),
  library(dbscan),
  library(class),
  library(TCGAbiolinks),
  library(patchwork),
  library(ggplotify),
  library(ComplexHeatmap),
  library(knitr),
  library(kableExtra),
  library(UpSetR),
  library(cowplot),
  library(doRNG),
  library(uwot),
  library(janitor),
  library(FactoMineR),
  library(seriation),
  library(rtracklayer),
  library(poibin),
  library(flextable),
  library(plot3D)
))
```

## Abstract

```{r child='abstract.Rmd'}
```

## Introduction

```{r child='introduction.Rmd'}
```

## Results
 
### A taxonomy of tumors based on disrupted molecular pathways
To study cancer pathways we obtained a set of 7,607 solid tumor samples from The
Cancer Genome Atlas (TCGA) through the [Genomic Data Commons
(GDC)](gdc.cancer.gov) portal. TCGA data because were most appropriate for our
study given the relative completeness of the patient metadata, particularly for
survival and staging. We chose to start with somatic mutations in exome
sequencing data because the affected target gene is known unambiguously.
Therefore, we focused on all missense, nonsense, frameshift, stop-gain, untranslated
region, and splicing mutations. In order to minimize bias from well-studied diseases
and processes, we selected 377 Reactome pathways [reactome.org](https://reactome.org/) 
of interest corresponding to basic cellular processes and biochemical pathways,
excluding gene sets that correspond to miscellaneous categories ( _e.g._
“transcription factors”) or disease associations ( _e.g._ “mutated in colon
cancer”) and filtered our gene list on membership in these pathways (total
of 8940 genes; see methods for details).

It does not matter for our purposes whether a single tumor is “enriched” for
mutations of individual pathways, chiefly because such metrics favor larger
pathways ( _i.e._ pathways with more member genes) over smaller ones. Data
binarized by pathway are likely to be noisy for several reasons. First, point
mutations can be deleterious (attenuating or hypomorphic) or activating
(neomorphic or hypermorphic) in genes that act functionally as either activators or repressors.
Within the scope of this study we accept this uncertainty and assume mutations
are generically disruptive to the pathway regardless of whether they are
activating or deactivating or whether they affect a gene that we might classify
as one thing or another ( _e.g._ "Tumor suppressor" or "Oncogene"). Second, we
know that low-expressed genes and non-expressed genes are known to accumulate
mutations at a higher rate due to transcription coupled repair
[@kandothMutationalLandscapeSignificance2013; @kimPancancerAnalysisSomatic2018;
@Pervasive_lesion_segregation; @expression-aware-annotation]. To address this
issue, we identified genes with low expression in each type of cancer and
eliminated them for that cancer type only (see methods for details). 
<strike> Highly expressed genes could also have high mutation rates owing to
transcription induced mutagenesis [@parkGenomicEvidenceElevated2012]. We felt
that this mechanism results in cell-type-specific biases that might be
biologically meaningful for predisposition to different classes of cancer
however, and therefore chose not to exclude these genes from our analysis. After
selecting our pathways and genes, we then compiled a matrix of the pathways,
assigning a Boolean value (a 1 or 0, indicating whether any pathway member gene
carries a mutation) to the pathway regardless of the number of genes carrying
mutations (**figure 1A**). </strike>

In order to classify tumors using this dataset, we used multiple correspondence
analysis (MCA) [@JSSv025i01]. First, we determined the number of dimensions
containing useful information. (**figures 1B, C**). Then we performed a UMAP
analysis, both in order to summarize the MCA graphically, and as a preprocessing
step to boost the performance of density based clustering. The resulting map was
notable for its lobed structure, with several large, robust projections
reproducible regardless of random seed setting. A representative version of this
3D UMAP is shown in **figure 1D** as a 2D projection, rotated to enhance the
visibility of the major features. Following this spatial mapping we attempted to
define groupings of similar tumors within the spatial map using HDBSCAN, which
performs hierarchical clustering and provides metrics of cluster stability and
probabilities of cluster membership for each node. However, HDBSCAN is sensitive
to several parameters; key for our analysis are the minimum number of tumor
samples in a cluster that capture the maximum number of tumor samples, measured
by probability of membership of $\geq 5\%$ in at least one class. Thus, we
created a score metric as the fraction of classified tumors with max probability
$< 5\%$ in one class and chose a cluster size of 92 to minimize the score
function (**figure 1E**). HDBSCAN with these settings resulted in ten distinct
high-density clusters which we then projected onto the UMAP (**figure 1F**).
This classified 6,038 out of 7,607 tumors but still resulted in a
significant fraction of unclassified tumor samples. Since we ultimately wish to
be able to classify any tumor using this scheme, we performed k Nearest
Neighbors (kNN) analysis, which computes a similarity metric to every tumor in
the set and then lets the $k$ most similar tumors "vote" as to the identity of
the query tumor sample based on their classification labels. We set $k$ to be
the square root of the number of tumor samples (87). Using this method, we
assigned cluster membership to the remaining tumors (**figure 1G**).

#### Independence from tissue-of-origin.

Having defined tumors in terms of their pathway disruption profile, we sought to
understand whether different cancer types segregated into one or more
predominant classes. To our surprise, most cancer types (with some notable
exceptions) were not heavily biased in one class, and all well-represented
cancer types had representative tumors in every class (for example **figure 1I**
and full tumor profiles in **supplementary figure S1**, see also interactive
media from **supplementary file 1 (html)**), suggesting that, in principle at
least, our novel pathway-disruption classification method identifies molecular
pathologies largely independent of tissue-of-origin considerations. As an
example of one type of cancer that does have a biased pathway profile,
pancreatic adenocarcinoma (PAAD) was biased in class 8, which is
distinguished  by KRAS mutations among others (**supplementary figure S3**). 
But even PAAD comprises tumors from the nine remaining classes, meaning that 
patients hosting these tumors have potentially different underlying molecular
pathologies.

#### Independence from molecular and histological subtype.

Many cancers have molecular or histological subtypes defined based on gene
expression or other -omics profiles or pathology lab results. These subtypes
often have different standards of care owing to different overall drug
sensitivity (or other factors). If the histological subtypes represent true
molecular phenotypes, one predicts that histological subtypes should segregate
with our pathway-based classes, therefore providing support for the classes as
proxies for molecular pathology sub-typing. To our surprise, we find a similar
result to the previous analysis of cancer types projected onto the UMAP of
pathway disruptions. For example we projected annotations for each of the breast
cancer subtypes, composed of Triple-negative/Basal-like, Her2 positive,
normal-like, and luminal A and B subtypes onto the UMAP. These are among the
most heavily studied molecular subtypes in cancer, which each have different
prognoses and standards of care. We did not observe any exclusive segregation by
pathway for these subtype annotations (**Figure 1J** and **supplemental
file 1**). We also projected histological subtype data for the remaining cancers
(see **supplementary figure S2** for the full set of projections); we find that
the subtypes, though often biased towards one or more classes, are almost never
exclusive to one of the classes. We interpret these data in aggregate to mean
that our pathway disruption classes do not correspond to molecular subtypes
within the parent cancer type. Furthermore, this result raises the possibility
that, molecular and histo-pathological subtypes could represent different
precursor cell origins within each cancer type. This interpretation still leaves
open the possibility that different precursor cell types, being susceptible to
different teratogens and selection pressures ( _e.g._ insulin dependence) could
be biased toward one or more of the pathway disruption profiles.

#### Independence from drivers of genome instability.

There are several well-known familial cancer-causing mutations that have been
interrogated extensively for differences in basic biology, survival and
treatment outcomes. However, the functions of these genes are related to risk
factors such as genome stability, proof-reading and DNA damage repair, and
telomere length. _BRCA1/2_ genes for example are key for DNA double-stranded
break repair [@moynihan-1999-brca1;@davies-2001-brca2] and confer risk for
breast, prostate and ovarian cancers. The mechanism of risk is thought to
involve loss of heterozygosity [@brca-loh], so we projected the somatic
mutations for _BRCA1_ and _-2_ genes onto the UMAP, but did not observe
segregation of these mutations into specific clusters (**figure 1K**). We made
similar projections for the mismatch repair (MMR) genes _MSH2_, _MSH6_, _MLH1_,
_MLH3_, _PMS1_ and _PMS2_; _BRIP1_, _RAD51_, _CHEK2_ and _APC_. None of these genes
except for _APC_ exhibited any remarkable specificity with respect to cluster
assignment (**supplementary figure S3**). To look at other risk factors such as
maintenance of DNA methylation levels and telomere length, we projected somatic
mutations of the _TET2_ and _TET3_ genes, plus _TERT_, _TEP1_, and _DKC1_, and
observed similar lack of segregation by cluster (**supplementary figure S3**).

#### Independence of stage, mutation count and mutation profile.

Tumor staging is based on physico-pathological criteria, including diameter,
which can vary greatly in importance between different tissues. Stage is used
clinically as a proxy for advancement toward a more deadly state and metastasis.
Given these assumptions, it is possible that more advanced tumors have common
pathway disruption profiles, and the UMAP which features a series of lobe-like
structures on a common backbone of tumor samples (**figure 1D**) could in
principle also reflect progression through a series of stages. In support of
this hypothesis, the backbone starts with a cluster of tumors (class 1) that has the fewest point mutations and culminates in a cluster (class 10) which
has nearly every pathway disrupted (**figure 1H**). However, outside of class 10 we don't observe
an obvious trend in the overall mutation burden across the backbone of the UMAP (**figure
1H**). Nonetheless, to test the hypothesis that the molecular-pathway disruption
classes represent advancement through stages, we projected staging data onto the
UMAP. Similar to tissue of origin and other categories of tumor, we did not
observe any bias amongst the stages into specific pathway disruption classes
(**figure 1L**), suggesting that stage is not an determining factor in pathway
disruption classes.

Finally, as a measure of tumor advancement, metastasis is the condition in which
certain phenotypic criteria are met. These phenotypes include (among others)
loss of differentiation, cell-cell contacts, epithelial to mesenchyme
transition, immune system evasion and tissue invasiveness [@hallmarks_II]. To determine whether
any of our classes correspond to an especially advanced stage of cancer accross
tissue types, we projected the metastasis data onto our UMAP, and surprisingly we observed an even distribution across classes (**figure 1M**). This
final observation suggests that our pathway-disruption classification is
dependent on particular combinations of gene mutations that hit different
pathway profiles that can each give rise to advanced stages of disease and
metastasis, regardless and independent of overall mutational burden (see Discussion for further analysis).

### Each class is defined by a specific set of pathway disruptions

In order to visualize pathway disruption enrichment across all cancers
(pan-cancer), we created waterfall graphs analagous to the oncoprint
visualizations on cBioportal [@cbioportal], replacing genes with pathways
(**figure 2A**). A complete list of pathway disruptions with percent mutated
samples and top genes is provided in **supplementary table ST1**). As expected,
these reveal the broad importance of many well known pathways that are disrupted
in cancer, including "PIP3 activates Akt signaling" (77% of samples),
"MAP1K/MAP3K signaling" (70% of samples), "Mitotic G2-G2/M phases" (67% of
samples), "Cellular senescence" (64% of samples), "G2/M Checkpoints" (62% of
samples), _etc._ . The top third of pathways drop off gradually in overall
mutation frequency (**figure 1A** and **supplementary table ST1**), possibly
owing to relative diversity among all tumors.

To discover what pathways drive the UMAP clustering, we calculated enrichment
for each pathway _in-class_ relative to all other classes combined (see methods)
and ranked pathways for each class from highest to lowest enrichment. We
visualized the enrichment as a heatmap (**figure 2B**). Using this approach, we
identified 25 pathways highly enriched (enrichment score > 0.3, see methods) in
class 2, 9 pathways enriched in class 3, 24 pathways in class 4, 10 pathways in
class 5, 10 pathways in class 6, 24 pathways in class 7, 24 pathways in class 8
(for which most pathways are putatively disrupted), 10 pathways in class 9, and
0 pathways for clusters 1 and 10. Classes 7, 8, and 9 were highly concordant in
terms of which pathways had enrichment. To explore the specific pathways marking
each class, we projected disruptions for each of the 377 pathways onto the UMAP
(**supplementary figure S4**). Classes 3 and 5 were distinguished by metabolic
pathways including RNA and protein biosynthesis (**supplementary figure S4**).
Similarly, class 4 was distinguished by mutations affecting regulation of DNA
and histone methylation ("DNA methylation", "PRC2 methylates histones and DNA",
and "Nucleosome assembly"). Classes 7-9 have in common mutations in
extracellular, intracellular, and immune-related signaling pathways (see
**figure3** and **supplementary figure S4**). Cluster 2 had the highest pathway
enrichment levels of the three, having mutations in hedgehog signaling,
"$\beta$-catenin degradation", "cellular response to hypoxia", "regulation of
cell cycle" and "apoptosis" among others.

### A combination of common driver mutations and tissue-specific pathway genes create tissue-specific class signatures

As discussed previously, prior efforts to extract common signatures from pan-cancer 
datasets met with difficulty in distinguishing tumor samples from their tissue-specific 
-omics data signatures. Given our pathway-disruption based classification, this raises 
the question, are tumor phenotypes driven entirely by common driver genes, or by “silent”
tissue-specific effectors ( _i.e._ too few samples to detect above statistical significance 
thresholds), or a combination of both? To answer this question, we compared each cluster 
relative to the background to find deferentially mutated genes. We ranked the odd ratios
and selected the top ten enriched and depleted genes (pvalue < 0.01) for each cluster 
(**Figure 4**; odds ratios plot). Clusters 7 and 8, which shared multiple enrichment in 
signaling pathways, are largely driven by mutations in _PI3K_ and its homologs and _Ras_ 
genes, respectively (compare _PIK3CA_ and _KRAS_ panels of **Supplementary figure S5**). 
Interestingly, cluster 9, which also shared multiple enrichment in signaling pathways 
with clusters 7 and 8, is enriched for both _PIK3CA_ and _KRAS_. This result suggests 
that tumors in this cluster could be a derivative either cluster 7 or 8. Clusters 3 and 
5, defined by enrichment in metabolic pathways, had mutations in ribosomal proteins and 
nuclear pore complex, respectively. Cluster 4 had mutations in genes responsible for 
nucleosome structure. Cluster 2 had mutations in genes involved in protein degradation. 
We also observed genes that were enriched for a cluster are depleted from others ( _e.g._ 
_TP53_ is enriched in cluster 6, but depleted in cluster 7; _PIK3CA_ is enriched in cluster
7, but depleted in clusters 3 and 8). Next, we investigated the proportion of samples per 
cancer type for the significant genes within a cluster (**Figure 4**; heatmap). Surprisingly, 
cluster-specific tumors were not predominated by one or more highly mutated genes across all 
cancers. Instead, when observing the mutation rate for these genes within samples that belong
to a cluster, the mutation rate is heterogeneous across tumors by tissue origin ( _e.g._ In 
cluster 4, CESC was enriched for _H2AFX_, OV was enriched for _HIST1H2BD_, and UCEC was enriched
for _HIST1H2AC_). Even among the top most mutationally enriched genes within clusters there is
no global pattern, indicating that our classes are not driven by individual genes, but rather 
networks as a whole. Taken together, our data identifies a framework of cancer type-specific 
mutation associated with a specific cluster.  

### Enrichment of pathways in metastasis is class-specific

Following the same logic we used to investigate cluster-specific enrichment of
pathways (**figure 2**) we applied this method to metastatic vs non-metastatic
tumors, as we find that metastatic tumors are distributed across all ten clusters
( _e.g._ **figure 1m**). Using all non-metastatic tumors as background, we found
very low levels of enrichment, less than $10\%$, in a handful of pathways. We
reasoned that the individual clusters might be too different to detect global
metastasis enrichment signal given the small sample size (n = 215 metastatic
tumor samples).

Therefore, we calculated cluster-specific enrichment in metastatic tumors and
found a total of 27 enriched pathways across all clusters (**table 1**). A number of
enrichments were found in multiple clusters represented by pathways that were
already shown to be enriched in a neighboring cluster. For example,
“Signaling by PTK6” is enriched in non-metastatic samples of cluster 8 (see 
**supplementary figure S4**) but not in 7 and 9. This pathway is enriched in
metastatic tumors of clusters 7 and 9 (p < 10-3, **table 1**). This is also true of 
“Erythropoietin activates RAS”, which is enriched in cluster 8 (**supplementary figure
S4**) and also in metastatic tumors of clusters 7 and 9. <strike>Cluster 1, one of two clusters 
for which we could not identify any distinguishing pathways (cluster 8 being the other), 
had enrichment in two pathways that were otherwise specific to non-metastatic tumors of other
clusters. These include “Cellular response to hypoxia” (enriched in clusters 6 &
7) and “TNFR2 non-canonical NF-kB pathway” (cluster 6 & 7).</strike> Cluster 4 metastatic
samples was enriched for “Fc epsilon receptor (FCERI) signaling”, a key neutrophil pathways, 
which are specific to clusters 2, 7 and 8 non-metastatic tumors. Thus, pathway lesions 
specific to metastases tend toward complementary mutations that are otherwise specific to
non-metastatic tumors of other clusters.

<mark>Statement does not apply anymore. The highest # of enriched path is 3. Not many are in 
metastasis. Only 2. “Fc epsilon receptor (FCERI) signaling” and "Signaling by SCF-KIT" </mark>
<strike>There were also a number of pathways that were enriched with lower specificity
(among 4 or more clusters) which were found to be enriched in metastases. We
identified 18 such pathways of which 12 had significantly higher mutation rates
in cluster-specific metastases, including “G1/S DNA Damage Checkpoints”,
“Deubiquitination”, “Sumoylation”, “M Phase”, “RHO GTPase Effectors”, “TCF
dependent signaling in response to WNT”, “Class I MHC mediated antigen
processing & presentation”, “Chaperonin-mediated protein folding”, “Intrinsic
Pathway for Apoptosis”, “Cellular Senescence”, “DNA Double-Strand Break
Response”, and “Pre-NOTCH Expression and Processing” (**Table 1**). </strike>

### Pathway disruption classes vary in short-term prognosis of survival

```{r child='survival.Rmd'}
```

## Discussion
 
```{r child='discussion.Rmd'}
```

## Materials & Methods

All code for producing the analyses and figures herein are included in this
fully reproducible manuscript. In addition, the R markdown files and all other
models are available from
[our repositories](https://github.com/dennishazelett/TTmanu) in the following
locations on the distributed version control site GitHub

### Selection of pathways

To understand the molecular mechanism of cancer at a pathway level, our pathway
analysis approach focused on Reactome (https://reactome.org/), a pathway knowledge-based 
database. The mapping files of ENSEMBL to pathways, pathway hierarchy relationship,
and complexes to top pathways were downloaded from the download page 
(https://reactome.org/download-data). Using the data, we imposed pathway criteria to
meet the basic cellular processes and biochemical pathways: (1) human-derived pathways
(_“HSA”_) (2) limited to grandchild node for each parent pathway (_e.g._ ‘Beta-catenin 
independent WNT signaling’ in ‘Signal Transduction’) (3) exclusion of pathways in the
parent pathway: Disease, Muscle contraction, and Reproduction or pathway names that 
include any of the following keywords: disease, defect, cancer, oncogenic, disorder, 
mutant, loss, infection, bacteria, or listeria. While some of the excluded pathways 
have been shown to play an important role in cancer, they are highly specialized 
(_e.g._ ‘PI3K/AKT Signaling in Cancer’). Additionally, for most of the excluded pathways,
a neutral version pathway of the pathway exist (_e.g._ ‘PIP3 activates Akt signaling’). 
Finally, we mapped the 18,577 Ensembl IDs from the TCGA dataset to the highly selected
Reactome pathways. This operation produced a lookup table that consisted of 377 pathways
mapped to 8,940 genes.  

### Filtering genes

We filtered likely erroneous mutations due to transcription coupled repair. Our approach
was to determine the status of each gene (expressed or not expressed) per tissue type in
order to exclude gene mutations that we have evidence for no expression. To do this, we 
obtained the TCGA RNA sequencing data adjusted for batch effect dataset (https://pancanatlas.xenahubs.net).
Using the data, we removed the genes and tumor samples that were not included in our analysis, 
grouped the tumor samples by tissue type and computed the mean expression value for each 
gene. A minimum threshold of 10 transcripts per million was set for expressed genes based 
on an inflection point observed when plotting the mean expression values of genes ranked 
by increasing expression values for each tissue type. Genes that did not meet this threshold 
were considered not expressed. This operation produced a lookup table for gene expression
status in each tumor sample for 18,127 genes.  

### Class and stage-specific Enrichment Calculations

We calculated the enrichment of pathways in one set of tumors as the relative
fraction (and probability of observing) of tumors with a mutated gene in the
pathway to all tumors, inclusive of the category of interest. To do this
operation, we use the beta distribution to permute a posterior distribution on
the fraction of tumors with a pathway mutated for each category based on the
observed set, and compared this to the posterior obtained from the full set of
tumors (all tumors) as the distribution of differences between all permuted
samples. We considered a pathway enriched if the $95\%$ range of credible
differences thus obtained excludes 0 (as a null hypothesis) and the mean
credible difference was greater than or equal to $20\%$ enrichment, which
excludes a large number of small differences that are not likely to be
biologically relevant.

### Models of survival based on population-specific longevity data

```{r child='survival-methods.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```

```{r child='../text/fig_legend.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 1
```{r child='../code/figure1v2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 2
```{r child='../code/figure2v2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 3
```{r child = '../code/figure3v2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 4
```{r child = '../code/figure4v2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 5
```{r child = '../code/figure2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 6
```{r child = '../code/figure4.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
#### Table 1
```{r child = '../code/table2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```
![](../figures/fig6B.png)
\newpage
![](../figures/fig6C.png)

## Supplementary Information

```{r }
#child='../code/supplements.Rmd'
```

## References
