---
title: "TTmanuscript"
author: "Dennis Hazelett"
date: "4/18/2020"
output: html_document
bibliography: references.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



# Title: A molecular taxonomy of tumors independent of tissue-of-origin

#### Authors: Peter T. Nguyen, Simon G. Coetzee, Daniel Lakeland and Dennis J. Hazelett
 
## Abstract: 
Cancer is a disease that affects metabolism and other basic cellular processes common to all cells. In spite of this, molecular signatures remain largely indistinguishable from tissue of origin, and general, unifying theories elude the field. To address this challenge we reduced omics mutation and expression profiling data to pathway memberships. We found that tumors can be classified according to pathway disruption in twelve distinct classes orthogonal to a tissue of origin (including histological subtypes), number of mutations and clinical stage. Moreover, known germline risk mutations such as BRCA1/2 are equally distributed among the classes. One class is distinguished by mutations in extracellular receptors and their ligands, particularly RTKs including the ERBB2/4, EGFR, Insulin Receptor and IGFR. This class is neatly subdivided into PI3K/Akt and Ras branches. Of these subclasses, the PI3K/Akt subgroup also has multiple lesions in immune-related pathways, including both innate immunity and T-cell receptor signaling. Another class is typified by mutations in the cellular response to hypoxia and defects in beta-catenin and hedgehog signaling. Intriguingly, we identify a novel class of mutations in proteasome component proteins that may be sufficient to cause compound defects across multiple signaling pathways in this class. A third class is typified by defects in metabolism, especially of RNA and protein biosynthesis. In addition to pathways that distinguish between the 3 major classes of tumor, we identify common mutations across tumors and stages, as well as pathways/mutations that distinguish between metastatic tumors and non-metastatic tumors. With these observations, we present a coherent picture of global tumor biology, in which genes and pathways can generally be assigned to one of three stages: initiation, expansion, and metastasis based on what groups the mutations are found. We discuss the implications for tumor biology and treatment.
 
## Introduction:
 
Recent advances in technology have made possible the study of cancer at unprecedented levels of detail and scale. A combination of array and next-generation sequencing technologies has made it possible to assay the germline variation and somatic mutations, expression profile, methylation status, copy number variation. To take advantage of these tools, several large consortia, including the cancer genome atlas (TCGA) and pan-cancer analysis of whole genomes (PCAWG) endeavored to sequence large number of tumors (order of thousands) and collect data from multiple assays that could be analyzed together, with an eye towards integration and a general understanding of the mechanisms of cancer. These efforts have consisted chiefly of recognizing and quantifying the importance of individual genes in cancer etiology as the product of positive selection processes, or classification of cancers relative to common patterns of aberration in the expression of those genes in various assays.

Some considerable progress has been made in analyzing these data. Statistical analyses identify hundreds of global and tissue-specific cancer driver genes [@deesMuSiC2012; @tamboreroOncodriveCLUST2013; @lawrenceDiscoverySaturationAnalysis2014; @kumarStatisticallyIdentifyingTumor2015; @tokheimEvaluatingEvaluationCancer2016; @jiangWITER2019; @zhaoModelingPositive2019] using approaches aimed at detecting when genes are mutated at a greater rate than expected due to chance. Based on these studies, it has been proposed that on the order of three mutations in key oncogenes and/or tumor suppressors would be sufficient to transform a normal cell to a cancerous state [@vogelsteinPathCancerThree2015]. 

Other studies have been aimed at providing a comprehensive view of mutations, gene expression and genomic signatures, with the express goal of understanding the common themes of all cancer independent of tissue of origin. The first of these (the “Pan-Cancer-12” analysis) identified a clade of 11 subtypes from 12 cancer types, using integrative analysis with co-equal weighting of gene expression, methylation, copy number and proteomics data [@hoadleyMultiplatformAnalysis2014]. The principal finding was that tissue-of-origin is the predominant driving factor, though a significant proportion of tumors (10%) could be reclassified independent of tissue-of-origin. In a second study involving 33 cancer types and a much greater number of tumors using a newer iCluster method [@shenIntegrativeClusteringMultiple2009], the same authors identify 28 clusters that could be further related or subdivided into organ specific groups, including pan-gastrointestinal, pan-gynecological, pan-squamous, pan-gynecological/squamous and pan-kidney [@hoadleyCellofOriginPatternsDominate2018]. The authors of this study analyzed these categories for pathway enrichment and noted multiple overlapping similarities between these latter categories.

More recently, attempts have been made to consider pan-cancer mutations and other data outside the context of enrichment at the gene level. Horn et al [@hornNetSigNetworkbasedDiscovery2018] introduce approaches to address the sparseness of mutations, allowing the potential importance of a gene to be influenced by the aggregate of positive selection in its nearest network neighbors. This method proves powerful especially when a small number (on the order of hundreds) of tumors are involved. However, the larger problem of cancer etiology remains: what mutations are required, in what order, and are there specific combinations of delinquent cellular processes sufficient to cause cancer? If there is a common logic to cancer, how is it to be detected underneath the overwhelming cell-of-origin signal?

In this study, we attempted to address this challenge by focusing on discrete molecular pathways as the unit of disruption in cancer. This operation has the virtue of stripping all cell-type-specific information from the mutation data. We use the resulting statistic to evaluate all cancers and show that, surprisingly, tumors may have distinct phenotypes completely independent of tissue of origin. Furthermore, tumors of varying cancer type and/or histological subtype may merely reflect the likely cell-of-origin that gave birth to them.

## Results:
 
### Classification of tumors by pathway disruption reveals 3 major groups
 
To study cancer pathways we obtained a set of 7,607 tumor samples (x donors) from GDC. We wanted to classify tumors by pathway disruption considered different types of data available from TCGA. We chose to start with somatic mutations in exome sequencing data because the affected target gene is known. Therefore, we focused on all missense, nonsense, frameshift, and splicing mutations. We selected 381 Reactome pathways (reactome.org) of interest corresponding to basic cellular processes and biochemical pathways, excluding gene sets that correspond to miscellaneous categories (e.g. “transcription factors”) or disease associations (e.g. “mutated in colon cancer”) and filtered our gene list on membership in these pathways. We then compiled a mutation matrix of the pathways, assigning a Boolean value (a 1 or 0, indicating whether any member gene carries a mutation) to the pathway regardless of the number of genes carrying mutations.
 
In order to classify tumors using this dataset, we used unsupervised clustering methods. Preliminary work with hierarchical clustering produced a set of 15 clusters (optimized with clusterProfiler(ref) was very promising, as the tumors appeared to cluster largely independent of tissue of origin (Supplementary Figure S1A). To determine whether our clusters were differentiated from random selections of tumors we plotted Kaplan Meier survival curves for each of the clusters first, and compared the results to randomized sets of tumors of equal-sized proportions (Figure S1B, C). In addition we fitted a Cox proportional hazards test model, accounting for age of diagnosis and cluster membership and found that 8 of the clusters were significantly different from a reference cluster (cluster 1), suggesting that cluster membership could in principle be rooted in biological differences affecting survival (Supplementary Table S1). These differences were small relative to differences in cancer by tissue of origin (Figure S1D).
 
We noted intriguing similarities across different clusters, which motivated us to use dimensional reduction with UMAP (arXiv:1802.03426). Varying parameters and random seeds resulted in consistent, robust representations of the tumor field with 5 major groupings (Figure 1A). We call these tentatively by roman numerals I-V. While many of the clusters from the prior analysis remained close in proximity (note the colors in Figure 1A), some redistribution occurred, potentially reflecting important differences that could not be captured using the hierarchical method. Two of the five groupings were predominated by clusters from the hierarchical method distinguished chiefly in that one (cluster 5) had very few mutations and the other cluster was mutant for nearly every pathway (clusters 11 and 14). To determine whether the overall structure of the graph was driven by mutation counts, we examined the distribution of mutations (for pathway genes only). While there is a subtle trend, mostly visible for groups I and V (violin plot, Figure 1B), mutation count does not appear to be the dominant feature. To test whether the 5 groups represent progression by stage (which are determined clinically by size, though multiple inconsistent staging paradigms exist), we also projected these annotations onto the UMAP and found tumors of every stage, including metastatic tumors, in every grouping about equally (Figure 1C).
 
The 15 hierarchical clusters were largely free of any tissue-of-origin bias, so we suspected that our 5 groups would be similarly independent from tissue of origin. To visualize tissue of origin relative to the groups we projected the annotations for 4 of the most common of the 23 cancer types in GDC, namely breast, prostate, ovarian and pancreatic cancers onto the UMAP, and did not observe any bias to group by these tissues of origin (Figures 1D). Full projections for all 23 cancers are presented in Supplementary Figure S2. Some cancers are defined by multiple subtypes of molecular or histologically distinguishing features, which are often the basis for different treatment therapies. One hypothesis to explain the lack of bias by cancer type is that the groupings are representative of these molecular subtypes. To probe this hypothesis, we projected annotations for each of the breast cancer subtypes, composed of Triple-negative/Basal-like, Her2 positive, normal-like, and lumenal A and B subtypes onto the UMAP. Perhaps surprisingly we did not detect any group biases for these or any other subtype annotations in GDC (see Supplementary Figure S3 for the full set of projections), suggesting that these subtypes are reflective of differences either in specialized cell-type-of-origin or other cancer risk mechanisms.
 
There are several well-known familial cancer-causing mutations that have been interrogated extensively for differences in basic biology, survival and treatment outcomes. However, the functions of these genes are related to risk factors such as genome stability, proof-reading and DNA damage repair, and telomere length. BRCA1/2 genes for example are key for DNA double-stranded break repair (ref) and confer risk for breast, prostate and ovarian cancers (refs). Since the mechanism of risk is thought to involve loss of heterozygosity, we projected the somatic mutations for BRCA1 and 2 genes onto the UMAP, but did not observe any segregation of these mutations by group (Figure 1E). We made similar projections for the MMR genes MSH2, MSH6, MLH, PMS1 and PMS2; BRIP1, RAD15, CHEK2 and APC. None of these genes exhibited any remarkable specificity with respect to group (except maybe APC??) (Supplementary Figure S4). To look at other risk factors such as maintenance of DNA methylation levels and telomere length we projected somatic mutations of the TET2 and TET3 genes, plus telomerase 1, 2, 3, and observed similar lack of segregation by group (Supplementary Figure S4).
 
### Determinants of Tumor Class
 
In order to discover whether any mutations of individual pathways corresponded to the major groupings (which we will hereafter refer to as classes) evident in the UMAP projection, we plotted the tumor samples harboring mutations for each of the 381 pathways used to produce it (Supplementary Figure S5), and then calculated enrichment for each pathway in-class relative to all other classes combined and ranked pathways for each class from highest to lowest enrichment. We visualized the enrichment as a heatmap, showing only the top significant pathways by x-fold enrichment (Figure 2A). Using this approach, we identified 6? pathways associated with class II, 34 pathways primarily enriched in group III tumors, and 10 pathways enriched in class IV tumors. Neither group I nor class V had any significantly enriched pathways, perhaps low and high mutation rates in these groups, respectively. Group II was distinguished by metabolic pathways including RNA and protein biosynthesis (Figure 2B). Most class II enriched pathways were not unique to this class but shared in one or more of the other classes, suggesting that this grouping reflects either greater heterogeneity or possibly a lower stage of tumor evolution. Class III typically has mutations in extracellular signaling, intracellular signaling, and immune-related pathways (Figure 2C). Class IV had the highest pathway enrichment levels of the three, having mutations in hedgehog signaling, beta-catenin degradation, cellular response to hypoxia, regulation of cell cycle and apoptosis among others. As groups III and IV were the most distinct of the 5, we chose to investigate these two groups further.
 
Group III contained interesting sub-patterns of enrichment, which we subdivided further into class IIIa and class IIIb sequentially from right following the global naming system. Common to classes IIIa and b were 20 or so pathways representing chiefly extracellular signaling receptors, including several cancer-associated receptor tyrosine kinases (RTK) such as EGFR, IGF1R, Insulin receptor, and ERBB2 and 4. In addition to these we observed SCF-Kit, DAP12, and Tie 2 signaling pathways, altogether suggesting that Class III tumors share mutations in extracellular signaling pathways generally. Since pathways like these may share many common components, we investigated whether tumors in this class typically have multiple mutations across signaling pathways, or whether a small number of critical shared components was driving the enrichment. We plotted mutations at the gene level, grouped in a supervised manner by pathway membership, and then performed unsupervised clustering of the tumors (Figure 3 or supplement figure?). We find….(please do this analysis)
 
Intriguingly, we noticed that the pathways enriched for class IIIa and b were related to RAS and PI3K/Akt signaling, respectively, which potentially represent the downstream components of the extracellular signaling pathways discussed above. Moreover, class IIIb is also enriched in several immune-associated pathways including for both innate immunity and T cell receptor signaling, suggesting an interesting correlation between the PI3K/Akt subgroup and potential loss (or hyperactivation) of immune function. To address this possibility, we isolated the PI3K/Akt, RAS and immune/TCR pathways and assessed whether the mutations tended to be biased toward likely inactivating or activating mutations ….
 
Class IV exhibited the highest pathway enrichment scores of the three major categories, though there is a significant number of tumors between class III and IV on the UMAP projection that have pathway characteristics of both. The top ten distinguishing pathways of Class IV tumors were cellular response to hypoxia, regulation of mitotic cell cycle, assembly of the pre-replicative complex, metabolism of polyamines, degradation of beta-catenin by the destruction complex, tumor necrosis factor receptor 2 non-canonical NF-kB pathway, AUF1 (hnRNP D0) binds and destabilizes RNA, regulation of apoptosis and 3 pathways associated with aspects of hedgehog signaling. To determine once again whether the genes driving pathway enrichment were commonly shared among these pathways we plotted and performed unsupervised clustering on tumors with an individual mutation matrix using every gene in each of the ten pathways, grouped by shared pathway membership (Figure 4A?). Surprisingly, this class of tumors fell into two major categories. The first contains many mutations, in nearly all of the hallmark pathways of this cluster, and the second subgroup contains far fewer mutations, but typically at least one mutation of any one of a large number of proteasome genes that happen to be common to the other 9 pathways. Moreover, it is apparent that some of these proteasome genes have a typically tissue-of-origin (or cancer-specific) mutation signature, suggesting a) that proteasome mutations could potentially substitute for a handful of more targeted mutations to these other pathways and b) that this could represent a novel class of cancer-promoting mutations that is obscured by virtue of having an abundance of targets with tissue-specific expression.
 
To further investigate this latter hypothesis, we examined the expression of the proteasome genes in this cohort of tumors, relating expression to percent of mutated samples by cancer (Figure 4B). We found a significant positive association between number of mutated samples (I’m making this up now, you have to determine whether it’s true and correct the record) and mean expression in the cancers by tissue-of-origin, thus confirming our suspicion that the proteasomal genes in class IV represent a novel class of tissue-specific specific cancer-promoting genes with a potential common mechanism.
 
### Shared mutated genes/pathways

### Oncoprint stuff
 
### Determinants of Metastatic vs non-metastatic tumors

Following the same logic we used to investigate cluster-specific enrichment of pathways (Figure 2) we applied this method to metastatic vs non-metastatic tumors, as we find that metastatic tumors are distributed across all 12 clusters (Figure 1C). Using all non-metastatic tumors as our background sample, we found very low levels of enrichment, less than 10%, in a handful of pathways (not shown). We reasoned that the clusters might be different enough to dilute cluster-specific enrichment. 

We, therefore, calculated cluster-specific enrichment in metastatic tumors and found 99 enrichments across all clusters (Table 1). A number of metastatic enrichments were found in multiple clusters represented by pathways that were already enriched in a neighboring cluster. For example, “Beta-catenin independent WNT signaling” is enriched in non-metastatic samples of clusters 6, 7, 11 and 12 (see supplemental figure S5) but not in 9 and 10. This pathway is enriched in metastatic tumors of clusters 9 and 10, plus the union of 9+10 (p < 10-3, Table 1). This is also true of “Erythropoietin activates RAS”, which is enriched in clusters 11 and 12 (Figure 2) and also in metastatic tumors of the sister groups 9 and 10, plus cluster 7. Cluster 1, one of two clusters for which we could not identify any distinguishing pathways (cluster 8 being the other), had enrichment in two pathways that were otherwise specific to non-metastatic tumors of other clusters. These include “Cellular response to hypoxia” (enriched in clusters 6 & 7) and “TNFR2 non-canonical NF-kB pathway” (cluster 6 & 7). Cluster 5 metastatic samples were enriched for two key neutrophil pathways, “Neutrophil Degranulation” and “Fc epsilon receptor (FCERI) signaling” which are specific to cluster 9 & 10 non-metastatic tumors. Thus, pathway lesions specific to metastases tend toward complementary mutations that are otherwise specific to non-metastatic tumors of other clusters. 

There were also a number of pathways that were enriched with lower specificity (among 4 or more clusters) which were found to be enriched in metastases. We identified 18 such pathways, and of these 12 had significantly higher mutation rates in cluster-specific metastases, including “G1/S DNA Damage Checkpoints”, “Deubiquitination”, “Sumoylation”, “M Phase”, “RHO GTPase Effectors”, “TCF dependent signaling in response to WNT”, “Class I MHC mediated antigen processing & presentation”, “Chaperonin-mediated protein folding”, “Intrinsic Pathway for Apoptosis”, “Cellular Senescence”, “DNA Double-Strand Break Response”, and “Pre-NOTCH Expression and Processing”. Of these pathways, we identified mutations in Sumoylation for 100% of the metastatic samples of clusters 3, 6, and 7 (Table 1). 

The enrichment we observe may be consistent with an interpretation in which the pathways themselves are promoting a metastatic phenotype, or at the very least that tumors bearing such mutations are more likely to turn metastatic since we don’t know the order in which they were acquired. The fact that many of the metastasis enrichments are cluster-specific and have a tendency to overlap with cluster-specific pathways from non-metastatic tumors of other clusters in a complementary fashion is suggestive that there is an order of progression with the metastatically enriched pathways representing later events.
 
### Correlation of microenvironment with tumor class
 
## Materials & Methods

## Discussion
 
### Classification of tumors independent of tissue-of-origin
 
The basic problem in tumor research of this kind, namely that the search space, ~20,000 protein-coding genes is greater than the number of tumor samples [@hornNetSigNetworkbasedDiscovery2018]. And that is not accounting for the fact that >30k non-coding genes and hundreds of thousands of non-coding regulatory elements exist with oncogenic or tumor suppressor potential. The bioinformatic challenge therefore is to define the problem precisely and then create a new dataset that addresses this problem. Hence, we define cancer as a malady of biochemical processes and pathways, and generated a dataset that scores each of these essential pathways as mutated or not. 

It is important that we chose point mutations in exome sequencing data for this task because we know unambiguously in most cases the target gene. In addition, we reduce pathway state to a boolean value because it does not matter for our purposes whether a single tumor is “enriched” for pathway mutations, chiefly because such metrics favor larger pathways over smaller ones. The data are likely to be noisy for several reasons. First, point mutations can be deleterious (attenuating or hypomorphic) or activating (neomorphic) in genes that are potentially either activators or repressors. In the current model, we accept this uncertainty and assume mutations are generically disruptive to the pathway. Second, low-expressed genes and non-expressed genes are known to accumulate mutations at a higher rate [@kandothMutationalLandscapeSignificance2013; @kimPancancerAnalysisSomatic2018]. To address this issue, we identified genes with low expression average for each type of cancer and eliminated them for that cancer only. Highly expressed genes also have high mutation rates owing to transcription induced mutagenesis [@parkGenomicEvidenceElevated2012]. We felt that this mechanism results in cell-type-specific biases that might be biologically meaningful for predisposition to different classes of cancer however, and therefore chose to include these genes in our analysis.
 
We hypothesize that cancer is the result of dysfunction of a limited number of basic cellular processes that are common to all living human cells, and thus all tumors belong on a spectrum of mutations regardless of tissue of origin. An alternative hypothesis is that every tumor belongs to one of a large number of syndromes that share some mechanisms and superficial similarity in treatment strategies. We refer to these as the “single disease” and the “multiple syndrome” hypotheses. The latter seems to have gained momentum in recent years since the publication of Encode, Roadmap and TCGA consortium papers, perhaps owing to the intractable complexity of genomic data in all its forms. The inescapable conclusion thus far is that all these data point to tissue-of-origin as the most important driver of tumor behavior. Our observations support the view of all tumors as manifestations of disease of cellular differentiation and metabolism, but by no means refute entirely the multiple syndrome hypothesis. 

We find that most tumors segregate into one of (at least) three biologically distinct clusters, regardless of the cancer type. Some cancer types (e.g. Pancreatic adenocarcinomas in the RAS arm cluster IIIa, and breast cancers being excluded from the same cluster) are unevenly distributed toward a specific class, we could not identify any type of cancer that was exclusive to a tumor class. Somewhat surprisingly to us, this finding extends to histological subtypes of breast, head and neck cancers, leukemias, etc. (see Figure 1E and Supplemental Figure S3). Indeed, it would be surprising if we did not observe bias for some cancers, as it has long been known that certain treatment regimens are likely to work for a majority of patients of a given cancer type. Nonetheless, the basis for some tumors being treatment-refractory in spite of clinical indications remains elusive. If supported by future studies, our analysis suggests general mechanisms underlying tumor evolution regardless of tissue or histological or cellular subtype and may form the basis for future separation of patients into different treatment regimens.

It could be that the classes we identified are consistent with a signature of overall disease progression, such that successive, adjacent classes on the UMAP projection exhibit increasing specificity. We identified no such trend and each class instead was associated with unique combinations of pathways. In class III--which contains mutations in extracellular signaling components--we identified subgroups (class IIIa and IIIb) which correspond to the different intracellular (downstream) components of the signaling pathways. Also consistent with the groups reflecting basic differences in biology rather than stages of tumor evolution, we found that our classifications were largely independent of the number of mutations or clinical stage (Figure 1B & C). In our view, these observations suggest that cancers have available to them at least three separate mechanisms to escape the selective pressures they are exposed to in situ.

Finally, we must remark on two of the classes, I and V, which for which we did not find distinctive associations with pathways. Class I has a relatively low level of mutated pathways overall, although it is broadly enriched in many of the same tumor-promoting pathways common to the other groups. We don’t believe this to be due to a true difference in stage of tumor evolution, as our data show clearly that this class is as likely to contain stage IV as it is to contain stage I tumors. We speculate this class represents a group of tumors with aberrations in methylation, gene expression, copy number or other genome rearrangements which we excluded from the present analysis. In the future, we hope to find ways to incorporate these other data types into a pathway-centric analysis as we have done here for point mutations and indels. Class V represents a group of hypermutated tumors that harbor so many mutations that virtually no pathway is unaffected. This group highlights one of the limitations of the approach that we’ve taken, in that it is likely that a significant fraction of the samples flagged as “mutant” for each pathway are actually unaffected, and thus if the mutation rate is high enough, those samples have low information content. Similar to class I we speculate that a future analysis of other data types may resolve these tumors into more meaningful categories. 
 
### Risk genes and tumor class
 
Certain mutations in risk genes increase risk for cancer quite dramatically. For example, carriers of mutations in BRCA1 and BRCA2 have increased risk for breast, ovarian, and prostate cancers (refs). The mechanism of risk in such cases is thought to be due to loss of heterozygosity (LOH) resulting in complete or near-complete loss of function(refs). In some cases it is known that carriers have a lower rate of chemo-resistance than non-carriers(refs), begging the question of whether carriers with LOH are exposed to distinctive selective pressure resulting in a constrained evolutionary outcome. If this hypothesis were true, we should have observed somatic mutations in BRCA1/2 genes clustering within one of our classes. Instead, BRCA mutations were evenly distributed amongst the classes (as in Figure 1E), suggesting that treatments or outcomes that distinguish between carriers and the general population are specific to BRCA gene biology. BRCA1/2 encode proteins whose function is involved in double-stranded break repair, and we speculate that genes involved in this and other aspects of genome stability are therefore upstream of the later events in tumor evolution that distinguish between classes II-IV. This further provides hope for combination therapies that account for carrier status and tumor class separately.
 
### (Include discussions of other germline LOH)
 
Our findings imply that there are separate processes in the etiology of cancer that can be broadly thought of as initiation, acquisition of cancer and metastasis. Initiation processes must include factors that relate to genome stability either directly or indirectly. Acquisition genes relate to selective pressures that all cells face; a complex web of signaling pathways governs cell division and migration in normal healthy cells, s.th. any cells that turn these pathways on in the absence of signal have a selective advantage. It is no wonder then that hormone-driven cancers coincide with menopause and advancement beyond middle age. With regard to metastasis….

### Two-step and 3-strike

## References: