---
title: "A molecular taxonomy of tumors independent of tissue-of-origin"
author: "Peter T. Nguyen$^{1}$, Simon G. Coetzee$^{1}$, Daniel L. Lakeland$^{2}$, and Dennis J. Hazelett$^{1,3}$"
output: 
  bookdown::html_document2:
    fig_captions: TRUE
    number_sections: FALSE
    toc: TRUE
    toc_float: TRUE
    toc_depth: 3
    code_folding: hide
    theme: flatly
    smart: true
bibliography: references.bib
editor_options: 
  chunk_output_type: console
---

###### Institutional affiliations: 1. The Center for Bioinformatics and Functional Genomics, Cedars-Sinai Medical Center, Los Angeles, California. 2. Lakeland Applied Sciences LLC, Los Angeles, CA. 3. Samuel Oschin Comprehensive Cancer Institute, Cedars-Sinai Medical Center, Los Angeles, CA.

```{r echo=FALSE, eval=FALSE}
  bookdown::pdf_book:
    number_sections: FALSE
    toc: FALSE
    keep_tex: yes
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
#library(rstan)
library(foreach)
library(sqldf)
library(tidyverse)
library(ggthemes)
library(data.table)
library(biomaRt)
library(dplyr)
library(GenomicRanges)
library(SummarizedExperiment)
library(reactome.db)
library(plyr)
library(cluster)
library(pheatmap)
library(clusterProfiler)
library(org.Hs.eg.db)
library(umap)
library(dbscan)
library(class)
library(TCGAbiolinks)
library(patchwork)
library(ggplotify)
library(ComplexHeatmap)
library(knitr)
library(kableExtra)
library(UpSetR)
```

## Abstract

```{r child='abstract.Rmd'}
```

## Introduction

```{r child='introduction.Rmd'}
```

## Results
 
### A taxonomy of tumors based on disrupted molecular pathways
 
To study cancer pathways we obtained a set of 7,607 tumor samples from GDC. We
chose to start with somatic mutations in exome sequencing data because the
affected gene is known almost unambiguously. Therefore, we focused on all
missense, nonsense, frameshift, and splicing mutations. In order to minimize
bias from well-studied diseases and processes, we selected 377 Reactome pathways
[reactome.org](https://reactome.org/) of interest corresponding to basic
cellular processes and biochemical pathways, excluding gene sets that correspond
to miscellaneous categories ( _e.g._ “transcription factors”) or disease
associations ( _e.g._ “mutated in colon cancer”) and filtered our gene list on
membership in these pathways (<mark>total of x genes;</mark> see methods for
details).

It does not matter for our purposes whether a single tumor is “enriched” for
multiple pathway mutations, chiefly because such metrics favor larger pathways (
_i.e._ pathways with more member genes) over smaller ones, and non-expressed
genes over expressed ones due to transcription coupled repair
[@kandothMutationalLandscapeSignificance2013; @kimPancancerAnalysisSomatic2018;
@Pervasive_lesion_segregation]. Data selected in this manner are likely to be
noisy for several reasons. First, point mutations can be deleterious
(attenuating or hypomorphic) or activating (neomorphic) in genes that act
functionally as either activators or repressors. Within the scope of this study
we must accept this uncertainty and assume mutations are generically disruptive
to the cell regardless of whether they are activating or deactivating or whether
they affect a gene that we might classify as one thing or another ( _e.g._
"Tumor suppressor" or "Oncogene"). Second, as previously mentioned, we know that
low-expressed genes and non-expressed genes are known to accumulate mutations at
a higher rate due to transcription coupled repair. To address this issue, we
identified genes with low expression in each type of cancer and eliminated them
for that cancer type only (see methods for details). <strike>Highly expressed
genes also have high mutation rates owing to transcription induced mutagenesis
[@parkGenomicEvidenceElevated2012]. We felt that this mechanism results in
cell-type-specific biases that might be biologically meaningful for
predisposition to different classes of cancer however, and therefore chose to
include these genes in our analysis.</strike>

After selecting our pathways and genes, we then compiled a matrix of the
pathways, assigning a Boolean value (a 1 or 0, indicating whether any pathway
member gene carries a mutation) to the pathway regardless of the number of genes
carrying mutations (this data is visualized in **figure 1a**).
<div style="text-align: right"> Generation of Somatic Mutation Data </div>
```{r data-set-creation, eval=FALSE}
## Data acquired from GDC
tcga_ssm <- fread("../data/simple_somatic_mutation.open.tsv.gz", header = T, sep = "\t")
## Somatic mutations focused in analysis
selected_mut <- c("3_prime_UTR_variant", "5_prime_UTR_premature_start_codon_gain_variant", "5_prime_UTR_variant", "frameshift_variant", "missense_variant", "splice_acceptor_variant", "splice_donor_variant", "stop_lost", "start_lost", "stop_gained")
## Filtering tcga_ssm for our selected mutations
tcga_ssm <- tcga_ssm[tcga_ssm$consequence_type %in% selected_mut, ]
## Selecting column that we are interested in
tcga_ssm <- data.frame(chr = paste0("chr",tcga_ssm$chromosome),
                   start = tcga_ssm$chromosome_start,
                   end = tcga_ssm$chromosome_end,
                   ref_allele = tcga_ssm$reference_genome_allele,
                   alt_allele = tcga_ssm$mutated_to_allele,
                   type = tcga_ssm$consequence_type,
                   mut_type = tcga_ssm$mutation_type,
                   ensembl_id = tcga_ssm$gene_affected,
                   donor_id = tcga_ssm$icgc_donor_id,
                   sample_id = tcga_ssm$submitted_sample_id,
                   stringsAsFactors = F)
## Removing duplicate rows
tcga_ssm <- tcga_ssm %>% distinct()
## Recategorizing the mutation type
tcga_ssm[which(tcga_ssm$type == "frameshift_variant"), "type"] <- tcga_ssm[which(tcga_ssm$type == "frameshift_variant"), "mut_type"] 
tcga_ssm <- tcga_ssm %>% mutate(type = case_when(type == "insertion of <=200bp" ~ "frameshins_variant", ## relabel values
                                                 type == "deletion of <=200bp" ~ "frameshdel_variant",
                                                 type == "splice_acceptor_variant" ~ "splice_variant",
                                                 type == "splice_donor_variant" ~ "splice_variant",
                                                 type == "start_lost" ~ "nonsense_variant",
                                                 type == "stop_gained" ~ "nonsense_variant",
                                                 type == "5_prime_UTR_premature_start_codon_gain_variant" ~ "UTR_5_prime_variant",
                                                 type == "5_prime_UTR_variant" ~ "UTR_5_prime_variant",
                                                 type == "3_prime_UTR_variant" ~ "UTR_3_prime_variant",
                                                 type == "missense_variant" ~ "missense_variant",
                                                 type == "stop_lost" ~ "nonstop_variant")) %>% 
  dplyr::select(-mut_type) ## rm mut_type col


## Preparing sample metadata ##

complete_sample <- tcga_ssm %>% .[,"sample_id"] %>% unique() %>% .[grep(pattern= "TCGA-", x = .)] # complete TCGA tumor samples

tcga_sample <- fread("../data/sample.tsv.gz", header = T, sep = "\t")
tcga_sample <- tcga_sample %>% dplyr::select(project_code, submitted_sample_id, submitted_specimen_id, icgc_donor_id, submitted_donor_id, study) %>% 
  .[.$submitted_sample_id %in% complete_sample,] %>%
  mutate(type = substring(.$submitted_sample_id, first = 14, last = 15)) %>% ## 01 is primary solid tumor
  filter(type == "01") ## filtering 01 samples
tcga_dup_id <- tcga_sample %>% .[.$icgc_donor_id %in% .[which(duplicated(.$submitted_donor_id)),]$icgc_donor_id,] %>% ## what are the dup donors
  filter(study == "PCAWG") ## selecting dup samples that are PCAWG
tcga_sample <- tcga_sample %>% .[!.$icgc_donor_id %in% tcga_dup_id$icgc_donor_id, ] %>% ## remove the dups from tcga_samples
  rbind(., tcga_dup_id) ## rbind the fixed dups

tcga_donor <- fread("../data/donor.tsv.gz", header = T, sep = "\t")
tcga_donor <- tcga_donor %>% dplyr::select(icgc_donor_id, project_code, submitted_donor_id, donor_sex, donor_vital_status, disease_status_last_followup, donor_age_at_diagnosis, donor_age_at_enrollment, donor_age_at_last_followup, donor_survival_time, donor_interval_of_last_followup) ## selecting columns that we are interested in

sample_sheet <- left_join(tcga_sample, tcga_donor, by = c("submitted_donor_id", "project_code", "icgc_donor_id"))

## Preparing gene metadata ##

# genelist <- unique(as.character(tcga_ssm$ensembl_id)) ## complete genelist
# ensembl_2_entrez_full_list <- bitr(genelist, 
#                                    fromType = "ENSEMBL", 
#                                    toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
load("../data/ensembl_2_entrez_full_list.Rda")
tcga_ssm <- tcga_ssm[tcga_ssm$ensembl_id %in% ensembl_2_entrez_full_list$ENSEMBL,]

## Now we only have 19067 genes
complete_ensembl <- as.data.frame(unique(as.character(tcga_ssm$ensembl_id)))
names(complete_ensembl) <- "ensembl_id"

selected_mut <- unique(as.character(tcga_ssm$type))
## For every mutation type.. select them.. and create a tsv file 
for(selected_mut in selected_mut) {
  print(selected_mut)
  df <-  tcga_ssm %>% filter(type == selected_mut) %>% ## filter data by selected_mut
    dplyr::select(sample_id, ensembl_id) %>% ## select column sample_id and ensembl_id
    mutate(mutation = 1) %>% ## create a new column with 1's
    distinct() %>% ## remove duplicated rows
    spread(., key = "sample_id", value = "mutation") %>% ## long to wide format
    remove_rownames() %>% ## move ensembl_id column to rownames
    column_to_rownames(var = "ensembl_id") %>% 
    replace(is.na(.), values = 0) %>% ## convert NA's to 0's
    .[, names(.) %in% sample_sheet$submitted_sample_id] ## rm any sample_id not included in complete_sample
  missing_gene <- as.data.frame(matrix(data = 0, nrow = length(complete_ensembl$ensembl_id[!complete_ensembl$ensembl_id %in% rownames(df)]), ncol = length(df)))
  rownames(missing_gene) <- complete_ensembl$ensembl_id[!complete_ensembl$ensembl_id %in% rownames(df)] ## insert missing ensembl_id into row names
  names(missing_gene) <- names(df) ## insert sample_id names into col names
  df <- rbind(df, missing_gene) ## rbind df and missing_gene
  rm(missing_gene) ## rm data
  gc()
  df <- df[match(complete_ensembl$ensembl_id, rownames(df)),] ## match df rownames with genelist
  missing_sample <- as.data.frame(matrix(data = 0, nrow = nrow(df), ncol = length(sample_sheet$submitted_sample_id[!sample_sheet$submitted_sample_id %in% names(df)])))
  rownames(missing_sample) <- rownames(df) ## insert ensembl_id into row names
  names(missing_sample) <- sample_sheet$submitted_sample_id[!sample_sheet$submitted_sample_id %in% names(df)] ## insert missing sample_id names into col names
  df <- cbind(df, missing_sample) ## rbind df and missing_gene
  rm(missing_sample) ## rm data
  gc()
  df <- df[, match(sample_sheet$submitted_sample_id, names(df))]
  assign(paste0(selected_mut), value = df) ## create df in r environment
  rm(df)
  gc()
  }

# ensembl <- useEnsembl(biomart="ensembl", dataset = "hsapiens_gene_ensembl")
 # genelist <- rownames(missense_variant)
 # gene_data <- getBM(
 #   attributes = c(
 #     'ensembl_gene_id',
 #     'external_gene_name',
 #     'gene_biotype',
 #     'chromosome_name', 
 #     'start_position',
 #     'end_position', 
 #     'strand'),
 #   filters = 'ensembl_gene_id',
 #   values = genelist,
 #   mart = ensembl)
 # save(gene_data, file = "gene_data.Rda")
load("../data/gene_data.Rda")
######################

missense_variant <- missense_variant[rownames(missense_variant) %in% gene_data$ensembl_gene_id,]
UTR_3_prime_variant <- UTR_3_prime_variant[rownames(UTR_3_prime_variant) %in% gene_data$ensembl_gene_id,]
UTR_5_prime_variant <- UTR_5_prime_variant[rownames(UTR_5_prime_variant) %in% gene_data$ensembl_gene_id,]
frameshdel_variant <- frameshdel_variant[rownames(frameshdel_variant) %in% gene_data$ensembl_gene_id,]
frameshins_variant <- frameshins_variant[rownames(frameshins_variant) %in% gene_data$ensembl_gene_id,]
splice_variant <- splice_variant[rownames(splice_variant) %in% gene_data$ensembl_gene_id,]
nonstop_variant <- nonstop_variant[rownames(nonstop_variant) %in% gene_data$ensembl_gene_id,]
nonsense_variant <- nonsense_variant[rownames(nonsense_variant) %in% gene_data$ensembl_gene_id,]


## Matching df to the ensembl_id orders in gene_data
match_ensembl <- function(x) {
  x <- x[match(gene_data$ensembl_gene_id, rownames(x)),]
}

missense_variant <- match_ensembl(missense_variant)
UTR_3_prime_variant <- match_ensembl(UTR_3_prime_variant)
UTR_5_prime_variant <- match_ensembl(UTR_5_prime_variant)
frameshdel_variant <- match_ensembl(frameshdel_variant)
frameshins_variant <- match_ensembl(frameshins_variant)
splice_variant <- match_ensembl(splice_variant)
nonstop_variant <- match_ensembl(nonstop_variant)
nonsense_variant <- match_ensembl(nonsense_variant)


genes <- GRanges(
  seqnames = paste("chr", gene_data$chromosome_name, sep=""),
  ranges = IRanges(
    start = gene_data$start_position,
    end = gene_data$end_position),
  strand = gene_data$strand,
  mcols = list(
    ensembl_gene_id = gene_data$ensembl_gene_name,
    hgnc_symbol = gene_data$hgnc_symbol))

gdc_tumor_coding_complete <- SummarizedExperiment(
  assays = list(
    missense   = missense_variant,
    splice     = splice_variant,
    frameshdel = frameshdel_variant,
    frameshins = frameshins_variant,
    utr3       = UTR_3_prime_variant,
    utr5       = UTR_5_prime_variant,
    nonsense   = nonsense_variant,
    nonstop    = nonstop_variant),
  rowRanges = genes,
  colData = sample_sheet)
save(gdc_tumor_coding_complete, file = "../data/gdc_tumor_coding_complete.Rdata")
rm(list = ls(all.names = T))
gc()
```
<div style="text-align: right"> Reactome Pathway Creation </div>
```{r reactome-pathway-creation, eval=FALSE}
load("../data/gdc_tumor_coding_complete.Rdata")
load("../data/gene_data.Rda")
## Preparing level_0 to level_1 pathway conversion table. level_0 is the master pathway. level_1 is the parent pathway
main2first <- fread(file = "../data/reactome_main2first_level.txt", header = T, sep = "\t")

main2first <- main2first %>% .[,2:3] %>%
  'colnames<-' (c("first_level", "main_level")) ## rename colnames

## Filtering out master pathways from first2second_level
first2second <- fread(file = "../data/reactome_first2second_level.txt", header = F, sep = "\t")
first2second <- first2second %>% 'colnames<-' (c("first_level", "second_level")) %>% ## rename colnames
  .[grep(pattern = "-HSA-", .$first_level)] %>% ## keep human pathways
  .[!.$first_level %in% unique(main2first$main_level),] %>% ## remove main level pathways from first level
  .[!.$first_level %in% unique(.$second_level),] ## remove second level pathways from first level

ensembl2reactome <- fread(file = "../data/Ensembl2Reactome_All_Levels.txt", header = F, sep = "\t") ## ensembl to pathway list acquired from reactome database
ensembl2reactome <- ensembl2reactome %>% .[,c(1:2,4)] %>% ## select columns ensembl, pathway_id, pathway_name columns
  'colnames<-' (c("ENSEMBL", "PATH_ID", "PATH_NAME")) %>% ## rename colnames
  .[grep(pattern = "-HSA-", .$PATH_ID),] %>% ## keep human pathways
  .[.$PATH_ID %in% first2second$second_level,] ## include only second level pathways

## Removing pathways that are not molecular functions such as pathways that related to disease, defect, bacteria, disorder, myosin and etc
rm_pathways <- c("R-HSA-1222499", "R-HSA-1226099", "R-HSA-1300642", "R-HSA-1300645", "R-HSA-162906",  "R-HSA-1643713", "R-HSA-168254",
                 "R-HSA-212436", "R-HSA-2160456", "R-HSA-2219528", "R-HSA-2474795", "R-HSA-2534343", "R-HSA-2644603", "R-HSA-2978092",
                 "R-HSA-3296482", "R-HSA-3304351", "R-HSA-3560782", "R-HSA-3781860", "R-HSA-388396",  "R-HSA-3906995", "R-HSA-4791275",
                 "R-HSA-5339562", "R-HSA-5387390", "R-HSA-5545483", "R-HSA-5576886", "R-HSA-5576890", "R-HSA-5576892", "R-HSA-5576893",
                 "R-HSA-5576894", "R-HSA-5578768", "R-HSA-5578775", "R-HSA-5579029", "R-HSA-5602358", "R-HSA-5609975", "R-HSA-5619084",
                 "R-HSA-5619102", "R-HSA-5632927", "R-HSA-5632928", "R-HSA-5632968", "R-HSA-5632987", "R-HSA-5663084", "R-HSA-5687613",
                 "R-HSA-6802957", "R-HSA-8862803", "R-HSA-8876384", "R-HSA-9005891", "R-HSA-9605310", "R-HSA-9616333", "R-HSA-9616334",
                 "R-HSA-9629232", "R-HSA-9630750", "R-HSA-9632693", "R-HSA-9645722", "R-HSA-9646303", "R-HSA-9646304")

## Ensembl to reactome pathway lookup table
gene2level2_lookup <- ensembl2reactome %>% dplyr::select(-PATH_NAME) %>% ## remove PATH_NAME col
  .[.$ENSEMBL %in% gene_data$ensembl_gene_id,] %>% ## only include genes that are in the gene_data 
  mutate(member = 1) %>% ## add a column for ensembl to pathway membership
  distinct() %>% ## keep unique rows
  spread(data = ., key = "PATH_ID", value = "member") %>% ## long to wide format
  column_to_rownames(var = "ENSEMBL") %>% ## ensembl_id as rownames
  .[,!names(.) %in% rm_pathways] %>% ## filter rm_pathways from lookup table
  replace(is.na(.), values = 0) ## replace NA's with 0's
#save(gene2level2_lookup, file = "gene2_level2_lookup.Rda")
## Pathway names to level 2 lookup table
pathname2level2 <- ensembl2reactome %>% dplyr::select(-ENSEMBL) %>% ## remove ENSEMBL col
  .[.$PATH_ID %in% names(gene2level2_lookup),] %>% ## only include pathways in gene2level2_lookup
  distinct(.) ## keep unique rows
#save(pathname2level2, file = "pathname2level2.Rda")
rm(rm_pathways, ensembl2reactome, main2first, first2second)
gc()
```
<div style="text-align: right"> Filter tissue-specific low expressed genes </div>
```{r filter-tissue-specific-low-expressed-genes, eval=FALSE}
gtf_file <- "../data/gencode.v30.annotation.gtf"
gencode_gtf <- rtracklayer::import(gtf_file)
gene_info <- as.data.frame(mcols(gencode_gtf)[, c("type", "gene_id", "transcript_id", "gene_type", "transcript_type", "gene_name")])#; rm(gencode_gtf)
## We only want transcript only 
gene_info <- gene_info[gene_info$type == "transcript",]
## We only want protein_coding only
gene_info <- gene_info[gene_info$gene_type == "protein_coding",]
gene_info <- gene_info[gene_info$transcript_type == "protein_coding",]
## Filtering for transcript_id, gene_id, gene_name columns
g2t <- gene_info %>% dplyr::filter(type == "transcript") %>% dplyr::select(transcript_id, gene_id, gene_name)
## Condense df to only having unique data. 
good.g2t <- as_tibble(unique(g2t))

## df with transcript_id and entrez id                      
gencode_entrez <- readr::read_tsv("../data/gencode.v30.metadata.EntrezGene.gz", col_names = FALSE)
head(gencode_entrez[1:10,])
colnames(gencode_entrez) <- c("transcript_id", "entrez_id")
## Joining data that are joined together by transcript_id
entrez_g2t <- left_join(good.g2t, gencode_entrez)
head(entrez_g2t[1:10,])
## Remove transcript_id column
entrez_g2t$transcript_id <- NULL
## Remove . and the numbers following from the ensembl id.
entrez_g2t$gene_id <- gsub("\\..*", "", entrez_g2t$gene_id)
## Removing data that does not have entrez id. 83091 out of 83647 had NA values
table(is.na(entrez_g2t$entrez_id))
entrez_g2t <- entrez_g2t[!is.na(entrez_g2t$entrez_id),]
entrez_g2t <- unique(entrez_g2t[, c("gene_id", "gene_name", "entrez_id")])
## Are there any duplicates in our df? Yes. Only one dupped ensembl id.
table(duplicated(entrez_g2t$gene_id))
## These are pseudogenes... We will remove them all.
dups_gene_id <- entrez_g2t[which(duplicated(entrez_g2t$gene_id)),]
## We're removing the rows that are associated with this pseudogene
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285607")),]
## It's gone
which(entrez_g2t$gene_id == "ENSG00000285607")
## Are there any duplicate gene names? Yes! 14 of them
table(duplicated(entrez_g2t$gene_name))
dups_gene_name <- entrez_g2t[which(duplicated(entrez_g2t$gene_name)),]
entrez_g2t[entrez_g2t$gene_name == "TBCE",]
## ENSG00000285053  is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285053")),]
entrez_g2t[entrez_g2t$gene_name == "PDE11A",]
## ENSG00000284741 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284741")),]
entrez_g2t[entrez_g2t$gene_name == "ATXN7",]
## ENSG00000285258 is not in the database anywhere
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285258")),]
entrez_g2t[entrez_g2t$gene_name == "MATR3",]
## ENSG00000280987 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000280987")),]
entrez_g2t[entrez_g2t$gene_name == "SOD2",]
## ENSG00000285441 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285441")),]
entrez_g2t[entrez_g2t$gene_name == "ABCF2",]
## ENSG00000285292 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285292")),]
entrez_g2t[entrez_g2t$gene_name == "PINX1",]
## ENSG00000258724 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000258724")),]
entrez_g2t[entrez_g2t$gene_name == "HSPA14",]
## ENSG00000284024 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284024")),]
entrez_g2t[entrez_g2t$gene_name == "IGF2",]
## ENSG00000284779 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284779")),]
entrez_g2t[entrez_g2t$gene_name == "ATF7",]
## ENSG00000267281 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000267281")),]
entrez_g2t[entrez_g2t$gene_name == "DIABLO",]
## ENSG00000284934 is a homolog
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284934")),]
entrez_g2t[entrez_g2t$gene_name == "GGT1",]
## ENSG00000286070 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286070")),]
entrez_g2t[entrez_g2t$gene_name == "SCO2",]
## ENSG00000130489 no longer exist
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000130489")),]
entrez_g2t[entrez_g2t$gene_name == "TMSB15B",]
## ENSG00000269226 is a readthrough
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000269226")),]

## Are there any dupped entrez id? 
dups_entrez <- entrez_g2t[which(duplicated(entrez_g2t$entrez_id)),]

entrez_g2t[entrez_g2t$entrez_id == "55486",]
## ENSG00000283765 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000283765")),]
entrez_g2t[entrez_g2t$entrez_id == "55300",]
## ENSG00000281028 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000281028")),]
entrez_g2t[entrez_g2t$entrez_id == "653082",]
## ENSG00000286094 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286094")),]
entrez_g2t[entrez_g2t$entrez_id == "8622",]
## ENSG00000284762 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000284762")),]
entrez_g2t[entrez_g2t$entrez_id == "221468",]
## ENSG00000286105 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286105")),]
entrez_g2t[entrez_g2t$entrez_id == "1124",]
## ENSG00000285162 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285162")),]
entrez_g2t[entrez_g2t$entrez_id == "102723849",]
## ENSG00000283765 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000273520")),]
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000262461")),]
entrez_g2t[entrez_g2t$entrez_id == "883",]
## ENSG00000286112 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286112")),]
entrez_g2t[entrez_g2t$entrez_id == "22891",]
## ENSG00000261130 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285551")),]
entrez_g2t[entrez_g2t$entrez_id == "54816",]
## ENSG00000285253 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285253")),]
entrez_g2t[entrez_g2t$entrez_id == "23370",]
## ENSG00000268861 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000268861")),]
entrez_g2t[entrez_g2t$entrez_id == "284391",]
## ENSG00000286098 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000286098")),]
entrez_g2t[entrez_g2t$entrez_id == "440519",]
## ENSG00000283201 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000283201")),]
entrez_g2t[entrez_g2t$entrez_id == "149951",]
## ENSG00000285382 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000285382")),]
entrez_g2t[entrez_g2t$entrez_id == "388820",]
## ENSG00000268861 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000278961")),]
entrez_g2t[entrez_g2t$entrez_id == "8471",]
## ENSG00000286098 not found on ncbi
entrez_g2t <- entrez_g2t[-c(which(entrez_g2t$gene_id == "ENSG00000260548")),]

## manual gene removal complete

load("../data/TCGA.RNA.Rda")


## Put entrez ids as rownames and shortening the TCGA ids since the last three portions of the TCGA barcode is coded to explain the test performed and etc. 
tumor_samples <- gdc_tumor_coding_complete %>% colData(.) %>% 
  as.data.frame(.) %>%
  dplyr::select(project_code, submitted_specimen_id) %>%
  mutate(submitted_specimen_id = substring(text = .$submitted_specimen_id, first = 1, last = 15))

TCGA.RNA <- TCGA.RNA %>% dplyr::select(-gene_symbol) %>%
  remove_rownames(.) %>% column_to_rownames(var = "gene_id") %>% 
  .[rownames(.) %in% entrez_g2t$entrez_id,] %>%
  'colnames<-' (substring(text = names(.), first = 1, last = 15)) %>% ## only need the first 15 character of barcode
  .[, names(.) %in% tumor_samples$submitted_specimen_id] ## only include tumor samples in our analysis

g2e_list <- entrez_g2t[entrez_g2t$entrez_id %in% rownames(TCGA.RNA),]

express_no_express_lookup <- list()
for(i in unique(tumor_samples$project_code)) {
  df <- which(tumor_samples$project_code == i) ## which rows are the spec cancer types
  df <- tumor_samples[df,] ## select only those rows
  gdc <- TCGA.RNA[, names(TCGA.RNA) %in% df$submitted_specimen_id] ## filter the norm_gdc_gene_exp dataset with those tumor samples
  mean_gdc <- as.data.frame(apply(gdc, MARGIN = 1, function(x) {
    mean(x)
  })) ## take the mean for each gene across the tumor samples
  names(mean_gdc) <- i
  express_no_express_lookup[[i]] <- mean_gdc ## store data in list
  rm(df, gdc, mean_gdc)
  gc
}
express_no_express_lookup <- do.call(cbind, express_no_express_lookup) ## cbind list
express_no_express_lookup[is.na(express_no_express_lookup)] <- 0

## 
gdc_ensembl_id <- gdc_tumor_coding_complete %>% rowRanges(.) %>% as.data.frame(.) %>% ## getting the ensembl_ids in our gdc dataset
  rownames(.)
g2e_list <- g2e_list[g2e_list$gene_id %in% gdc_ensembl_id,] ## subsetting the g2e_list to only have ensembl_ids in our gdc dataset


express_no_express_lookup <- express_no_express_lookup[rownames(express_no_express_lookup) %in% g2e_list$entrez_id, ]
express_no_express_lookup[express_no_express_lookup <= 10] <- 0 ## Set threshold to >10 for gene expression
express_no_express_lookup[express_no_express_lookup >10] <- 1 

## convert entrez id to ensembl_id
express_no_express_lookup <- express_no_express_lookup %>% .[match(g2e_list$entrez_id, rownames(.)),] %>% ## match the entrez_id express_no_express_lookup with g2e_list
  'rownames<-' (g2e_list$gene_id) ## replace entrez_id with 

## Booleanize mutation count for each tumor sample
missense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$missense) 
splice <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$splice)
frameshdel <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshdel)
frameshins <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshins)
utr3 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr3)
utr5 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr5)
nonstop <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonstop)
nonsense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonsense)

gdc_mut_count_patient <- missense + splice + frameshdel + frameshins + utr3 + utr5 + nonstop + nonsense ## add up all the mutation count
rm(missense, splice, frameshdel, frameshins, utr3, utr5, nonstop, nonsense)
gc()
gdc_mut_count_patient <- gdc_mut_count_patient[rownames(gdc_mut_count_patient) %in% rownames(express_no_express_lookup),] ## filter data with genes in the express_no_express_lookup
gdc_mut_count_patient[gdc_mut_count_patient > 0] <- 1 ## booleanize data, anything greater than 1 is equal to 1

gdc_gene_exp_adj <- list()
for(i in unique(tumor_samples$project_code)) {
  df <- which(tumor_samples$project_code == i) ## which rows are my spec cancer types
  df <- tumor_samples[df,] ## select only those rows
  gdc <- gdc_mut_count_patient[, names(gdc_mut_count_patient) %in% rownames(df)] ## filter the norm_gdc_gene_exp dataset with those tumor samples
  exp_data <- as.data.frame(express_no_express_lookup[, i]) ## select correct lookup column for the cancer type
  rownames(exp_data) <- rownames(express_no_express_lookup) ## Insert rownames to exp_data
  gdc <- gdc[match(rownames(exp_data), rownames(gdc)),] ## Match ensembl_id of exp_data with gdc data
  gdc_adj <- as.data.frame(apply(gdc, MARGIN = 2, function(x) {
    exp_data * x
  })) ## Function to multiply lookup to the data for the specific cancer type
  names(gdc_adj) <- names(gdc) ## Rename gdc_adj with tumor samples names
  gdc_gene_exp_adj[[i]] <- gdc_adj ## Store data to list
  rm(df, gdc, exp_data, gdc_adj)
  gc()
}
## Combining list into a dataframe.
gdc_gene_exp_adj <- do.call(cbind, gdc_gene_exp_adj) ## cbind list
names(gdc_gene_exp_adj) <- gsub("^.*\\.", "", names(gdc_gene_exp_adj)) ## tumor sample names have cancer type name in it. Removing it using gsub
rm(gdc_mut_count_patient, express_no_express_lookup, TCGA.RNA, gdc_tumor_coding_complete, gencode_gtf, dups_entrez, dups_gene_id, dups_gene_name, g2t, good.g2t, gene_info, g2e_list, entrez_g2t, gencode_entrez)
gc()
save(gdc_gene_exp_adj, file = "gdc_gene_exp_adj.Rda")
```
<div style="text-align: right"> Genes to pathway transformation </div>
```{r genes-to-pathways-transformation, eval=FALSE}
load("../data/gdc_gene_exp_adj.Rda")
load("../data/gene2_level2_lookup.Rda")
sample_list <- as.character(names(gdc_gene_exp_adj)) ## Create sample list
complete_path <- as.character(names(gene2level2_lookup)) ## Get names of pathways

df_gene <- as.data.frame(t(gdc_gene_exp_adj)) ## Transpose data so samples are rows and genes are columns
df_gene <- df_gene > 0 ## if 1 == TRUE if 0 == FALSE
df_gene <- as.data.frame(df_gene) ## turn matrix back to df

df_gene <- df_gene %>% rownames_to_column(var = "sample") %>% 
  gather(key = "gene", value = "mutation", -sample) %>% ## Wide to long format
  .[.$mutation,] %>% ## Keep TRUE's (which are 1's); Remove FALSE's (0's)
  dplyr::select(-mutation) ## Remove column named mutation

path_lookup <- as.data.frame(t(gene2level2_lookup)) ## Transpose data so paths are rows and genes are columns
path_lookup <- path_lookup > 0 ## if 1 == TRUE if 0 == FALSE
path_lookup <- as.data.frame(path_lookup) ## turn matrix back to df

path_lookup <- path_lookup %>% rownames_to_column(var = "path") %>%
  gather(key = "gene", value = "path_member", -path) %>% ## Wide to long format
  .[.$path_member,] %>% ## Keep TRUE's (which are 1's); Remove FALSE's (0's)
  dplyr::select(-path_member)

df_lookup <- df_gene %>% left_join(., path_lookup, by = "gene") %>% ## Match df_gene and lookup dataframes
  na.omit(.) ## Omit NA's that were generated due to no matches

## getting the genes involved in our analysis
genes_in_pathanalysis <- unique(df_lookup$gene)
write.table(genes_in_pathanalysis, file = "genes_in_pathanalysis.tsv", quote = FALSE, sep = "\t", row.names =  FALSE) ## save data

df_lookup <- as.data.frame(table(df_lookup$sample, df_lookup$path)) ## Generate 1's for the matches

df_lookup <- df_lookup %>% spread(key = "Var1", value = "Freq") %>% ## Long to wide format
  remove_rownames() %>%
  column_to_rownames(var = "Var2") ## path_id to rownames

## Insert missing tumor samples due to no mutations
df_missing <- sample_list[!(sample_list %in% names(df_lookup))] 
mx <- as.data.frame(matrix(data = 0, nrow = nrow(df_lookup), ncol = length(df_missing))) ## Create mx to store data
names(mx) <- df_missing ## Missing tumor sample names in col names
rownames(mx) <- rownames(df_lookup) ## Path names in rownames
df_lookup <- cbind(df_lookup, mx) ## cbind data

## Insert missing pathway due to no mutations
path_missing <- complete_path[!c(complete_path %in% rownames(df_lookup))]
mx <- as.data.frame(matrix(data = 0, nrow = length(path_missing), ncol = ncol(df_lookup))) ## Create mx to store data
names(mx) <- names(df_lookup) ## Insert tumor sample names
rownames(mx) <- path_missing ## Insert missing path names
df_lookup <- rbind(df_lookup, mx) ## cbind data

pathway_order <- rownames(df_lookup) ## need to perserve the order of pathway for the next operation
gdc_reactome_path_analysis <- df_lookup %>% mutate_if(is.numeric, ~1 * (. > 0)) %>% as.data.frame() %>% ## Mutate anything >0 to 1's (Booleanize data)
  'rownames<-' (pathway_order) %>%
  rownames_to_column(var = "PATH_ID") %>%
  left_join(., pathname2level2, by = "PATH_ID") %>%
  dplyr::select(-PATH_ID) %>% 
  remove_rownames() %>% column_to_rownames(var = "PATH_NAME")
rm(df_missing, path_missing, mx, df_gene, sample_list, complete_path, pathway_order, df_lookup)
gc()
save(gdc_reactome_path_analysis, file = "gdc_reactome_path_analysis.Rda")
```
<div style="text-align: right"> Figure 1 data </div>
```{r tumor-clusters, eval = FALSE}
####################
## Figure 1A data ##
####################
load("../data/gdc_reactome_path_analysis.Rda")
load("../data/gdc_tumor_coding_complete.Rdata")

path_daisy <- daisy(t(gdc_reactome_path_analysis), type = list(symm  = c(1:col(gdc_reactome_path_analysis))))
path_daisy <- as.dist(as.matrix(path_daisy))

path_daisy_row <- daisy(gdc_reactome_path_analysis, type = list(symm  = c(1:nrow(gdc_reactome_path_analysis))))
path_daisy_row <- as.dist(as.matrix(path_daisy_row))

pathway_dendro <- pheatmap(gdc_reactome_path_analysis,
         show_rownames = F,
         show_colnames = F,
         cellheight = 1,
         cellwidth = 0.5,
         clustering_distance_cols = path_daisy,
         clustering_distance_rows = path_daisy_row,
         cluster_cols = T,
         cluster_rows = T,
         cutree_cols = 19,
         annotation_legend = F,
         legend_breaks = c(0, 0.25, 0.75, 1),
         legend_labels = c("","Not Mut", "Mut", ""),
         color = c("grey", "firebrick3"),
         clustering_method = "ward.D2",
         legend = T)
dev.off()
pathway_dendro <- pathway_dendro$tree_row$order ## getting pathway row ordering
df <- rownames(gdc_reactome_path_analysis)
pathway_dendro <- df[pathway_dendro]

tumor_samples <- gdc_tumor_coding_complete %>% colData(.) %>%  ## get sample_id data from gdc
  as.data.frame(.) %>%
  dplyr::select(project_code, submitted_specimen_id) %>%  
  mutate(project_code = gsub(pattern = "\\-.*", replacement = "", x = .$project_code))

cancer_type <- tumor_samples %>% .$project_code %>% unique %>% .[order(.)] ## get unique cancer_type

cancer_dendro <- list() ## getting cancer type specific tumor sample clustering info

for(n in cancer_type) {
  cancer_info <- tumor_samples %>% rownames_to_column(var = "sample_id") %>% ## move sample_id to column
    filter(project_code == n) %>% ## filter for specific cancer type
    .[, "sample_id"] ## retain only sample_id
  cancer_path <- gdc_reactome_path_analysis %>% .[, names(.) %in% cancer_info] ## only including filtered sample_id
  cancer_heatmap <- pheatmap(cancer_path[pathway_dendro,], ## using pathway row order 
                          show_rownames = F,
                          show_colnames = F,
                          cellheight = 1,
                          cellwidth = 0.5,
                          cluster_cols = T,
                          cluster_rows = F,
                          annotation_legend = F,
                          color = c("grey", "firebrick3"),
                          clustering_method = "single", ## ordering is single linkage
                          legend = T)
  cancer_order <- cancer_heatmap$tree_col$order ## cancer order
  cancer_path <- names(cancer_path) ## getting the sample_id original order 
  cancer_dendro[[n]] <- cancer_path[cancer_order] ## reorder cancer by heatmap order
  rm(cancer_info, cancer_path)
  gc()
}
dev.off()

for(n in cancer_type[1]) { ## getting the order of clustering by cancer type by alphabetical order
  print(n)
  cancer_order <- as.data.frame(do.call(cbind, cancer_dendro[n])) ## read in first list
  names(cancer_order) <- "sample_id"
  cancer_order$type <- n
  for(p in cancer_type[-1]) {
    print(p)
    df1 <- as.data.frame(do.call(cbind, cancer_dendro[p])) ## read in rest of list in loop
    names(df1) <- "sample_id"
    df1$type <- p
    cancer_order <- rbind(cancer_order, df1) ## combine the two df and repeat the loop until end of cancer_type
    rm(df1)
    gc()
  }
}

cancer_dendro <- cancer_order %>% .[,"sample_id"] %>% as.character(.) ## sample_id ordering

cancer_disrupt_heatmap <- gdc_reactome_path_analysis %>% rownames_to_column(var = "path_name") %>%  ## move rownames to column
  gather(., key = "sample_id", value = "membership", -path_name) ## wide to long format
cancer_disrupt_heatmap$membership <- as.character(cancer_disrupt_heatmap$membership) ## turn boolean values into categories

## set levels based on clustering algorithm above
cancer_disrupt_heatmap$path_name <- factor(cancer_disrupt_heatmap$path_name, levels = c(rev(pathway_dendro)))
cancer_disrupt_heatmap$sample_id <- factor(cancer_disrupt_heatmap$sample_id, levels = c(cancer_dendro))

cancer_disrupt_heatmap <- left_join(cancer_disrupt_heatmap, cancer_order, by = "sample_id") ## add in cancer type information for faceting
save(cancer_disrupt_heatmap, file = "cancer_disrupt_heatmap.Rda") 

####################
## Figure 1C data ##
####################
gdc_path <- gdc_reactome_path_analysis %>% t() %>% as.data.frame() %>%
  rownames_to_column(var = "sample") %>% ## move path_names to column
  .[match(rownames(tumor_samples), .$sample),]
  
## umap analysis
set.seed(3296)
gdc.umap <- umap(gdc_path[,2:length(gdc_path)]) ## default n_component is 2
gdc_umap_coors <- data.frame(x_2d = gdc.umap$layout[,1], y_2d = gdc.umap$layout[,2])

####################
## Figure 1E data ##
####################
## To identify clusters using hdbscan, use umap in the third dimension
set.seed(3296)
gdc.umap.3d <- umap(gdc_path[,2:length(gdc_path)], n_components = 3)
gdc.umap.3d <- data.frame(x_3d = gdc.umap.3d$layout[,1], y_3d = gdc.umap.3d$layout[,2], z_3d = gdc.umap.3d$layout[,3]) ## coors of umap
rownames(gdc.umap.3d) <- gdc_path$sample ## insert sample_id in rownames
hdb_cluster_assignment <- hdbscan(gdc.umap.3d, minPts =  114) ## min group size for cluster is 114 (value before cluster number drops significantly)
umap_2d_3d_coors <- data.frame(x_2d = gdc_umap_coors$x_2d, y_2d = gdc_umap_coors$y_2d, x_3d = gdc.umap.3d$x_3d, y_3d = gdc.umap.3d$y_3d, z_3d = gdc.umap.3d$z_3d, clust = as.factor(hdb_cluster_assignment$cluster), member_prob = hdb_cluster_assignment$membership_prob, row.names = rownames(gdc.umap.3d)) ## clust 0 is unclassified cluster

####################
## Figure 1F data ##
####################
## reclassifying clust 0 using our classified cluster as training test and unclassified as test set
umap_2d_3d_coors$knn_clust <- umap_2d_3d_coors$clust
umap_2d_3d_coors$knn_clust[umap_2d_3d_coors$clust==0]<-class::knn(train = umap_2d_3d_coors[,c("x_3d", "y_3d", "z_3d")][which(umap_2d_3d_coors$clust!=0),],
                                                                  test = umap_2d_3d_coors[,c("x_3d", "y_3d", "z_3d")][which(umap_2d_3d_coors$clust==0),],
                                                                  cl = umap_2d_3d_coors$clust[umap_2d_3d_coors$clust!=0], 
                                                                  k=15)

####################
## Figure 1G data ##
####################
missense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$missense) ## reading in mutation data from SE
splice <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$splice)
frameshdel <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshdel)
frameshins <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshins)
utr3 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr3)
utr5 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr5)
nonstop <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonstop)
nonsense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonsense)

mutation_count <- missense + splice + frameshdel + frameshins + utr3 + utr5 + nonstop + nonsense ## add up all the mutation count
rm(missense, splice, frameshdel, frameshins, utr3, utr5, nonstop, nonsense)
gc()
mutation_count <-mutation_count %>% .[rownames(.) %in% rownames(gene2level2_lookup),] %>% ## filter data with genes in our pathway lookup table
  colSums() %>% as.data.frame() %>% ## sum the mutation count
  'colnames<-' (c("mut_count")) %>% ## rename colname
  rownames_to_column(var = "V1") ## move rownames to column

umap_2d_3d_coors <- left_join(umap_2d_3d_coors, mutation_count, by = "V1") 

####################
## Figure 1H data ##
####################
tumor_samples <- gdc_tumor_coding_complete %>% colData(.) %>% 
  as.data.frame(.) %>%
  dplyr::select(project_code) %>% 
  mutate(project_code = gsub(pattern = "\\-.*", replacement = "", x = .$project_code)) %>%
  rownames_to_column(var = "V1")

umap_2d_3d_coors <- left_join(umap_2d_3d_coors, tumor_samples, by = "V1")

####################
## Figure 1I data ##
####################
## getting subtype information on our cancer type
subtypes <- PanCancerAtlas_subtypes() 
tumor_subtypes <- substring(text = tumor_samples$submitted_specimen_id, first = 1, last = 12) ## sample_id for subsetting subtypes
subtypes <- subtypes %>% dplyr::select(pan.samplesID, Subtype_Selected) %>% ## select the relevant columns
  mutate(Subtype_Selected = gsub(pattern = ".*\\.", replacement = "", x = .$Subtype_Selected)) %>% ## remove the cancer acronym at the beginning
  mutate(pan.samplesID = substring(text = .$pan.samplesID, first = 1, last = 12)) %>% ## interested in 
  distinct() %>% ## only unique columns
  .[.$pan.samplesID %in% tumor_subtypes, ] %>% ## include sample_ids that are in our analysis
  filter(Subtype_Selected != "Normal") %>%
  .[!.$pan.samplesID %in% .[which(duplicated(.$pan.samplesID)),]$pan.samplesID, ] ## remove dups

#save(subtypes, file = "subtypes.Rda")
load("../data/subtypes.Rda")

umap_2d_3d_coors <- umap_2d_3d_coors %>%
  mutate(pan.samplesID = substring(text = .$V1, first = 1, last = 12)) %>%
  left_join(., subtypes, by = "pan.samplesID") %>% ## left_join subtype data to umap
  dplyr::select(-pan.samplesID)

##########################
## Figure 1K and L data ##
##########################
## getting tumor_metastasis, staging
tcga_clinical_tumor_staging <- list()
for(n in cancer_type) {
  print(n)
  cancername <- paste0("TCGA-",n)
  query <- GDCquery(project = cancername,
                  data.category = "Clinical",
                  data.type = "Clinical Supplement",
                  data.format = "BCR Biotab",
                  file.type = "patient")
  GDCdownload(query)
  clinical <- GDCprepare(query)[[1]]
  if("ajcc_pathologic_tumor_stage" %in% colnames(clinical) & "ajcc_tumor_pathologic_pt" %in% colnames(clinical) == T) {
    cat("Yep, ajcc info in there!\n")
    clinical <- clinical[-c(1:2), c("bcr_patient_barcode","ajcc_pathologic_tumor_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")]
  }
  if(!"ajcc_pathologic_tumor_stage" %in% colnames(clinical) & "clinical_stage" %in% colnames(clinical) & "ajcc_tumor_pathologic_pt" %in% colnames(clinical)== T) {
    cat("Nope, ajcc stage not there, but pathologic scores is!\n")
    clinical <- clinical[-c(1:2), c("bcr_patient_barcode", "clinical_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")]
    names(clinical) <- c("bcr_patient_barcode","ajcc_pathologic_tumor_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")
  }
  if(!"ajcc_pathologic_tumor_stage" %in% colnames(clinical) & "clinical_stage" %in% colnames(clinical) & !"ajcc_tumor_pathologic_pt" %in% colnames(clinical) & "pathologic_T" %in% colnames(clinical) == T) {
    cat("Nope, ajcc info not there!\n")
    clinical <- clinical[-c(1:2), c("bcr_patient_barcode", "clinical_stage", "pathologic_T", "pathologic_M")]
    names(clinical) <- c("bcr_patient_barcode","ajcc_pathologic_tumor_stage", "ajcc_tumor_pathologic_pt", "ajcc_metastasis_pathologic_pm")
    }
  if(!"ajcc_pathologic_tumor_stage" %in% colnames(clinical) & !"clinical_stage" %in% colnames(clinical) == T) next
  if("ajcc_pathologic_tumor_stage" %in% colnames(clinical) & !"ajcc_tumor_pathologic_pt" %in% colnames(clinical) == T) next
  tcga_clinical_tumor_staging[[n]] <- clinical
}

tcga_clinical_tumor_staging <- do.call(rbind, tcga_clinical_tumor_staging) ## rbind the list together
tcga_clinical_tumor_staging <- tcga_clinical_tumor_staging[tcga_clinical_tumor_staging$bcr_patient_barcode %in% tumor_subtypes,] ## We have 6633 out of 7607 tumor data
missing_sample <- as.data.frame(matrix(data = "NA", nrow = length(tumor_subtypes[!tumor_subtypes %in% tcga_clinical_tumor_staging$bcr_patient_barcode]), ncol = length(tcga_clinical_tumor_staging) -1 ))

tcga_clinical_tumor_staging <- missing_sample %>% 'colnames<-' (names(tcga_clinical_tumor_staging[-1])) %>% ## colnames based on tcga_clinical_tumor_staging
  mutate(bcr_patient_barcode = tumor_subtypes[!tumor_subtypes %in% tcga_clinical_tumor_staging$bcr_patient_barcode]) %>%
  .[,c(4,1,2,3)] %>% ## reorder columns for rbinding
  rbind(tcga_clinical_tumor_staging, .) ## rbind

save(tcga_clinical_tumor_staging, file = "tcga_clinical_tumor_staging.Rda")

umap_2d_3d_coors <- umap_2d_3d_coors %>%
  mutate(bcr_patient_barcode = substring(text = .$V1, first = 1, last = 12)) %>%
  left_join(., tcga_clinical_tumor_staging, by = "bcr_patient_barcode") %>% ## left_join clinical data to umap
  dplyr::select(-bcr_patient_barcode)

M0 <- umap_2d_3d_coors$ajcc_metastasis_pathologic_pm %in% c("cM0 (i+)", "M0")
M1 <- umap_2d_3d_coors$ajcc_metastasis_pathologic_pm %in% c("M1", "M1a", "M1b", "M1c")
NA.value <- umap_2d_3d_coors$ajcc_metastasis_pathologic_pm %in% c("[Discrepancy]", "[Not Applicable]", "[Not Available]", "MX")
  
umap_2d_3d_coors$ajcc_metastasis_pathologic_pm[M0] <- "M0"
umap_2d_3d_coors$ajcc_metastasis_pathologic_pm[M1] <- "M1"
umap_2d_3d_coors$ajcc_metastasis_pathologic_pm[NA.value] <- "NA"

stage1 <- umap_2d_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage I", "Stage IA", "Stage IA1", "Stage IA2", "Stage IB", "Stage IB1", "Stage IB2", "Stage IC")
stage2 <- umap_2d_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage II", "Stage IIA", "Stage IIA1", "Stage IIA2", "Stage IIB", "Stage IIC")
stage3 <- umap_2d_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage III", "Stage IIIA", "Stage IIIB", "Stage IIIC", "Stage IIIC1", "Stage IIIC2")
stage4 <- umap_2d_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage IV", "Stage IVA", "Stage IVB", "Stage IVC")
unknown <- umap_2d_3d_coors$ajcc_pathologic_tumor_stage %in% c("Stage X", "[Discrepancy]", "[Not Applicable]", "I/II NOS", "NA", "[Not Available]")

umap_2d_3d_coors$ajcc_pathologic_tumor_stage[stage1] <- "Stage I"
umap_2d_3d_coors$ajcc_pathologic_tumor_stage[stage2] <- "Stage II"
umap_2d_3d_coors$ajcc_pathologic_tumor_stage[stage3] <- "Stage III"
umap_2d_3d_coors$ajcc_pathologic_tumor_stage[stage4] <- "Stage IV"
umap_2d_3d_coors$ajcc_pathologic_tumor_stage[unknown] <- "Not Available"

write.table(x = umap_2d_3d_coors, file = "umap_2d_3d_coors.tsv", quote = FALSE, sep = "\t", row.names = FALSE) ## save the data
```

```{r fig-1a-1l, warning=FALSE, fig.height=11, fig.width=8.5}
load("../data/cancer_disrupt_heatmap.Rda")
umap_cancertype <- fread("../data/umap_2d_3d_coors.tsv")

kelly.colours <- c("gray95", "gray13", "gold2", "plum4", "darkorange1",
                   "lightskyblue2", "firebrick", "burlywood3", "gray51", "springgreen4",
                   "lightpink2", "deepskyblue4", "lightsalmon2", "mediumpurple4",
                   "orange", "maroon", "yellow3", "brown4", "yellow4", "sienna4", "chocolate", "gray19")

fig1A <- ggplot(data = cancer_disrupt_heatmap, aes(y=sample_id, x=path_name, fill = membership)) + 
  geom_tile(size = 0.6) +
  scale_fill_manual(values = c("grey", "firebrick3")) +
  facet_grid(type ~ ., scales = "free", space = "free", switch = "y") +
scale_y_discrete(position = "left") +
  labs(tag = "A") +
  ylab("") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        strip.text.y = element_text(size = 12, hjust = 0),
        strip.text.y.left = element_text(angle = 0),
        strip.background = element_blank(),
        legend.position = "none") 

## fixing the text cutoff in heatmap
 fig1Ag <- ggplotGrob(fig1A)
 
 for(i in which(grepl("strip-l", fig1Ag$layout$name))){
   fig1Ag$grobs[[i]]$layout$clip <- "off"
 }
 
 fig1A <- as.ggplot(fig1Ag)
 rm(fig1Ag)

## umap monochrome 2d
fig1C <- ggplot(umap_cancertype, aes(x=x_2d, y=y_2d)) +
  geom_point(colour = "grey") +
  xlab("umap_1") +
  ylab("umap_2") +
  theme_classic() +
  labs(tag = "C") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank())
## umap hdb_scan
fig1E <- ggplot(subset(umap_cancertype, clust !=0), aes(x = x_2d, y = y_2d, color = as.factor(clust))) +
  geom_point(data = umap_cancertype, aes(x = x_2d, y = y_2d), colour = "grey") +
  geom_point() + scale_color_manual(values = kelly.colours[-c(1:2)]) +
  guides(color = F) +
  theme_classic() +
  xlab("umap_1") +
  ylab("umap_2") +
  theme_classic() +
  labs(tag = "E") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank())

## umap knn_clust 
fig1F <- ggplot(subset(umap_cancertype), aes(x = x_2d, y = y_2d, color = as.factor(knn_clust))) +
  geom_point() + scale_color_manual(values = kelly.colours[-c(1:2)]) +
  guides(color = F) +
  theme_classic() +
  xlab("umap_1") +
  ylab("umap_2") +
  theme_classic() +
  labs(tag = "F") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank())

umap_cancertype$knn_clust <- factor(umap_cancertype$knn_clust, levels = c(1:12))

## violin mutation count
fig1G <- ggplot(umap_cancertype, aes(x = knn_clust, y = mut_count)) +
    geom_violin(aes(fill = knn_clust)) + #draw_quantiles = c(0.25, 0.5, 0.75)) 
    geom_boxplot(aes(fill = knn_clust), width=0.2) +
    theme_classic() +
    scale_y_log10() +
    xlab("") +
    ylab("Log(Mutation Count)") +
    labs(color = "", tag = "G") +
    theme(axis.text.x = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.title.x = element_text(size = 12),
          axis.title.y = element_text(size = 12),
          legend.position = "none") +
  scale_fill_manual(values = kelly.colours[-c(1:2)])

## umap by cancer type
fig1H <- ggplot(umap_cancertype[umap_cancertype$project_code %in% c("COAD", "BRCA", "HNSC", "PAAD"),], aes(x = x_2d, y = y_2d, colour = project_code)) +
  geom_point(data = umap_cancertype, aes(x = x_2d, y = y_2d), colour = "grey") +
  geom_point() + 
  theme_classic() +
  ylab("umap_2") +
  xlab("umap_1") +
  labs(tag = "H", color = "") +
  scale_color_manual(labels = c("BRCA", "COAD", "HNSC", "PAAD"), 
                     values = c('#fb9a99', '#b15928', '#6a3d9a', 'springgreen4')) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

## umap of brca subtype
fig1I <- ggplot(umap_cancertype %>% filter(project_code == "BRCA" & !is.na(Subtype_Selected)), aes(x = x_2d, y = y_2d, colour = Subtype_Selected)) +
  geom_point(data = umap_cancertype, aes(x = x_2d, y = y_2d), colour = "grey") +
  geom_point() +
  theme_classic() +
  ylab("umap_2") +
  xlab("umap_1") +
  labs(tag = "I", color = "") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

## umap brca somatic mutations
load("../data/gdc_gene_exp_adj.Rda")
load("../data/gene_data.Rda")
brca <- gene_data %>% .[.$external_gene_name %in% c("BRCA1", "BRCA2"),] %>% dplyr::select(ensembl_gene_id, external_gene_name) ## Getting BRCA gene information
brca <- gdc_gene_exp_adj %>% .[rownames(.) %in% brca$ensembl_gene_id, ] %>% ## we only want brca gene
  rownames_to_column(var = "ensembl_gene_id") %>%
  left_join(., brca, by = "ensembl_gene_id") %>% 
  remove_rownames()  %>% column_to_rownames(var = "external_gene_name") %>% ## replace ensembl_id with gene_name
  dplyr::select(-ensembl_gene_id) %>% ## get ride of ensembl_id  col
  t() %>% as.data.frame() %>% ## correct orientation for umap_cancertype df
  mutate(BRCA1 = case_when(BRCA1 == "1" ~ "BRCA1",
                           BRCA1 == "0" ~ "NA"),
         BRCA2 = case_when(BRCA2 == "1" ~ "BRCA2",
                           BRCA2 == "0" ~ "NA")) %>%
  mutate(mut_type = gsub('NA,|,NA', '', paste(.$BRCA1, .$BRCA2, sep = ","))) %>% ## combine mutation cause some patients can have both
  dplyr::select(-BRCA1, -BRCA2) %>% ## remove col we don't need
  rownames_to_column(var = "V1") %>% ## getting data ready for leftjoin 
  left_join(., umap_cancertype[,c("V1", "x_2d", "y_2d", "project_code")], by = "V1") ## leftjoin by sample id



fig1J <- ggplot(brca %>% filter(mut_type != "NA"), aes(x = x_2d, y = y_2d, colour = mut_type)) +
  geom_point(data = brca, aes(x = x_2d, y = y_2d), colour = "grey") +
  geom_point() + 
  theme_classic() +
  ylab("umap_2") +
  xlab("umap_1") +
  labs(color = "", tag = "J") +
  scale_color_manual(labels = c("BRCA1", "BRCA1,BRCA2", "BRCA2"), 
                     values = c('#33a02c', '#e31a1c', '#ff7f00')) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

## umap stage
fig1K <- ggplot(subset(umap_cancertype, ajcc_pathologic_tumor_stage != "Not Available"), aes(x = x_2d, y = y_2d, colour = ajcc_pathologic_tumor_stage)) +
  geom_point(data = umap_cancertype, aes(x = x_2d, y = y_2d), colour = "grey") +
  geom_point() + 
  theme_classic() +
  ylab("umap_2") +
  xlab("umap_1") +
  labs(tag = "K", color = "") +
  scale_color_manual(labels = c("Stage I", "Stage II", "Stage III", "Stage IV"), 
                     values = c('darkgoldenrod3','dodgerblue3','yellow3','tomato2')) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom") 

## umap metastasis
fig1L <- ggplot(umap_cancertype, aes(x = x_2d, y = y_2d)) +
  geom_point(data = umap_cancertype, aes(x = x_2d, y = y_2d), colour = "grey") +
  geom_point(data = subset(umap_cancertype, ajcc_metastasis_pathologic_pm == "M0"), aes(x = x_2d, y = y_2d, colour = ajcc_metastasis_pathologic_pm)) + 
  geom_point(data = subset(umap_cancertype, ajcc_metastasis_pathologic_pm == "M1"), aes(x = x_2d, y = y_2d, colour = ajcc_metastasis_pathologic_pm)) + 
  ylab("umap_2") +
  xlab("umap_1") +
  labs(tag = "L", color = "") +
  scale_color_manual(labels = c("M0", "M1"), 
                     values = c('#a6cee3','#1f78b4')) +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "bottom")

layout <- '
AAAABBCCDD
AAAAEEEFFF
AAAAGGGG##
AAAAGGGG##
AAAAHHIIJJ
AAAAKKLL##
'

fig1A + plot_spacer() + fig1C + plot_spacer() + fig1E + fig1F + fig1G + fig1H + fig1I + fig1J + fig1K + fig1L + plot_layout(design = layout)

```

<p style="FONT-SIZE:10px; COLOR:#000000; LINE-HEIGHT:12px;">**Figure 1:
Classification of tumors by pathway disruptions.** a) Each of 377 selected
Reactome pathways (columns) is classified as disrupted if one or more genes is
mutated in the tumor sample (rows). Tumors are grouped by tissue of origin using
standardized abrreviations from the TCGA project. b) MCA analysis was performed
restricted to the 36 informative components. c) The resulting MCA was projected
down to 3 dimensions for the purpose of density based clustering (only the first
2 dimensions are shown). Each dot represents a tumor d) HDBSCAN was used to
perform hierarchical density based clustering tuned to minimize the fraction of
unassigned tumors. e) HDBSCAN clustered tumors were visualized on the UMAP
projection; tumors are colored by cluster membership (see legend). This results
in some tumors remaining unclassified (grey points). f) kNN was used to assign
membership to the remaining tumors. g) Violin plots of sample mutation counts
for tumors in each class. h) Projection of cancer type identity onto UMAP.
Abbrevs: BRCA = Breast cancer, COAD = Colorectal adenocarcinoma, HNSC = Head and
Neck Squamous Cell, PAAD = Pancreatic adenocarcinoma. i) Projection of breast
cancer subtypes onto the UMAP. j) projection of BRCA1/2 somatic mutation onto the
UMAP. k) projection of tumor stage onto the UMAP,
regardless of cancer type. l) Projection of metastatic status onto the UMAP.
Abbrevs: M0 = non-metastatic tumors, M1 = metastatic tumors.</p>

In order to classify tumors using this dataset, we used multiple correspondence
analysis (MCA) [@JSSv025i01]. First we determined the number of dimensions
containing useful information (**figure 1b**). Then we performed a UMAP
analysis, both in order to summarize the MCA graphically, and as a preprocessing
step to boost the performance of density based clustering. The resulting map was
notable for its lobed structure, with several large, robust projections
reproducible regardless of random seed setting. A representative version of this
UMAP is shown in **figure 1c** as a 2D projection. Following this spatial
mapping we attempted to define groupings of similar tumors within the spatial
map using HDBSCAN, which performs hierarchical clustering and provides metrics
of cluster stability and probabilities of cluster membership for each node.
However, HDBSCAN is sensitive to several parameters; key for our analysis are
the minimum number of tumor samples in a cluster that capture the maximum number
of tumor samples, measured by probability of membership of $\geq 5\%$ in at
least one class. Thus, we created a score metric as the fraction of classified
tumors with max probability $< 5\%$ in one class and chose a cluster size of 95
to minimize the score function (**figure 1d**). HDBSCAN with these settings
resulted in 12 distinct high-density clusters which we then projected onto the
UMAP (**figure 1e**). This classified <mark>x out of 7,607 tumors</mark> but
still resulted in a significant fraction of unclassified tumor samples. Since we
ultimately wish to be able to classify any tumor using this scheme, we performed
k Nearest Neighbors (kNN) analysis, which computes a similarity metric to every
tumor in the set and then lets the $k$ most similar tumors "vote" as to the
identity of the query tumor sample based on their classification labels. We set
$k$ to be the square root of the number of tumor samples (87). Using this
method, we assigned cluster membership to the remaining tumors (**figure 1f**).

#### Independence from tissue-of-origin.

Having defined tumors in terms of their pathway disruption profile, we sought to
understand whether different cancer types segregated into one or more
predominant classes. To our surprise, most cancer types (with some notable
exceptions) were not heavily biased in one class, and all well-represented
cancer types had representative tumors in every class (for example **figure 1h**
and full tumor profiles in **supplementary figure S1**, see also interactive
media from **supplementary file 1 (html)**), suggesting that, in principle at
least, our novel pathway-disruption classification method identifies molecular
pathologies largely independent of tissue-of-origin considerations. As an
example of one type of cancer that does have a biased pathway profile,
pancreatic adenocarcinoma (PAAD) was biased in class 9 & 10, which is
distinguished  by KRAS mutations among others. But even PAAD comprises tumors
from <mark>check; 7 out of 10 of the remaining classes</mark>, meaning that
patients hosting these tumors have potentially different underlying molecular
pathologies.

#### Independence from molecular and histological subtype.

Many cancers have molecular or histological subtypes defined based on gene
expression or other -omics profiles or pathology lab results. These subtypes
often have different standards of care owing to different overall drug
sensitivity (or other factors). If the histological subtypes represent true
molecular phenotypes, one predicts that histological subtypes should segregate
with our pathway-based classes, therefore providing support for the classes as
proxies for molecular pathology sub-typing. To our surprise, we find a similar
result to the previous analysis of cancer types projected onto the UMAP of
pathway disruptions. For example we projected annotations for each of the breast
cancer subtypes, composed of Triple-negative/Basal-like, Her2 positive,
normal-like, and lumenal A and B subtypes onto the UMAP. These are among the
most heavily studied molecular subtypes in cancer, which each have different
prognoses and standards of care. We did not observe any exclusive segregation by
pathway for these subtype annotations in GDC (**Figure 1i** and **supplemental
file 1**). We also projected histological subtype data for the remaining cancers
(see **supplementary figure S2** for the full set of projections); we find that
the subtypes, though often biased towards one or more classes, are almost never
exclusive to one of the classes. We interpret these data in aggregate to mean
that our pathway disruption classes do not correspond to molecular subtypes
within the parent cancer type. Furthermore, this result raises the possibility
that, molecular and histo-pathological subtypes could represent different
precursor cell origins within each cancer type. This interpretation still leaves
open the possibility that different precursor cell types, being susceptible to
different teratogens and selection pressures ( _e.g._ insulin dependence) could
be biased toward one or more of the pathway disruption profiles.

#### Independence from drivers of genome instability.

There are several well-known familial cancer-causing mutations that have been
interrogated extensively for differences in basic biology, survival and
treatment outcomes. However, the functions of these genes are related to risk
factors such as genome stability, proof-reading and DNA damage repair, and
telomere length. _BRCA1/2_ genes for example are key for DNA double-stranded
break repair [@moynihan-1999-brca1;@davies-2001-brca2] and confer risk for
breast, prostate and ovarian cancers (refs). The mechanism of risk is thought to
involve loss of heterozygosity [@brca-loh], so we projected the somatic
mutations for _BRCA1_ and _-2_ genes onto the UMAP, but did not observe
segregation of these mutations into specific clusters (**figure 1j**). We made
similar projections for the mismatch repair (MMR) genes _MSH2_, _MSH6_, _MLH_,
_PMS1_ and _PMS2_; _BRIP1_, _RAD15_, _CHEK2_ and _APC_. None of these genes
except for _APC_ exhibited any remarkable specificity with respect to cluster
assignment (**supplementary figure S3**). To look at other risk factors such as
maintenance of DNA methylation levels and telomere length we projected somatic
mutations of the _TET2_ and _TET3_ genes, plus telomerases 1, 2, and 3, and
observed similar lack of segregation by cluster (**supplementary figure S3**).

#### Independence of stage, mutation count and mutation profile.

Tumor staging is based on physico-pathological criteria, including diameter,
which can vary greatly in importance between different tissues. Stage is used
clinically as a proxy for advancement toward a more deadly state and metastasis.
Given these assumptions, it is possible that more advanced tumors have common
pathway disruption profiles, and the UMAP which features a series of lobe-like
structures on a common backbone of tumor samples (**figure 1c**) could in
principle also reflect progression through a series of stages. In support of
this hypothesis, the backbone culminates in a group of tumors (class 8) which
has nearly every pathway disrupted. However, outside of class 8 we don't observe
an obvious trend in the mutation count across the backbone of the UMAP (**figure
1g**). Nonetheless, to test the hypothesis that the molecular-pathway disruption
classes represent advancement through stages, we projected these data onto the
UMAP. Similar to tissue of origin and other categories of tumor, we did not
observe any bias amongst the stages into specific pathway disruption classes
(**figure 1k**), suggesting that stage is not an determining factor in pathway
disruption classes.

Finally, as a measure of tumor advancement, metastasis is the condition in which
certain phenotypic criteria are met. These phenotypes include (among others)
loss of differentiation, cell-cell contacts, epithelial to mesenchyme
transition, immune system evasion and tissue invasiveness. To determine whether
any of our classes correspond to an especially advanced stage of cancer accross
tissue types, we projected the metastasis data onto our UMAP, and surprisingly,
again, we observed an even distribution across classes (**figure 1l**). This
final observation suggests that our pathway-disruption classification is
dependent on particular combinations of gene mutations that hit different
pathway profiles that can each give rise to advanced stages of disease and
metastasis, regardless and independent of overall mutational burden.

### Each class is defined by a specific set of pathway disruptions

In order to visualize pathway disruption enrichment across all cancers
(pan-cancer), we created oncoprint graphs analagous to the commonly use method,
replacing genes with pathways (**figure 2**). A complete list of pathway
disruptions with percent mutated samples and top genes is provided in
**supplementary table ST1**). These reveal the broad importance of many well
known pathways that are disrupted in cancer, including <mark>PI3K/Akt
activation, DNA repair etc fix this with top 5 or so quoted pathway
labels</mark>. The top third of pathways drop off gradually in overall mutation
frequency, possibly owing to relative diversity among all tumors.
<div style="text-align: right"> Figure 2 Data </div>
```{r oncoprint-top100-plus-panel-data-generation, eval = FALSE}
tumor_clusters <- fread("../data/umap_2d_3d_coors.tsv") ## load in data
load("../data/gdc_reactome_path_analysis.Rda")

######################
## oncoprint top100 ##
######################
onco_color <- c("0" = "grey", "1" = "firebrick3")

top100path <- gdc_reactome_path_analysis %>% rowSums() %>% as.data.frame() %>% rownames_to_column(var = "path_name") %>% arrange(desc(.)) %>% .[1:100,] %>% .[, "path_name"] ## getting the top100 mutated pathways in class

top100path <- gdc_reactome_path_analysis[top100path, ] ## selecting the top 100 paths

top100path[top100path == "1"] <- "MUT" ## categorize boolean values
top100path[top100path == "0"] <- " "
  col <- c("MUT" = "firebrick3")
heatmap_legend <- list(title = "", at = c("MUT"),
                       labels = c("Mutation"))
alter_fun = list(
    background = function(x, y, w, h) {
        grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), 
            gp = gpar(fill = "#CCCCCC", col = NA))
    },
    # bug red
    MUT = function(x, y, w, h) {
        grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), 
            gp = gpar(fill = col["MUT"], col = NA))
    })
  column_title <- "Top 100 Pathway PanCancer"
  onco_res <- oncoPrint(top100path,
    alter_fun = alter_fun, col = col,
    column_title = column_title, heatmap_legend_param = heatmap_legend,
    remove_empty_columns = F, remove_empty_rows = F, right_annotation = NULL,
    top_annotation = NULL
    )
  
clust_col_order <- names(top100path[,onco_res@column_order])
path_row_order <- rownames(top100path[onco_res@row_order,])

onco_ggplot <- gdc_reactome_path_analysis %>% .[rownames(.) %in% path_row_order,] %>% ## only include the top 10 paths
  rownames_to_column(var = "path_name") %>% ## path_name to col
  gather(., key = "sample_id", value = "mutation_status", -path_name) %>% ## wide to long format
  mutate(path_name = factor(.$path_name, levels = c(rev(path_row_order))), ## factor to force oncoprint order
         sample_id = factor(.$sample_id, levels = c(clust_col_order)))
save(onco_ggplot, file = "onco_ggplot.Rda")


#####################
## oncoprint panel ##
#####################
clust_col_order <- list()
path_row_order <- list()

for(n in 1:12) {
  class_spec_sample_id <- tumor_clusters %>% .[,c("V1", "knn_clust")] %>% filter(knn_clust == n) %>% .[,.$V1] ## get class spec sample ids
  top10path <- gdc_reactome_path_analysis %>% .[, names(.) %in% class_spec_sample_id] %>% rowSums() %>% as.data.frame() %>% rownames_to_column(var = "path_name") %>% arrange(desc(.)) %>% .[1:10,] %>% .[, "path_name"] ## getting the top10 mutated pathways in class
  top10path <- gdc_reactome_path_analysis[top10path, class_spec_sample_id]
  top10path[top10path == "1"] <- "MUT" ## categorize boolean values
  top10path[top10path == "0"] <- " "
  col <- c("MUT" = "firebrick3")
heatmap_legend <- list(title = "", at = c("MUT"),
                       labels = c("Mutation"))
alter_fun = list(
    background = function(x, y, w, h) {
        grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), 
            gp = gpar(fill = "#CCCCCC", col = NA))
    },
    # bug red
    MUT = function(x, y, w, h) {
        grid.rect(x, y, w-unit(0.5, "mm"), h-unit(0.5, "mm"), 
            gp = gpar(fill = col["MUT"], col = NA))
    })
  column_title <- "test"
  onco_res <- oncoPrint(top10path,
    alter_fun = alter_fun, col = col,
    column_title = column_title, heatmap_legend_param = heatmap_legend,
    remove_empty_columns = F, remove_empty_rows = F, right_annotation = NULL,
    top_annotation = NULL
    )
  clust_col_order[[n]] <- names(top10path[,onco_res@column_order])
  path_row_order[[n]] <- rownames(top10path[onco_res@row_order,])
}

save(clust_col_order, path_row_order, file = "onco_clust.Rda")
```

```{r figure-2-plot, fig.width=72, fig.height=48}
## top100 pathways pancancer oncoprint
load("../data/onco_ggplot.Rda") ## load data
onco_color <- c("0" = "grey", "1" = "firebrick3") ## color scheme for oncoprint

fig2A <- ggplot(data = onco_ggplot, aes(x=sample_id, y=path_name)) + 
  geom_tile(aes(fill = as.character(mutation_status))) +
  scale_fill_manual(values = onco_color) +
  xlab(label = "sample_id") +
  labs(tag = "A") +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.line = element_blank(),
        legend.position = "none") 

## top10 pathways for each cluster
load("../data/onco_clust.Rda") ## load in data
load("../data/gdc_reactome_path_analysis.Rda")

plots_onco <- list() ## creating a list to store all the ggplots in the for loop below
onco_color <- c("0" = "grey", "1" = "firebrick3") ## color scheme for oncoprint
figure_labs <- LETTERS[seq(from = 2, to = 13)]
for(n in 1:12) {
  onco_ggplot <- gdc_reactome_path_analysis %>% .[rownames(.) %in% path_row_order[[n]],] %>% ## only include the top 10 paths
  .[,names(.) %in% clust_col_order[[n]]] %>% ## only include specific tumor class
  rownames_to_column(var = "path_name") %>% ## path_name to col
  gather(., key = "sample_id", value = "mutation_status", -path_name) %>% ## wide to long format
  mutate(path_name = factor(.$path_name, levels = c(rev(path_row_order[[n]]))), ## factor to force oncoprint order
         sample_id = factor(.$sample_id, levels = c(clust_col_order[[n]])))
  plots_onco[[n]] <- ggplot(data = onco_ggplot, aes(x=sample_id, y=path_name)) + ## ggplot code
  geom_tile(aes(fill = as.character(mutation_status))) +
  scale_fill_manual(values = onco_color) +
  xlab(label = paste0("Class ", n)) +
  labs(tag = figure_labs[n]) +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.line = element_blank(),
        legend.position = "none") 
}

## setting patchwork panel layout
layout <- '
AABCD
AABCD
AABCD
AAEFG
AAEFG
AAEFG
AAHIJ
AAHIJ
AAHIJ
AAKLM
AAKLM
AAKLM
'

fig2A + plots_onco[[1]] + plots_onco[[2]] + plots_onco[[3]] + plots_onco[[4]] + plots_onco[[5]] + plots_onco[[6]] + plots_onco[[7]] + plots_onco[[8]] + plots_onco[[9]] + plots_onco[[10]] + plots_onco[[11]] + plots_onco[[12]] + plot_layout(design = layout)
```
<p style="FONT-SIZE:10px; COLOR:#000000; LINE-HEIGHT:12px;"><b>Figure 2:
Pan-cancer enrichment of pathway disruptions.</b> "Oncoprint" style graphic in
which each column represents a tumor and each row represents a pathway. The
pathways are ranked such that the pathway with the most disruptions across all
7,607 tumors is the first row _etc._ and tumors are ordered to prioritize the
top pathway disruptions successively. All 377 pathways are shown. Red signifies
pathway disrupted pathways, grey signifies not disrupted.</p>

To discover what pathways drive the UMAP clustering, we calculated enrichment
for each pathway _in-class_ relative to all other classes combined (see methods)
and ranked pathways for each class from highest to lowest enrichment. We
visualized the enrichment as a heatmap (**figure 3**). Using this approach, we
identified 3 pathways highly enriched (enrichment score > 0.2, see methods) in
class 1, 11 pathways enriched in class 2, 9 pathways in class 3, 12 pathways in
class 4, 23 pathways in class 5, 31 pathways in class 6, 35 pathways in class 7,
0 pathways in class 8 (for which most pathways are putatively disrupted), 25
pathways in class 9, 28 pathways in class 10, 24 pathways in class 11, and 29
pathways in class 12. Some pairs of classes, in particular classes 6 & 7,
classes 9 & 10, and 11 & 12 were highly concordant in terms of which pathways
had enrichment. To explore the specific pathways marking each class, we
projected disruptions for each of the 377 pathways onto the UMAP
(**supplementary figure S4**). <mark>Class x</mark> was distinguished by
metabolic pathways including RNA and protein biosynthesis (**supplementary
figure S4**). Most class 4 enriched pathways were not unique to this class but
shared in class 10 & 12, suggesting that this grouping reflects either greater
heterogeneity or possibly a lower stage of tumor evolution (as distinct from
clinical ``stage''). Classes 9-12 have in common mutations in extracellular
signaling, intracellular signaling, and immune-related pathways (see **figures
3** and **supplementary figure S4**). Cluster group 6-7 had the highest pathway
enrichment levels of the three, having mutations in hedgehog signaling,
$\beta$-catenin degradation, cellular response to hypoxia, regulation of cell
cycle and apoptosis among others. As cluster groups 9-12 and 6-7 were the most
distinct of the 12, we chose to investigate these two groups further.
<div style="text-align: right"> Figure 3 data </div>
```{r group-pathway-enrichment-data-generation, eval=FALSE}
tumor_clusters <- fread("../data/umap_2d_3d_coors.tsv") ## load in data
tumor_clusters <- tumor_clusters %>% as.data.frame() %>% dplyr::select(V1, knn_clust) ## select the relevant columns

## Getting the total number of tumor samples in each cluster
n_tumor_in_clust <- as.data.frame(table(tumor_clusters$knn_clust)) %>% ## table the number of tumors in each cluster
  'colnames<-' (c("clust", "tot_n_clust")) %>% ## move cluster number to rownames
  mutate(clust = as.character(clust)) ## convert factor to character


## Organizing rBeta data
load("../data/gdc_reactome_path_analysis.Rda")
clust_group <- n_tumor_in_clust$clust
cluster_path_count <- list()
for(n in clust_group) {
  print(n)
  sample_id <- tumor_clusters %>% filter(knn_clust == n) %>% ## filter for tumor class
    .[,"V1"] ## get sample_ids
  path_mutation <- gdc_reactome_path_analysis %>% .[,names(.) %in% sample_id] %>% ## getting the tumor samples from path analysis
    rowSums() %>% as.data.frame() %>% ## getting the observed mutation count of tumor class
    'colnames<-' (n) ## change colname to cluster number
  cluster_path_count[[n]] <- path_mutation
  rm(path_mutation, sample_id, n)
  gc()
}
cluster_path_count <- do.call(cbind, cluster_path_count) ## Combine the list together

global_path_mut <- cluster_path_count %>% rowSums() %>% as.data.frame() %>% ## global pathway mutation count
  .[rep(1:ncol(.), each = 12)] %>% ## replicate the column 12 times for the 12 tumor classes
  'colnames<-' (clust_group) %>% ## rename column with tumor class
  rownames_to_column(var = "path") %>% ## move rownames to column (pathways) %>%
  gather(., key = "clust", value = "global_mut", -path)

rbeta_path_analysis <- cluster_path_count %>% rownames_to_column(var="path") %>%
  gather(key = "clust", value = "clust_mut", -path) %>% ## Wide to long format
  left_join(., global_path_mut, by = c("path", "clust")) %>% ## add global mutation count 
  left_join(., n_tumor_in_clust, by = "clust") %>% ## add total number of tumor samples per clust 
  mutate(global_sample = sum(n_tumor_in_clust$tot_n_clust)) ## add total number of tumor samples

rm(global_path_mut, cluster_path_count, n_tumor_in_clust, clust_group, tumor_clusters, gdc_reactome_path_analysis)
gc()

enrich_prob_uncertainity_in_bg <- data.frame(path = rbeta_path_analysis$path,
                                            clust = rbeta_path_analysis$clust,
                                            fg_mut = rbeta_path_analysis$clust_mut,
                                            fg_not_mut = rbeta_path_analysis$tot_n_clust - rbeta_path_analysis$clust_mut,
                                            bg_mut = rbeta_path_analysis$global_mut,
                                            bg_not_mut = rbeta_path_analysis$global_sample - rbeta_path_analysis$global_mut
                                            )
fg_mut_rate <- list()
bg_mut_rate <- list()
for(n in 1:max(as.numeric(unique(enrich_prob_uncertainity_in_bg$clust)))) {
  print(n)
  df <- subset(enrich_prob_uncertainity_in_bg, clust == n)
  fg_denominator <- nrow(df) * (df$fg_mut[1] + df$fg_not_mut[1]) ## n_path * tot_n_clust_x
  fg_numerator <- sum(df$fg_mut)
  bg_denominator <- nrow(df) * (df$bg_mut[1] + df$bg_not_mut[1])
  bg_numerator <- sum(df$bg_mut)
  fg_mut_rate[[n]] <- fg_numerator / fg_denominator
  bg_mut_rate[[n]] <- bg_numerator / bg_denominator
  rm(df)
}

fg_mut_rate <- fg_mut_rate %>% do.call(rbind, .) %>% as.data.frame() %>% ## rbind the mut rates from list
  'colnames<-' ("fg_mut_rate") %>% ## rebind col names
  rownames_to_column(var = "clust") ## move rownames to col
bg_mut_rate <- bg_mut_rate %>% do.call(rbind, .) %>% as.data.frame() %>% ## rbind the mut rates from list
  'colnames<-' ("bg_mut_rate") %>% ## rebind col names
  rownames_to_column(var = "clust") ## move rownames to col

enrich_prob_uncertainity_in_bg <- enrich_prob_uncertainity_in_bg %>% left_join(., fg_mut_rate, by = "clust") %>%
  left_join(., bg_mut_rate, by = "clust") %>%
  mutate(rel_odds = .$fg_mut_rate / .$bg_mut_rate) %>%
  mutate(denominator = (.$bg_mut * .$rel_odds) + (.$bg_not_mut / .$rel_odds)) %>%
  mutate(x_adj =(.$fg_mut + .$fg_not_mut) * (.$bg_mut * .$rel_odds) / .$denominator) %>%
  mutate(y_adj = (.$fg_mut + .$fg_not_mut) - .$x_adj)
rm(fg_mut_rate, bg_mut_rate, rbeta_path_analysis)
gc()

moderate_fg <- apply(enrich_prob_uncertainity_in_bg[,c("fg_mut", "fg_not_mut")], MARGIN = 1, function(x) {
  rbeta(n = 1e5, shape1=x["fg_mut"], shape2=x["fg_not_mut"])
})

moderate_bg <- apply(enrich_prob_uncertainity_in_bg[,c("x_adj", "y_adj")], MARGIN = 1, function(x) {
  rbeta(n = 1e5, shape1=x["x_adj"], shape2=x["y_adj"])
})

## ModerateEnrichment probability 
prob <- moderate_fg > moderate_bg ## boolean data
mean_prob <- apply(prob, MARGIN = 2, function(x) {
  mean(x)
})
rm(prob)
gc()
## Moderate Difference
CI <- 0.025 ## setting a 95% confidence interval

diff_fg_bg <- moderate_fg - moderate_bg ## fg - bg
rm(moderate_fg, moderate_bg, prob)
gc()

sig_fg_bg <- as.data.frame(apply(diff_fg_bg, MARGIN = 2, function(x) {
  sum(x < 0 ) ## out of 1e5 simulations, how many times is fg-bg less than 0
}))
quant_diff <- t(apply(diff_fg_bg, MARGIN = 2, function(x) {
  quantile(x, c(CI, 0.5, 1-CI))
}))

moderate_enrich_prob <- data.frame(path = enrich_prob_uncertainity_in_bg$path,
                                   clust = enrich_prob_uncertainity_in_bg$clust,
                                   probability = mean_prob,
                                   signif = sig_fg_bg[,1],
                                   difference = quant_diff[,2],
                                   lower = quant_diff[,1],
                                   upper = quant_diff[,3])
rm(enrich_prob_uncertainity_in_bg, mean_prob, quant_diff, sig_fg_bg, diff_fg_bg)
gc()
#save(moderate_enrich_prob, file = "moderate_enrich_prob.Rda")
load("../data/moderate_enrich_prob.Rda")

diff_heatmap <- moderate_enrich_prob[,c("path", "clust", "difference")]
diff_heatmap <- spread(diff_heatmap, key = "clust", value = "difference")
diff_heatmap <- diff_heatmap %>% remove_rownames() %>% column_to_rownames(var="path")

effect_size <- pheatmap(diff_heatmap,
         show_rownames = F,
         show_colnames = T,
         cluster_cols = T,
         cluster_rows = T,
         cellwidth = 10,
         cellheight = 0.76,
         color = rev(c('#7f3b08','#b35806','#e08214','#fdb863','#fee0b6','white', '#d8daeb','#b2abd2','#8073ac','#542788','#2d004b')),
         breaks = c(seq(from = -1, -0.2, 0.2), -0.0001, 0.0001, seq(from = 0.2, to = 1, by = 0.2)),
         clustering_method = "ward.D2")

effect_row_order <- effect_size$tree_row$order
effect_row_order <- diff_heatmap %>% .[effect_row_order,] %>% rownames() ## getting row order of pheatmap

effect_heatmap <- diff_heatmap %>% rownames_to_column(var = "path_name") %>% 
  gather(., key = "cluster", value = "effect_size", -path_name) 

effect_heatmap$path_name <- factor(effect_heatmap$path_name, levels = c(rev(effect_row_order)))
effect_heatmap$cluster <- factor(effect_heatmap$cluster, levels = c(1:12))
effect_segments <- c(seq(from = -1, -0.2, 0.2), -0.0001, 0.0001, seq(from = 0.2, to = 1, by = 0.2))

effect_heatmap <- effect_heatmap %>% mutate(discrete_color = case_when(
  effect_size > effect_segments[1] & effect_size <= effect_segments[2] ~ "1",
  effect_size > effect_segments[2] & effect_size <= effect_segments[3] ~ "2",
  effect_size > effect_segments[3] & effect_size <= effect_segments[4] ~ "3",
  effect_size > effect_segments[4] & effect_size <= effect_segments[5] ~ "4",
  effect_size > effect_segments[5] & effect_size <= effect_segments[6] ~ "5",
  effect_size > effect_segments[6] & effect_size <= effect_segments[7] ~ "6",
  effect_size > effect_segments[7] & effect_size <= effect_segments[8] ~ "7",
  effect_size > effect_segments[8] & effect_size <= effect_segments[9] ~ "8",
  effect_size > effect_segments[9] & effect_size <= effect_segments[10] ~ "9",
  effect_size > effect_segments[10] & effect_size <= effect_segments[11] ~ "10",
  effect_size > effect_segments[11] & effect_size <= effect_segments[12] ~ "11"
))
effect_heatmap$discrete_color <- factor(effect_heatmap$discrete_color, levels = c(1:11))

save(effect_heatmap, file = "effect_heatmap.Rda")
```

```{r group-pathway-enrichment heatmap}
load("../data/effect_heatmap.Rda") ## load data

effect_colors = c("1" = "#2d004b", "2" = "#542788", "3" = "#8073ac", "4" = "#b2abd2", "5" = "#d8daeb", "6" = "white", "7" = "#fee0b6", "8" = "#fdb863", "9" = "#e08214","10" = "#b35806","11" ="#7f3b08") ## color scheme

fig3 <- ggplot(data = effect_heatmap, aes(x=cluster, y=path_name)) + 
  geom_tile(size = 0.6, aes(fill = discrete_color)) +
  scale_fill_manual(values = effect_colors) +
  theme(axis.text.x = element_text(size = 12),
        axis.text.y = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        legend.position = "none")

fig3_legend <- ggplot(data = data.frame(x = 1:11, y = 1), aes(x = x, y = y, fill = as.factor(x))) +
  geom_bar(stat = "identity", width = 1) +
  theme_classic() +
  scale_fill_manual(values = effect_colors) + 
  scale_x_continuous(labels= c("-1", "-0.5","0", "0.5", "1")) +
  theme(legend.position = "none",
        axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.line = element_blank())

## setting patchwork panel layout
layout <- '
AAAAA
AAAAA
AAAAA
AAAAA
AAAAA
AAAAA
AAAAA
AAAAA
#BBB#
'

fig3 + fig3_legend + plot_layout(design = layout)
```

<p style="FONT-SIZE:10px; COLOR:#000000; LINE-HEIGHT:12px;">**Figure 3. Pathway-specific enrichment within clusters.** Heatmap shows relative enrichment of each pathway (rows) within each numbered cluster (columns). The annotation at left indicates the Reactome pathway group at the second level (check this).</p>

### Gene-Pathway redundancy contributes to cluster identity

<mark>This section will have to be rewritten to reflect Simon's new clusters.</mark>

Since pathways like these share many common genes, we investigated whether
tumors in this class  have multiple mutations across signaling pathways, or
whether a small number of driver genes account for enrichment. The split between
clusters 9 & 10 and 11 & 12 is largely driven by mutations in _PI3K_ and its
homologs and _Ras_ genes, respectively (compare _PIK3CA_ and _KRAS_ panels of
**supplementary figure S4**). Furthermore, clusters 9 and 11 are distinguished
from 10 and 12 by general lack of P53 mutations (compare **Figure 1f** with the
_TP53_ panel of **Supplementary Figure S4**).
 
In contrast to clusters 9-12, the group represented by clusters 6 & 7 exhibited
the highest pathway enrichment scores of the three major categories, though
there is a significant number of tumors spatially distributed between cluster
groupings 9-12 and 6-7 on the UMAP projection that have pathway characteristics
of both. The top ten distinguishing pathways of class 6/7 tumors were "cellular
response to hypoxia," "regulation of mitotic cell cycle," "assembly of the
pre-replicative complex," "metabolism of polyamines," "degradation of
$\beta$-catenin by the destruction complex", "tumor necrosis factor receptor 2
non-canonical _NF-$\kappa$B_ pathway", " _AUF1_ (hnRNP D0) binds and
destabilizes RNA", "regulation of apoptosis" and 3 other pathways associated
with aspects of hedgehog signaling. To determine once again whether the genes
driving pathway enrichment were commonly shared among these pathways we plotted
and performed unsupervised clustering on tumors with an individual mutation
matrix using every gene in each of the ten pathways, grouped by shared pathway
membership (**figure 4**). Surprisingly, this class of tumors was not
predominated by 1 or more highly mutated genes across all cancers. Instead,
there is a whole set of genes in the proteosomal pathway whose activity
contributes to all the aforementioned pathways, and the majority of tumors in
these classes have a single mutation in one of these genes. Nonetheless, a
significant fraction <mark>(~1/3 <-- need numbers!)</mark> do not have
proteasomal mutations but instead harbor mutations in genes representing a
majority of the same pathways, suggesting that most if not all of the pathways
are individually important and that proteasomal mutations may be sufficient to
drive the net effect of the other mutations in this cluster. We annotated the
tumors in this class according to tissue of origin to determine whether
proteasome mutations have a tissue-of-origin (or cancer-specific) bias, and
indeed we could observe some clustering. These data could explain how
tissue-specific gene mutation signatures might result in general cancer
phenotypes by converging at the pathway level.

```{r figure-4-data-generation, eval = FALSE}
tumor_clusters <- fread("../data/umap_2d_3d_coors.tsv") ## load in data
load("../data/gene2_level2_lookup.Rda")
load("../data/pathname2level2.Rda")
load("../data/gdc_reactome_path_analysis.Rda")
load("../data/gene_data.Rda")

## Parameters ##
filtered_class <- c(6,7) ## filtering for only class 6 and 7 samples
n_path <- 10 ## looking at the top10 path

## Data Generation ##
class_spec <- tumor_clusters %>% dplyr::select(V1, knn_clust) %>% filter(knn_clust %in% filtered_class) %>% .[,.$V1] ## sample_id for class specific

toppath <- gdc_reactome_path_analysis %>% .[, names(.) %in% class_spec] %>% rowSums() %>% as.data.frame() %>% rownames_to_column(var = "path_name") %>% arrange(desc(.)) %>% .[1:n_path,] %>% .[, "path_name"] ## getting the top10 mutated pathways

genes_in_path <- gene2level2_lookup %>% .[, match(pathname2level2$PATH_ID, names(.))] %>% ## match path ids
  'colnames<-' (pathname2level2$PATH_NAME) %>% ## replace with path_names
  dplyr::select(c(toppath)) %>% ## filter by top 10 pathways
  rownames_to_column(var = "ensembl_gene_id") %>% left_join(., gene_data[,c("ensembl_gene_id", "external_gene_name")], by = "ensembl_gene_id") %>% column_to_rownames(var = "external_gene_name") %>% dplyr::select(-ensembl_gene_id) %>% ## replace ensembl gene id with gene name
  .[rowSums(.[, -1])>0,]

df <- upset(genes_in_path, 
      sets = names(genes_in_path),
      mb.ratio = c(0.55, 0.45), 
      order.by = "freq", 
      point.size = 3.5,
      line.size = 2.0,
      mainbar.y.label = "Gene Intersection",
      sets.x.label = "Total Gene Per Pathway",
      text.scale = c(2.5, 2.4, 2, 2.1, 2.4, 1.5),
      number.angles = 0)

make_combinations <- function(x) {
  l <- length(x)
  mylist <- lapply(1:l, function(y) {
    combn(x, y, simplify = FALSE)
  })
  mylist
}

path_combo <- make_combinations(names(genes_in_path)) ## possible pathway combinations from 1 to 10

top10path <- genes_in_path %>% rowSums() %>% as.data.frame() %>% ## getting the sums of genes in pathway
  'colnames<-' ("sum_top10") %>% ## rename col
  rownames_to_column(var = "gene_name") ## make rownames a col

path_match_in_set <- list() ## making a list for matchings in sets

for(n in 1:10) {
  message(n)
  path_combo_set <- path_combo[[n]] ## selecting combination set (1 combo, 2 combo, 3 combo, etc)
  path_match_within_set <- list() ## making a list for matching within sets
  for(p in 1:length(path_combo_set)) {
    message(p)
    path_name <- path_combo_set[[p]] ## selecting specific combination pathway names
    spec_combo <- genes_in_path %>% dplyr::select(all_of(path_name)) ## select pathway combo
    if(length(spec_combo) > 1) {
      spec_combo <- spec_combo %>% rowSums() %>% as.data.frame() %>% ## rowSums the pathway to get the gene total membership
        'colnames<-' ("combo") %>% filter(combo == n) ## rename col and filter for combo that is n (n combo)
    }
    if(length(spec_combo) == 1) {
      names(spec_combo) <- "combo"
    }
    spec_combo <- spec_combo %>% rownames_to_column(var = "gene_name") %>% left_join(., top10path, by = "gene_name")
    spec_combo$match <- spec_combo$combo == spec_combo$sum_top10
    any_matches <- as.character(unique(spec_combo$match)) ## are there any matches?
    if(!"TRUE" %in% any_matches) { ## if no matches then skip
      next
      }
    if("TRUE" %in% any_matches) { ## if are matches then do this
      path_match_within_set[[p]] <- spec_combo %>% filter(match == "TRUE") %>% dplyr::select(gene_name) %>% ## filter for trues and select only gene names
        mutate(path_combo = paste0(paste(path_name, collapse=", "))) ## insert combo type
    }
  }    
  if(length(path_match_within_set) > 0) {
    path_match_in_set[[n]] <- path_match_within_set %>% do.call(rbind, .) %>% as.data.frame()
  }
}
path_match_in_set <- path_match_in_set %>% do.call(rbind, .) %>% as.data.frame() ## combine the list together

save(path_match_in_set, file = "path_match_in_set.Rda")
load("../data/path_match_in_set.Rda")

### getting clustering order from pheatmap
most_genes <- path_match_in_set %>% .$path_combo %>% table() %>% as.data.frame() %>% arrange(desc(Freq)) %>% ## arranging pathway combination based on most genes involved
  'colnames<-' (c("path_combo", "Freq")) %>% dplyr::select(path_combo) %>%
  rownames_to_column(var = "order") 

gene_order <- path_match_in_set %>% left_join(., most_genes, by = "path_combo") %>% ## left join the data together
  arrange(order) %>% .$gene_name ## getting the gene name order


## now need to get the gene mutation data and tumor samples 
load("../data/gdc_tumor_coding_complete.Rdata")

## Booleanize mutation count for each tumor sample
missense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$missense) 
splice <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$splice)
frameshdel <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshdel)
frameshins <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshins)
utr3 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr3)
utr5 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr5)
nonstop <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonstop)
nonsense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonsense)

gdc_mut_count_patient <- missense + splice + frameshdel + frameshins + utr3 + utr5 + nonstop + nonsense ## add up all the mutation count
rm(missense, splice, frameshdel, frameshins, utr3, utr5, nonstop, nonsense, gdc_tumor_coding_complete)
gc()
gdc_mut_count_patient[gdc_mut_count_patient > 0] <- 1 ## booleanize data, anything greater than 1 is equal to 1

filtered_mut_data <- gdc_mut_count_patient %>% rownames_to_column(var = "ensembl_gene_id") %>% ## ensembl id to col
  left_join(., gene_data[,c("ensembl_gene_id", "external_gene_name")], by = "ensembl_gene_id") %>% ## left_join data by ensembl_gene_id
  column_to_rownames(var = "external_gene_name") %>% dplyr::select(-ensembl_gene_id) %>% ## replace ensembl_id with gene_name
  .[rownames(.) %in% gene_order, names(.) %in% class_spec] ## filter by genes in top10 path of filtered class and tumor samples associated with it
rm(gdc_mut_count_patient)
gc()

save(filtered_mut_data, gene_order, file = "filtered_mut_data.Rda")

fig4_pheatmap <- pheatmap(filtered_mut_data[gene_order,],
         show_rownames = T,
         show_colnames = F,
         treeheight_row = 0,
         treeheight_col = 0,
         cellheight = 5,
         cellwidth = 1,
         cluster_cols = T,
         cluster_rows = F, ## turn off row clustering
         annotation_legend = F,
         legend_breaks = c(0, 0.25, 0.75, 1),
         legend_labels = c("","Not Mut", "Mut", ""),
         color = c("grey", "firebrick3"),
         clustering_method = "ward.D2",
         fontsize = 10,
         legend = T)

## getting sample orders
pheatmap_order <- fig4_pheatmap$tree_col$order ## getting sample clustering order
sample_order <- names(filtered_mut_data) ## getting the sample_id original order 
sample_order <- sample_order[pheatmap_order] ## reorder cancer by heatmap order

## stopped here ## ignore bottom using as reference

class67_gene_2_path_heatmap <- filtered_mut_data %>% rownames_to_column(var = "gene_name") %>%  ## move rownames to column
  gather(., key = "sample_id", value = "membership", -gene_name) ## wide to long format
class67_gene_2_path_heatmap$membership <- as.character(class67_gene_2_path_heatmap$membership) ## turn boolean values into categories

## set levels based on clustering algorithm above
class67_gene_2_path_heatmap$gene_name <- factor(class67_gene_2_path_heatmap$gene_name, levels = c(rev(rownames(filtered_mut_data))))
class67_gene_2_path_heatmap$sample_id <- factor(class67_gene_2_path_heatmap$sample_id, levels = c(sample_order))

class67_gene_2_path_heatmap <- left_join(class67_gene_2_path_heatmap, path_match_in_set, by = "gene_name") ## add in path_combo for faceting 
save(class67_gene_2_path_heatmap, file = "class67_gene_2_path_heatmap.Rda") 
rm(list = ls(all.names = T))
gc()
```

```{r figure-4, eval = FALSE}
load("../data/class67_gene_2_path_heatmap.Rda") ## load data
load("../data/path_match_in_set.Rda")

### getting facet ordering
most_genes <- path_match_in_set %>% .$path_combo %>% table() %>% as.data.frame() %>% arrange(desc(Freq)) %>% ## arranging pathway combination based on most genes involved
  'colnames<-' (c("path_combo", "Freq")) %>% 
  rownames_to_column(var = "order") 

fig4 <- ggplot(data = class67_gene_2_path_heatmap, aes(y=gene_name, x=sample_id, fill = membership)) + 
  geom_tile(size = 0.6) +
  scale_fill_manual(values = c("grey", "firebrick3")) +
  #facet_grid(factor(path_combo, levels = most_genes$path_combo) ~ ., scales = "free", space = "free") +
  #scale_y_discrete(position = "left") +
  labs(tag = "A") +
  ylab("") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.ticks = element_blank(),
        strip.text.y = element_blank(),
        strip.background = element_blank(),
        legend.position = "none") 
fig4
ggsave("fig4_ggplot_heatmap.png", height = 100, width = 17, limitsize = FALSE)

## fixing the text cutoff in heatmap
 fig1Ag <- ggplotGrob(fig1A)
 
 for(i in which(grepl("strip-l", fig1Ag$layout$name))){
   fig1Ag$grobs[[i]]$layout$clip <- "off"
 }
 
 fig1A <- as.ggplot(fig1Ag)
 rm(fig1Ag)

```

![](../figures/figure4_temp.png)
<p style="FONT-SIZE:10px; COLOR:#000000; LINE-HEIGHT:12px;">**Figure 4: Some genes with multiple pathway memberships drive clustering.** Each column is a tumor in class 6 or 7, each row represents a gene with mutation (red) or not mutated (grey). The genes are organized according to which of the top distinguishing pathways for this class that they effect. Some genes affect multiple pathways, as indicated by the bars at left. The highlighted genes at right are all proteasomal subunits.</p>

### A combination of common driver mutations and tissue-specific pathway genes create tissue-specific class signatures

<mark>As discussed previously, prior efforts to extract common signatures from
pan-cancer datasets met with difficulty in distinguishing tumor samples from
their tissue-specific -omics data signatures. Given our pathway-disruption based
classification, this raises the question, are tumor phenotypes driven entirely
by common driver genes, or by "silent" tissue-specific effectors ( _i.e._ too
few samples to detect above statistical significance thresholds), or a
combination of both? To answer this question, we ranked the top cluster-specific
pathways for each cluster, and then tabulated the most frequently mutated genes
for each pathway as a function of cancer type (**Table 1** and **Supplementary
tables ST2-ST13**). We find that in most clusters the top pathways are
determined by mutations in a handful of driver genes, TP53, PIK3CA, KRAS etc.
However in many cancers we find divergence outside these top driver mutations,
wherein etc etc</mark>
<div style="text-align: right"> Table 1 Data </div>
```{r table-1-data-generation, eval = FALSE}
tumor_clusters <- fread("../data/umap_2d_3d_coors.tsv") ## load in data
load("../data/moderate_enrich_prob.Rda") ## load data
load("../data/gene2_level2_lookup.Rda")
load("../data/gene_data.Rda")
load("../data/gdc_gene_exp_adj.Rda")
load("../data/pathname2level2.Rda")

id_to_names <- gene2level2_lookup %>% rownames_to_column(var = "ensembl_gene_id") %>% ## convert ensembl to gene name
  left_join(., gene_data[,c("ensembl_gene_id", "external_gene_name")], by = "ensembl_gene_id") %>%
  remove_rownames() %>% column_to_rownames(var = "external_gene_name") %>%
  dplyr::select(-ensembl_gene_id) %>%
  t() %>% as.data.frame() %>% rownames_to_column(var = "PATH_ID") %>% ## convert path_id to path_name
  left_join(., pathname2level2, by = "PATH_ID") %>%
  remove_rownames() %>% column_to_rownames(var = "PATH_NAME") %>%
  dplyr::select(-PATH_ID) %>%
  t() %>% as.data.frame()

gene_mutation <- gdc_gene_exp_adj %>% rownames_to_column(var = "ensembl_gene_id") %>% ## convert ensembl to gene name
  left_join(., gene_data[,c("ensembl_gene_id", "external_gene_name")], by = "ensembl_gene_id") %>%
  remove_rownames() %>% column_to_rownames(var = "external_gene_name") %>%
  dplyr::select(-ensembl_gene_id)

## parameters ##
enrich_cutoff <- 0.30
howmanypaths <- 5 ## top _ pathways; how many top pathways do you want to see
show_perc <- TRUE
show_n <- TRUE
mutated_gene_position <- 1
################

clust_spec_enrich <- moderate_enrich_prob %>% filter(signif <= 2500 & difference >= enrich_cutoff) ## signif is 95% CI with a specified cutoff

clust_spec_enrich_top5 <- list()
for(n in unique(clust_spec_enrich$clust)) {
  message(n)
  clust_spec_enrich_top5[[n]] <- clust_spec_enrich %>% filter(clust == n) %>% ## filter for specific cluster
    arrange(desc(difference)) %>% ## arrange enrichment score in decreasing order 
    dplyr::select(path, clust) %>% .[1:5,]
}
clust_spec_enrich_top5 <- clust_spec_enrich_top5 %>% do.call(rbind, .) %>% as.data.frame() ## combine list from for loop
rm(clust_spec_enrich)

## finding the top mutated gene for each cluster in each pathway for the top 5 pathways 
type <- unique(tumor_clusters$project_code) ## type of cancer type in data
cluster_spec_enrich_table <- list() ## list for forloop

for(d in unique(clust_spec_enrich_top5$clust)) {
  message(d)
  clust_toppath <- clust_spec_enrich_top5 %>% filter(clust == d) %>% .$path %>% as.character() ## get the top pathway names for spec class
  class_spec <- tumor_clusters %>% dplyr::select(V1, knn_clust, project_code) %>% filter(knn_clust %in% d) ## get the smaple ids and cancer type for spec class
  top_genes_in_path_cancer_type <- list()
  for(n in clust_toppath) { ## for each of the top paths do this
    message(n)
    gene_in_path <- id_to_names %>% .[,which(names(.) == n)] %>% as.data.frame() %>% ## select pathway
    'rownames<-' (rownames(id_to_names)) %>% 
    filter(. == 1) %>% ## filter for genes that are in that pathway
    rownames(.)
    top_gene_in_path <- list()
    for(t in type) { ## looping through cancer type within class
      message(t)
      class_spec <- tumor_clusters %>% dplyr::select(V1, knn_clust, project_code) %>% filter(knn_clust == d) ## sample_id for class
      if(t %in% class_spec$project_code) { ## if cancer type is in class then run this 
        class_type_spec <- class_spec %>% filter(project_code == t) %>% column_to_rownames(var = "V1") %>% rownames() ## get cancer type specific tumor classes
      if(length(class_type_spec) == 1) { ## if you only have 1 sample_id in the class
        top_mut_gene <- gene_mutation %>% rownames_to_column(var = "gene") %>% .[.$gene %in% gene_in_path, names(.) %in% c("gene",  class_type_spec)] %>% remove_rownames() %>% column_to_rownames(var = "gene") %>% 'colnames<-' ("obs_count") %>% 
          mutate(n_sample = length(class_type_spec),
               perc_mut = round((obs_count / length(class_type_spec)) * 100, digits = 1)) %>% 
        rownames_to_column(var = "gene") %>% arrange(desc(obs_count))
      }
      if(length(class_type_spec) > 1) { ## if you have more than 1 sample_id in the class
        top_mut_gene <- gene_mutation %>% .[rownames(.) %in% gene_in_path, names(.) %in% class_type_spec] %>% 
        rowSums() %>% as.data.frame() %>% 'colnames<-' ("obs_count") %>%
        mutate(n_sample = length(class_type_spec),
               perc_mut = round((obs_count / length(class_type_spec)) * 100, digits = 1)) %>% 
        rownames_to_column(var = "gene") %>% arrange(desc(obs_count))
      }
      if(show_perc == TRUE & show_n == FALSE) {
         top_gene_in_path[[t]] <- paste0(top_mut_gene$gene[mutated_gene_position], " (", top_mut_gene$perc_mut[mutated_gene_position],"%)")  
      }
      if(show_perc == TRUE & show_n == TRUE) {
         top_gene_in_path[[t]] <- paste0(top_mut_gene$gene[mutated_gene_position], " (", top_mut_gene$perc_mut[mutated_gene_position],"%)", "    n = ", length(class_type_spec))  
      }
      if(show_perc == FALSE) {
         top_gene_in_path[[t]] <- paste0(top_mut_gene$gene[mutated_gene_position])  
      }
      }
      if(!(t %in% class_spec$project_code)){ ## if cancer type is not in class then insert NA
        top_gene_in_path[[t]] <- "NA" 
      }
    }
    top_gene_in_path <- top_gene_in_path %>% do.call(rbind, .) %>% as.data.frame() %>% 'colnames<-' (n)
    clust_row <- data.frame(Class = d, row.names = n) %>% t() ## creating a row for cluster membership
    top_genes_in_path_cancer_type[[n]] <- rbind(clust_row, top_gene_in_path)
  }
  cluster_spec_enrich_table[[n]] <- top_genes_in_path_cancer_type %>% do.call(cbind, .) %>% as.data.frame() %>% 
  t() %>% as.data.frame() %>% rownames_to_column(var = "Pathway")
  }

cluster_spec_enrich_table1 <- cluster_spec_enrich_table %>% do.call(rbind, .) %>% as.data.frame()
save(cluster_spec_enrich_table1, file = "cluster_spec_enrich_table1.Rda")
rm(list = ls(all.names = T))
gc()
```

```{r table-1}
load("../data/cluster_spec_enrich_table1.Rda")
cluster_spec_enrich_table1 %>% kable(caption = "Table 1: Most mutated gene in class-specific pathways (enrichment score >= 0.30) for specific cancer", row.names = FALSE) %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"), full_width = FALSE, font_size = 12) %>%
  column_spec(1, bold = T)
```
<p style="FONT-SIZE:10px; COLOR:#000000; LINE-HEIGHT:12px;">**Table 1:** The numbers in this table are completely wrong, it is just a place holder, please ignore</p>

### Enrichment of pathways in metastasis is class-specific

Following the same logic we used to investigate cluster-specific enrichment of
pathways (**figure 3**) we applied this method to metastatic vs non-metastatic
tumors, as we find that metastatic tumors are distributed across all 12 clusters
( _e.g._ **figure 1l**). Using all non-metastatic tumors as background, we found
very low levels of enrichment, less than $10\%$, in a handful of pathways. We
reasoned that the individual clusters might be too different to detect global
metastasis enrichment signal given the small sample size (n = 215 metastatic
tumor samples).

Therefore, we calculated cluster-specific enrichment in metastatic tumors and
found 99 enriched pathways across all clusters (**table 2**). A number of
enrichments were found in multiple clusters represented by pathways that were
already shown to be enriched in a neighboring cluster. For example,
“Beta-catenin independent WNT signaling” is enriched in non-metastatic samples
of clusters 6, 7, 11 and 12 (see **supplementary figure S4**) but not in 9 and
10. This pathway is enriched in metastatic tumors of clusters 9 and 10, plus the
union of 9+10 (p < 10-3, **table 2**). This is also true of “Erythropoietin
activates RAS”, which is enriched in clusters 11 and 12 (**supplementary figure
S4**) and also in metastatic tumors of clusters 9 and 10, plus cluster 7.
Cluster 1, one of two clusters for which we could not identify any
distinguishing pathways (cluster 8 being the other), had enrichment in two
pathways that were otherwise specific to non-metastatic tumors of other
clusters. These include “Cellular response to hypoxia” (enriched in clusters 6 &
7) and “TNFR2 non-canonical NF-kB pathway” (cluster 6 & 7). Cluster 5 metastatic
samples were enriched for two key neutrophil pathways, “Neutrophil
Degranulation” and “Fc epsilon receptor (FCERI) signaling” which are specific to
cluster 9 & 10 non-metastatic tumors. Thus, pathway lesions specific to
metastases tend toward complementary mutations that are otherwise specific to
non-metastatic tumors of other clusters.
<div style="text-align: right"> Table 2 Data </div>
```{r table-2-data-generation, eval = FALSE}
tumor_clusters <- fread("../data/umap_2d_3d_coors.tsv") ## load in data
load("../data/gdc_reactome_path_analysis.Rda")

tumor_clusters <- tumor_clusters %>% dplyr::select(V1, ajcc_metastasis_pathologic_pm, knn_clust)

m0_clust <- tumor_clusters[tumor_clusters$ajcc_metastasis_pathologic_pm == "M0",] ## filter for nonmetastatic samples
m1_clust <- tumor_clusters[tumor_clusters$ajcc_metastasis_pathologic_pm == "M1",] ## filter for metastatic samples

prepare_rbeta_data_p1 <- function(data, path_analysis) {
  n_tumor_in_group <- as.data.frame(table(data$ajcc_metastasis_pathologic_pm)) ## Getting the total number of tumor samples in each group
  path_analysis <- path_analysis[,names(path_analysis) %in% data$V1] ## Filter only group specific cancers
  data <- data[match(names(path_analysis), data$V1),] ## Match tumor sample ids
  table(names(path_analysis) == data$V1) ## good they match now
  ## Now we are going to build group specific cluster data
  n_tumor_in_clust <- as.data.frame(table(data$knn_clust)) ## getting total n in each cluster
  names(n_tumor_in_clust) <- c("clust", "tot_n_clust") ## rename colnames
  clust_group <- as.character(n_tumor_in_clust$clust) ## Getting the cluster numbers
  cluster_path_count <- list() ## Creating a list for the forloop
  for(n in clust_group) {
    print(n)
    df <- subset(data, knn_clust == n) ## subset cluster number
    df <- df$V1 ## getting tumor sample id
    df1 <- path_analysis[,names(path_analysis) %in% df] ## getting the tumor samples from path analysis
    df1 <- as.data.frame(rowSums(df1))
    names(df1) <- n
    cluster_path_count[[n]] <- df1
  }
  cluster_path_count <- do.call(cbind, cluster_path_count) ## Combine the list together
  rbeta_path_df <- cluster_path_count %>% rownames_to_column(var="path") 
  rbeta_path_df <- rbeta_path_df %>% gather(key = "clust", value = "mutation", -path) ## Wide to long format
  global_path_mut <- as.data.frame(rowSums(cluster_path_count))
  global_path_mut <- global_path_mut[rep(1:ncol(global_path_mut), each = max(as.numeric(as.character(rbeta_path_df$clust))))] ## Repeat column 12 times
  names(global_path_mut) <- as.character(clust_group) ## Replace names with cluster numbers
  global_path_mut <- global_path_mut %>% rownames_to_column(var="path") %>% gather( key = "clust", value = "global", -path) ## Wide to long format
  rbeta_path_df <- left_join(rbeta_path_df, global_path_mut, by = c("path", "clust")) ## left_join rbetadf and global
  rbeta_path_df <- left_join(rbeta_path_df, n_tumor_in_clust, by = "clust")
  rbeta_path_df$global_sample <- as.numeric(sum(n_tumor_in_clust$tot_n_clust)) ## Getting total tumor sample number
  names(rbeta_path_df) <- c("path", "clust", "clust_mut", "global_mut", "tot_n_clust", "global_sample")
  rbeta_path_df$tot_n_clust <- as.numeric(as.character(rbeta_path_df$tot_n_clust)) ## convert tot_n_clust numeric
  return(rbeta_path_df)
}

m0_rbeta <- prepare_rbeta_data_p1(data = m0_clust, path_analysis = gdc_reactome_path_analysis)
## Combining cluster 6/7 cluster 910 cluster 1112
combo_group <- c("6", "9", "11")
compo_group_df <- list() ## list for loop
for(n in combo_group) {
  print(n) ## first cluster in set
  m <- as.numeric(n) + 1 ## second cluster in set
  m <- as.character(m)
  df <- m0_rbeta[m0_rbeta$clust == n,] ## get rbeta info for first cluster in set
  df1 <- m0_rbeta[m0_rbeta$clust == m,] ## get rbeta info for second cluster in set
  tot_clust_mut <- df$clust_mut + df1$clust_mut ## add both clust muts
  tot_n_clust <- df$tot_n_clust + df1$tot_n_clust ## add both clust n's
  p <- paste0(n,m)
  df2 <- m0_rbeta[1:377,]
  df2$clust <- p ## replace with new clust name
  df2$clust_mut <- tot_clust_mut ## replace with new clust mut count
  df2$tot_n_clust <- tot_n_clust ## replace with new clust size
  compo_group_df[[p]] <- df2 ## store in list and continue loop
}
compo_group_df <- do.call(rbind, compo_group_df) ## rbind list
m0_rbeta <- rbind(m0_rbeta, compo_group_df) ## rbind m0_rbeta and the compo group together
save(m0_rbeta, file = "m0_rbeta.Rda")

## repeat same function above with metastasis sample
m1_rbeta <- prepare_rbeta_data_p1(data = m1_clust, path_analysis = gdc_reactome_path_analysis)
compo_group_df <- list() ## list for loop
for(n in combo_group) {
  print(n) ## first cluster in set
  m <- as.numeric(n) + 1 ## second cluster in set
  m <- as.character(m)
  df <- m0_rbeta[m0_rbeta$clust == n,] ## get rbeta info for first cluster in set
  df1 <- m0_rbeta[m0_rbeta$clust == m,] ## get rbeta info for second cluster in set
  tot_clust_mut <- df$clust_mut + df1$clust_mut ## add both clust muts
  tot_n_clust <- df$tot_n_clust + df1$tot_n_clust ## add both clust n's
  p <- paste0(n,m)
  df2 <- m0_rbeta[1:377,]
  df2$clust <- p ## replace with new clust name
  df2$clust_mut <- tot_clust_mut ## replace with new clust mut count
  df2$tot_n_clust <- tot_n_clust ## replace with new clust size
  compo_group_df[[p]] <- df2 ## store in list and continue loop
}
compo_group_df <- do.call(rbind, compo_group_df) ## rbind list
m1_rbeta <- rbind(m1_rbeta, compo_group_df) ## Combine the new groups
save(m1_rbeta, file = "m1_rbeta.Rda")

## function to compiling data for rbeta analysis
prepare_rbeta_data_p2 <- function(data) {
  enrich_prob <- data.frame(path = data$path,
                               clust = data$clust,
                               fg_mut = data$clust_mut,
                               fg_not_mut = data$tot_n_clust - data$clust_mut,
                               bg_mut = data$global_mut,
                               bg_not_mut = data$global_sample - data$global_mut)
fg_mut_rate <- list()
bg_mut_rate <- list()
for(n in unique(as.character(enrich_prob$clust))) {
  print(n)
  df <- subset(enrich_prob, clust == n)
  fg_denominator <- nrow(df) * (df$fg_mut[1] + df$fg_not_mut[1]) ## n_path * tot_n_clust_x
  fg_numerator <- sum(df$fg_mut)
  bg_denominator <- nrow(df) * (df$bg_mut[1] + df$bg_not_mut[1])
  bg_numerator <- sum(df$bg_mut)
  fg_mut_rate[[n]] <- fg_numerator / fg_denominator
  bg_mut_rate[[n]] <- bg_numerator / bg_denominator
}

fg_mut_rate <- as.data.frame(do.call(rbind, fg_mut_rate)) ## rbind the mut rates from list
bg_mut_rate <- as.data.frame(do.call(rbind, bg_mut_rate)) ## rbind the mut rates from list
enrich_prob_df <- enrich_prob
enrich_prob_df$fg_mut_rate <- rep(x = fg_mut_rate$V1, each = length(unique(enrich_prob_df$path)))
enrich_prob_df$bg_mut_rate <- rep(x = bg_mut_rate$V1, each = length(unique(enrich_prob_df$path)))
enrich_prob_df$rel_odds <- enrich_prob_df$fg_mut_rate / enrich_prob_df$bg_mut_rate
enrich_prob_df$denominator <- (enrich_prob_df$bg_mut * enrich_prob_df$rel_odds) +   (enrich_prob_df$bg_not_mut / enrich_prob_df$rel_odds)
enrich_prob_df$x_adj <- round((enrich_prob_df$fg_mut + enrich_prob_df$fg_not_mut) * (enrich_prob_df$bg_mut * enrich_prob_df$rel_odds) / enrich_prob_df$denominator)
enrich_prob_df$y_adj <- (enrich_prob_df$fg_mut + enrich_prob_df$fg_not_mut) -   enrich_prob_df$x_adj
return(enrich_prob_df)
}

m0_rbeta <- prepare_rbeta_data_p2(m0_rbeta)                               
m1_rbeta <- prepare_rbeta_data_p2(m1_rbeta)     

save(m0_rbeta, m1_rbeta, file = "m0_m1_rbeta.Rda")
rm(list = ls(all.names = T)) ## cleaning up data
gc()
load("../data/m0_m1_rbeta.Rda")
## Metastatic (foreground) - Nonmetastatic (background) Rbeta
moderate_fg <- apply(m1_rbeta[,c("fg_mut", "fg_not_mut")], MARGIN = 1, function(x) {
  rbeta(n = 1e5, shape1=x["fg_mut"], shape2=x["fg_not_mut"])
})
moderate_bg <- apply(m0_rbeta[,c("fg_mut", "fg_not_mut")], MARGIN = 1, function(x) {
  rbeta(n = 1e5, shape1=x["fg_mut"], shape2=x["fg_not_mut"])
})

metastasis_cluster_moderate <- moderate_fg - moderate_bg  ## Fg - Bg ... to see how many times the Fg is larger than the Bg
metastasis_cluster_moderate <- as.data.frame(metastasis_cluster_moderate)

## ModerateEnrichment probability 
prob <- moderate_fg > moderate_bg ## boolean data
mean_prob <- apply(prob, MARGIN = 2, function(x) {
  mean(x)
})
## Moderate Difference
CI <- 0.025

diff_fg_bg <- moderate_fg - moderate_bg ## fg - bg
rm(moderate_fg, moderate_bg, prob)
gc()
quant_diff <- t(apply(diff_fg_bg, MARGIN = 2, function(x) {
  quantile(x, c(CI, 0.5, 1-CI)) ## 95% CI
}))

sig_fg_bg <- as.data.frame(apply(diff_fg_bg, MARGIN = 2, function(x) {
  sum(x < 0 ) ## out of 1e5 simulations, how many times is fg-bg less than 0
}))

m1_minus_m0_per_cluster_rbeta <- data.frame(path = m1_rbeta$path,
                                   clust = m1_rbeta$clust,
                                   probability = mean_prob,
                                   signif = sig_fg_bg[,1],
                                   difference = quant_diff[,2],
                                   lower = quant_diff[,1],
                                   upper = quant_diff[,3])
save(m1_minus_m0_per_cluster_rbeta, file = "m1_minus_m0_per_cluster_rbeta.Rda")
rm(list = ls(all.names = T))
gc()
```

```{r table-2}
load("../data/m1_minus_m0_per_cluster_rbeta.Rda")

table2 <- m1_minus_m0_per_cluster_rbeta %>% filter(difference >= 0.20 & signif <= 2500) %>% ## filter effect size >= 0.20 and enrichment at a 95% CI
  dplyr::select(path, clust, signif, difference) %>% 'colnames<-' (c("Pathway", "Cluster", "Signif", "Enrichment_Score")) ## select relevant column and rename col

table2$log10pvalue <- round(-log10((table2$Signif + 1) / 1E5), digits = 2) ## Getting p value ## adding 1 for pseudo count.. 1E5 is the n simulations performed
table2$Enrichment_Score <- round(table2$Enrichment_Score, digits = 2)
table2$Signif <- NULL ## we don't need this anymore

table2 %>% knitr::kable(format = "html") %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"), full_width = FALSE, font_size = 12) %>%
  column_spec(1, bold = T, color = "black")
```

There were also a number of pathways that were enriched with lower specificity
(among 4 or more clusters) which were found to be enriched in metastases. We
identified 18 such pathways of which 12 had significantly higher mutation rates
in cluster-specific metastases, including “G1/S DNA Damage Checkpoints”,
“Deubiquitination”, “Sumoylation”, “M Phase”, “RHO GTPase Effectors”, “TCF
dependent signaling in response to WNT”, “Class I MHC mediated antigen
processing & presentation”, “Chaperonin-mediated protein folding”, “Intrinsic
Pathway for Apoptosis”, “Cellular Senescence”, “DNA Double-Strand Break
Response”, and “Pre-NOTCH Expression and Processing” (**Table 2**).

### Pathway disruption classes vary in short-term prognosis of survival

If our molecular pathway disruption classes represent distinct biological states
apart from tissue of origin, they may have different prognoses within cancer
types or across all cancers. Kaplan-Meier plots reveal large differences between
cancer types as expected, and some modest differences between our classes (not
shown). These analyses are profoundly limited by confounding factors of age and
stage at diagnosis, sex, ethnicity, and tissue-specific disease progression, to
name the obvious ones. To explore these ideas, we tested models of survival
using public longevity data from the CDC, accounting for age, gender and
ethnicity. To build on this relatively accurate model of baseline risk (see
methods), we assume that individual cancer type captures some variation not
accounted for in the longevity data ( _e.g._ all tissue-specific factors,
including stage [see methods]), thus allowing us to avoid introducing
uncertainty by modeling each of those factors explicitly. Therefore, our model
estimates the effect of cancer type and class-specific cancer effects
independently. We found that cancer types, as expected, have a range of
prognoses relative to the general population, as shown in **figure 5a**. The
least deadly was prostate cancer (PRAD), and the most deadly was glioblastoma
(GBM) which has a risk multiplier of between 15 and 22 relative to the
background risk of death in the population. Among the deadliest cancers outside
of GBM were stomach (STAD), melanoma (SKCM) and pancreatic (PAAD) cancers. Apart
from these top 4 cancers which ranged from 6-22 in relative risk of age-adjusted
death, the remaining cancer types ranged from about 1 to 5 in magnitude
(**figure 5a**).

![](../figures/fig6B.png)
![](../figures/fig6C.png)
<p style="FONT-SIZE:10px; COLOR:#000000; LINE-HEIGHT:12px;">**Figure 5: relative survival of pathway clusters. a) Posterior distributions of estimated relative age-dependent survival by cancer type. b) Posterior distributions of estimated class-dependent survival for each cancer type. See methods for model details.</p>

Our estimates of tissue-specific class effects, in contrast, ranged from 0-3,
reflecting that some classes are either less deadly or more deadly than other
classes within each cancer types, where the null hypothesis corresponds to an
effect size=1 (**figure 5b**). For several cancers ( _e.g._ PRAD, kidney
chromophobe (KICH), diffuse large B-cell lymphoma (DBLC), thyroid cancer (THCA))
the posterior estimates are largely indestinguishable from the prior, reflecting
that either there were too few mortalities in the data to make an estimate (as
expected for PRAD and THCA) or two few samples period. We did not observe
class-specific trends that held true across cancer types, which could result
from different cancers having different standards of care for example. In
support of this interpretation, we also tested a factored model which considered
both cancer type and each class independent of cancer type (**supplementary
figure S7??**). Though we were able to successfully fit these models, they
exhibited much greater uncertainty, suggesting an inferior specification.
(<mark>Dan are there additional arguments to be made for the inferiority of the
factored model, here?</mark>)

```{r survival-model, eval=FALSE}

```

![**Figure 6B**](../figures/fig6B.png)

![**Figure 6C**](../figures/fig6C.png)

```{r umap-global-trends, fig.cap="Placeholder for figure 1", results=FALSE, eval=FALSE}
set.seed(21)
plot(x=rnorm(1000), y=rnorm(1000))
```


## Materials & Methods

All code for producing the analyses and figures herein are included in this
fully reproducible manuscript. In addition, the R markdown files and all other
models are available from repositories in the following locations on the
distributed version software hub GitHub.

### Selection of pathways

Pathway analysis provides additional information that genes and gene ontologies
alone lack: 1) it aggregates molecular events across multiple genes in the same
pathway and 2) the results, interpretable as genetic alterations, are related 
to molecular functions such as “intrinsic pathway for apoptosis” or “cellular 
response to hypoxia”. Using the Reactome Pathway Database, we selected the 
second sub-pathways for each pathway (e.g. ‘Beta-catenin independent WNT 
signaling’ in ‘Signal Transduction’) which we believed was a suitable 
representation of the basic cellular processes and biochemical pathways. 
We also excluded pathways categorized under cancer, disease, gene defect, 
disorder, infection, reproduction, and muscle contraction (e.g. ‘PI3K/AKT 
Signaling in Cancer’). While some of the excluded pathways have been shown
to play an important role in the cancer, they focus on a specific point 
mutations. In addition, for most of the excluded pathways, there is a global
pathway (e.g. ‘PIP3 activates Akt signaling’) that exist. Finally, we mapped
the 18,577 Ensembl IDs from the GDC dataset to the highly selected pathways.
This operation produced a lookup table that consisted of 377 pathways mapped 
to 9,910 genes.  

### <mark>Filtering genes <- Peter to write</mark>

### Class and stage-specific Enrichment Calculations

We calculated the enrichment of pathways in one set of tumors as the relative
fraction (and probability of observing) of tumors with a mutated gene in the
pathway to all tumors, inclusive of the category of interest. To do this
operation, we use the beta distribution to permute a posterior distribution on
the fraction of tumors with a pathway mutated for each category based on the
observed set, and compared this to the posterior obtained from the full set of
tumors (all tumors) as the distribution of differences between all permuted
samples. We considered a pathway enriched if the $95\%$ range of credible
differences thus obtained excludes 0 (as a null hypothesis) and the mean
credible difference was greater than or equal to $20\%$ enrichment, which
excludes a large number of small differences that are not likely to be
biologically relevant.

### Models of survival based on population-specific longevity data

We modeled baseline longevity in the U.S. with these different factors.
<mark>Key assumptions? (Room here for further descriptions and equations)</mark>
Using this base survival model, we constructed a likelihood function with
parameters describing the overall lethality of each cancer type as a modifier
$k_{tissue}$ of the rate parameter $r_{0}$, and cluster specific modifiers of
$k$ and optimized model fit using stan[@stanref]. The $k$ in our model can be
taken as, roughly, the death rate relative to the baseline rates as depicted in
**supplementary figure S5**. Thus, a value of $k=2$ means that a given cancer is
twice as deadly.

With an accurate model for baseline longevity in place, we sought to model
cancer as an event based modifier of long-term survival. Thus any factors or
events such as cancer would modify the baseline risk function as a multiplier.
The resulting coefficients can thus be interpreted as increasing baseline risk
by <mark>x-fold.</mark> Baseline risk was modeled as cumulative survival for
female vs. male caucasians, blacks and asians (hispanics are not clearly marked
in the tumor data; see **supplementary figure S5** for comparisons of raw CDC
data and baseline risk models). Thus we are able to accurately model baseline
risk of death taking these factors into account. The data themselves encompass
all other causes of death for these populations including cancer, thus freeing
our model from concerns about other confounding factors ( _e.g._ Diabetes). For
demonstration purposes we also illustrate some examples of the expected survival
for a population that receives a diagnosis with doubled risk of death at ages
40, 50, 60, 70, and 80 (**supplementary figure S6**).

As an aside, we did not explicitly account for stage at diagnosis in our model
for the following reasons. In the tumor sample data, stage at diagnosis is
confounded with cancer type because some cancers are screened agressively (
_e.g._ colorectal and prostate cancers) while others are diagnosed typically
after they become problematic for the patient's lifestyle ( _e.g._ ovarian and
pancreatic cancers). Secondly, such a model specification would suffer from
added noise because the staging data are not well standardized across cancer
types, have different criteria, and because it is unclear what the relationship
between stage and advancement of disease is (for example some sub-stage 4 tumors
are metastatic). To compound this latter issue, our tumor set is vastly
under-powered given the uneven representation of stage across cancer types.

## Discussion
 
```{r child='discussion.Rmd'}
```

## Supplementary Information
```{r supplementary-figure-s1, fig.width= 8.5, fig.height= 11}
umap_cancertype <- fread("../data/umap_2d_3d_coors.tsv") ## load data

umap_bg <- umap_cancertype[,-"project_code"] ## get background data for facet
  
ggplot(umap_cancertype, aes(x = x_2d, y = y_2d, colour = project_code)) + 
  geom_point(data = umap_bg, aes(x = x_2d, y = y_2d), colour = "grey", size = 1) +
  geom_point(size = 1) +
  theme_classic() +
  coord_fixed() +
  xlab("UMAP 1") + ylab("UMAP 2") +
  scale_colour_manual(values = rep("firebrick3", 23)) +
  theme(axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none") +
  facet_wrap(~ project_code, ncol = 5) ## facet by cancer_type
```
![**Supplementary figure S1. Distribution of cancer type on UMAP clusters.** Each panel shows the identity of individual tumors by one of 23 cancer types projected onto the cluster UMAP identified in **Figure 1 A** of the main text._]

```{r supplementary-figure-s2-data-generation}
umap_cancertype <- fread("../data/umap_2d_3d_coors.tsv") ## load data
umap_bg <- umap_cancertype[,-"project_code"] ## get background data for facet

ggplot(umap_cancertype, aes(x = x_2d, y = y_2d, colour = Subtype_Selected)) + 
  geom_point(data = umap_bg, aes(x = x_2d, y = y_2d), colour = "grey", size = 0.5) +
  geom_point(size = 0.5) +
  theme_classic() +
  coord_fixed() +
  xlab("UMAP 1") + ylab("UMAP 2") +
  theme(axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none") +
  facet_wrap(~ project_code, ncol = 5) ## facet by cancer type
```

![**Supplementary Figure S2. Histological subtypes.** _Each panel shows the histological subtypes of one of the 23 cancers projected on to the tumors of the UMAP from **Figure 1A of the main text.**_]
<div style="text-align: right"> Supplementary Figure S3 Data </div>
```{r supplementary-figure-s3-data-generation, eval = FALSE}
load("../data/gdc_tumor_coding_complete.Rdata")
load("../data/gene_data.Rda")

## Booleanize mutation count for each tumor sample
missense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$missense) 
splice <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$splice)
frameshdel <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshdel)
frameshins <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$frameshins)
utr3 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr3)
utr5 <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$utr5)
nonstop <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonstop)
nonsense <- as.data.frame(SummarizedExperiment::assays(gdc_tumor_coding_complete)$nonsense)

gdc_mut_count_patient <- missense + splice + frameshdel + frameshins + utr3 + utr5 + nonstop + nonsense ## add up all the mutation count
rm(missense, splice, frameshdel, frameshins, utr3, utr5, nonstop, nonsense)
gc()
gdc_mut_count_patient[gdc_mut_count_patient > 0] <- 1 ## booleanize data, anything greater than 1 is equal to 1

##oncogene
oncogene <- c("AKT2","BCL2","CTNNB1","ERBB2","KRAS","MDM2","MYC","PIK3CA","TET2","USP6") ## list of selected oncogene
##tumor_supressor
tumor_suppressor <- c("APC","BRIP1","CDKN2C","CHEK2","MSH2","PTEN","RB1","SMARCA4","TP53","VHL") ## list of selected tumor suppressor

selected_mut <- gene_data %>% dplyr::select(external_gene_name, ensembl_gene_id) %>% filter(external_gene_name %in% c(oncogene, tumor_suppressor)) ## getting ensembl_id for the oncogenes for filtering gdc data

selected_mut <- gdc_mut_count_patient %>% filter(rownames(.) %in% selected_mut$ensembl_gene_id) %>% ## filter by ensembl ids
  rownames_to_column(var = "ensembl_gene_id") %>% left_join(., selected_mut, by = "ensembl_gene_id") %>% ## replace ensembl ids with gene name 
  column_to_rownames(var = "external_gene_name") %>% dplyr::select(-ensembl_gene_id) %>%
  t() %>% as.data.frame() %>% rownames_to_column(var = "V1")  ## prep for left_join


selected_mut <- umap_cancertype %>% dplyr::select(V1, x_2d, y_2d) %>% ## select relevant columns
  left_join(., selected_mut, by = "V1") %>%
  melt(data = ., id.vars = c("V1", "x_2d", "y_2d")) %>%
  filter(value != 0)

save(selected_mut, file = "selected_mut.Rda")
```

```{r supplementary-figure-s3}
load("../data/selected_mut.Rda")
umap_cancertype <- fread("../data/umap_2d_3d_coors.tsv") ## load data
umap_bg <- umap_cancertype[,-"project_code"] ## get background data for facet

ggplot(selected_mut, aes(x = x_2d, y = y_2d, colour = as.character(value))) + 
  geom_point(data = umap_bg, aes(x = x_2d, y = y_2d), colour = "grey", size = 0.5) +
  geom_point(size = 0.5) +
  theme_classic() +
  coord_fixed() +
  xlab("UMAP 1") + ylab("UMAP 2") +
  scale_colour_manual(values = rep("firebrick3", 20)) +
  theme(axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none") +
  facet_wrap(~ factor(variable, levels = c("AKT2","BCL2","CTNNB1","ERBB2","KRAS","MDM2","MYC","PIK3CA","TET2","USP6", 
                                           "APC","BRIP1","CDKN2C","CHEK2","MSH2","PTEN","RB1","SMARCA4","TP53","VHL")),
                      ncol = 5) ## facet by gene
```

![**Supplementary Figure S3. Common oncogenic mutations.** _Each panel shows one of 20 commonly mutated oncogenic genes projected on the UMAP from **Figure 1A of the main text.**_]

<div style="text-align: right"> Supplementary Figure S4 Data </div>
```{r supplementary-figure-s4-data-generation, eval = FALSE}
load("../data/gdc_reactome_path_analysis.Rda") ## load data
umap_cancertype <- fread("../data/umap_2d_3d_coors.tsv") 
n_path <- 377 ## how many pathways do we want to look at starting from the most to least mutated pathway

toppath <- gdc_reactome_path_analysis %>% rowSums() %>% as.data.frame() %>% rownames_to_column(var = "path_name") %>% arrange(desc(.)) %>% .[1:n_path,] %>% .[, "path_name"] ## order pathway by most to least mutated pathway

selected_path <- gdc_reactome_path_analysis %>% t() %>% as.data.frame() %>% rownames_to_column(var = "V1") ## prepare pathway for left_join

selected_path <- umap_cancertype %>% dplyr::select(V1, x_2d, y_2d) %>% ## select relevant columns
  left_join(., selected_path, by = "V1") %>%
  melt(data = ., id.vars = c("V1", "x_2d", "y_2d")) %>%
  filter(value != 0)

save(selected_path, file = "selected_path.Rda")
```

```{r supplementary-figure-s4, out.width = '90%'}
load("../data/selected_path.Rda") ## load data
load("../data/gdc_reactome_path_analysis.Rda")
umap_cancertype <- fread("../data/umap_2d_3d_coors.tsv")
umap_bg <- umap_cancertype[,-"project_code"] ## get background data for facet

n_path <- 377 ## number of pathways we want to see

toppath <- gdc_reactome_path_analysis %>% rowSums() %>% as.data.frame() %>% rownames_to_column(var = "path_name") %>% arrange(desc(.)) %>% .[1:n_path,] %>% .[, "path_name"] ## order pathway by most to least mutated pathway.. needed for facet

ggplot(selected_path, aes(x = x_2d, y = y_2d, colour = as.character(value))) + 
  geom_point(data = umap_bg, aes(x = x_2d, y = y_2d), colour = "grey", size = 0.5) +
  geom_point(size = 0.5) +
  theme_classic() +
  coord_fixed() +
  xlab("UMAP 1") + ylab("UMAP 2") +
  scale_colour_manual(values = rep("firebrick3", 20)) +
  theme(axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none") +
  facet_wrap(~ factor(variable, levels = toppath),
                      ncol = 5) ## facet by pathways

```

![**Supplementary Figure S4. Projection of pathway disruptions onto the UMAP.**_]

<div style="text-align: right"> Supplementary Figure ST1 Data </div>
```{r supplemental-table-st1-data-generation, eval = FALSE}
load("../data/gdc_tumor_coding_complete.Rdata")
load("../data/gdc_reactome_path_analysis.Rda")
load("../data/gene2_level2_lookup.Rda")
load("../data/pathname2level2.Rda")
load("../data/gene_data.Rda")
load("../data/gdc_gene_exp_adj.Rda")

## Parameters ##
n_path <- 377 ## how many top pathways do you want to see. 1 = top path, 2 = top path and second most
mutated_gene_position <- c(1,2,3) ## what gene do you want to see. 1 = top gene, 2 = second most, 3 = third
show_perc <- TRUE ## show perc for mutated gene


toppath <- gdc_reactome_path_analysis %>% rowSums() %>% as.data.frame() %>% rownames_to_column(var = "path_name") %>% arrange(desc(.)) %>% .[1:n_path,] %>% .[, "path_name"] ## getting the top mutated pathways

id_to_names <- gene2level2_lookup %>% rownames_to_column(var = "ensembl_gene_id") %>% ## convert ensembl to gene name
  left_join(., gene_data[,c("ensembl_gene_id", "external_gene_name")], by = "ensembl_gene_id") %>%
  remove_rownames() %>% column_to_rownames(var = "external_gene_name") %>%
  dplyr::select(-ensembl_gene_id) %>%
  t() %>% as.data.frame() %>% rownames_to_column(var = "PATH_ID") %>% ## convert path_id to path_name
  left_join(., pathname2level2, by = "PATH_ID") %>%
  remove_rownames() %>% column_to_rownames(var = "PATH_NAME") %>%
  dplyr::select(-PATH_ID) %>%
  t() %>% as.data.frame()

gene_mutation <- gdc_gene_exp_adj %>% rownames_to_column(var = "ensembl_gene_id") %>% ## convert ensembl to gene name
  left_join(., gene_data[,c("ensembl_gene_id", "external_gene_name")], by = "ensembl_gene_id") %>%
  remove_rownames() %>% column_to_rownames(var = "external_gene_name") %>%
  dplyr::select(-ensembl_gene_id)

table_st1 <- list() ## list for forloop 

for(n in toppath) {
  message(n)
  ## finding top mutated gene in pathway
  gene_in_path <- id_to_names %>% .[,which(names(.) == n)] %>% as.data.frame() %>% ## select pathway
    'rownames<-' (rownames(id_to_names)) %>% 
    filter(. == 1) %>% ## filter for genes that are in that pathway
    rownames(.)
  top_mut_gene <- gene_mutation %>% .[rownames(.) %in% gene_in_path, ] %>% ## only include genes in paths
      rowSums() %>% as.data.frame() %>% 'colnames<-' ("obs_count") %>% ## sum the pathway mutation
      mutate(n_sample = length(gene_mutation),
             perc_mut = round((obs_count / length(gene_mutation)) * 100, digits = 1)) %>% 
      rownames_to_column(var = "gene") %>% arrange(desc(obs_count))
  top_gene_in_path <- list()
  for(p in mutated_gene_position) {
    message(p)
    if(show_perc == TRUE) {
    top_gene_in_path[[p]] <- paste0(top_mut_gene$gene[p], " (", top_mut_gene$perc_mut[p],"%)")  
    }
    if(show_perc == FALSE) {
    top_gene_in_path[[p]] <- paste0(top_mut_gene$gene[p])  
    }
  }
  top_gene_in_path <- top_gene_in_path %>% do.call(rbind, .) %>% as.data.frame() %>% ## rbind list
   'rownames<-' (mutated_gene_position) %>% t() %>% as.data.frame()
  ## calculating the percent mutated samples
  spec_path <- data.frame(Pathway = n,
                          Percent_Mutated = round(gdc_reactome_path_analysis %>% .[n,] %>% sum() / length(gdc_reactome_path_analysis) * 100, digits = 1))
  table_st1[[n]] <- cbind(spec_path, top_gene_in_path)
}
table_st1 <- as.data.frame(do.call(rbind, table_st1))
save(table_st1, file = "table_st1.Rda")  
rm(list = ls(all.names = T))  
gc()
```

```{r supplemental-table-st1}
load("../data/table_st1.Rda")

table_st1 %>% dplyr::select(-Pathway) %>% knitr::kable(format = "html") %>%
  kable_styling(bootstrap_options = c("striped", "hover", "condensed"), full_width = FALSE, font_size = 12) %>%
  column_spec(1, bold = T, color = "black")
```

![**Supplementary Table ST1. Frequency of pathway disruption and top genes within pathways.** _A complete list of pathway disruptions with percent mutatied samples and top genes_]


![**Supplemental figure S6. Survival fit independent of disease status.** _The red and blue curves show the survival of people in the US by female and male, respectively. The orange and green curves show the model fits._](../figures/fig6A.png)


## References