---
title: "A molecular taxonomy of tumors independent of tissue-of-origin"
author: Peter T. Nguyen$^{1}$, Simon G. Coetzee$^{1}$, Daniel L. Lakeland$^{2}$, and
  Dennis J. Hazelett$^{1,3}$
bibliography: references.bib
output:
  word_document:
    reference_docx: "../text/word-style-reference-ttmanu.docx"
always_allow_html: TRUE
editor_options:
  chunk_output_type: console
---

###### Institutional affiliations: 1. The Center for Bioinformatics and Functional Genomics, Cedars-Sinai Medical Center, Los Angeles, California. 2. Lakeland Applied Sciences LLC, Los Angeles, CA. 3. Samuel Oschin Comprehensive Cancer Institute, Cedars-Sinai Medical Center, Los Angeles, CA.

```{r echo=FALSE, eval=FALSE}
  bookdown::pdf_book:
    number_sections: FALSE
    toc: FALSE
    keep_tex: yes

output:
  word_document:
    reference_docx: word-styles-reference-01.docx
  number_sections: FALSE
  toc: FALSE
  
output: 
bookdown::html_document2:
  fig_captions: TRUE
  number_sections: FALSE
  toc: TRUE
  toc_float: TRUE
  toc_depth: 3
  code_folding: hide
  theme: flatly
  smart: true
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, dpi=600, warning = FALSE)
suppressPackageStartupMessages(c(
  library(rstan),
  library(foreach),
  library(sqldf),
  library(tidyverse),
  library(ggthemes),
  library(data.table),
  library(biomaRt),
  library(dplyr),
  library(GenomicRanges),
  library(SummarizedExperiment),
  library(reactome.db),
  library(plyr),
  library(cluster),
  library(pheatmap),
  library(clusterProfiler),
  library(org.Hs.eg.db),
  library(dbscan),
  library(class),
  library(TCGAbiolinks),
  library(patchwork),
  library(ggplotify),
  library(ComplexHeatmap),
  library(knitr),
  library(kableExtra),
  library(UpSetR),
  library(cowplot),
  library(doRNG),
  library(uwot),
  library(janitor),
  library(FactoMineR),
  library(seriation),
  library(rtracklayer),
  library(poibin),
  library(flextable),
  library(plot3D),
  library(ggfortify),
  library(maftools),
  library(survival),
  library(ggalluvial),
  library(grid),
  library(gridBase)
))
```

## Abstract

```{r child='abstract.Rmd'}
```

## Introduction

```{r child='introduction.Rmd'}
```

## Results
 
### A taxonomy of tumors based on disrupted molecular pathways

To study cancer pathways we obtained a set of 7,607 solid tumor samples from The
Cancer Genome Atlas (TCGA) through the [Genomic Data Commons
(GDC)](gdc.cancer.gov) portal. TCGA data were most appropriate for our study
given the relative completeness of the patient metadata, particularly for
survival and staging. We chose to analyze somatic mutations in exome sequencing
data because the affected target gene is known unambiguously. Therefore, we
selected all missense, nonsense, frameshift, stop-loss, untranslated region, and
splicing mutations. In order to minimize bias from well-studied diseases and
processes, we selected 377 [Reactome pathways](https://reactome.org/) (see
supplemental table ST1) of interest corresponding to basic cellular processes
and biochemical pathways, excluding gene sets that correspond to miscellaneous
categories ( _e.g._ “transcription factors”) or disease associations ( _e.g._
“mutated in colon cancer”) and filtered our gene list on membership in these
pathways (total of 8,940 genes; see methods for details).

To avoid bias toward larger pathways ( _i.e._ pathways with more member genes),
we counted pathways as disrupted if one or more member genes were mutated. We do
not attempt to calculate enrichment for mutations within a pathway. Data
binarized by pathway are likely to be noisy for several reasons. First, point
mutations can be deleterious (attenuating or hypomorphic) or activating
(neomorphic or hypermorphic) in genes, and these can in turn act functionally as
either activators or repressors. For this study we assume a significant fraction
of these mutations are generically disruptive to normal pathway activity if they
are activating or deactivating or affect a gene that one might classify as a
tumor suppressor or oncogene. Second, we know that low-expressed genes and
non-expressed genes accumulate mutations at a higher rate due to transcription
coupled repair [@kandothMutationalLandscapeSignificance2013;
@kimPancancerAnalysisSomatic2018;
@Pervasive_lesion_segregation; @expression-aware-annotation]. To address this
issue, we identified genes with low expression in each type of cancer and
eliminated them for that cancer type only (see methods for details). Highly
expressed genes could also have high mutation rates owing to transcription
induced mutagenesis [@parkGenomicEvidenceElevated2012]. We felt that such
mechanisms result in cell-type-specific biases that might be biologically
meaningful for predisposition to different classes of cancer however, and
therefore chose not to exclude these genes from our analysis. After selecting
our pathways and genes, we then compiled a matrix of the pathways, assigning a
Boolean value of 1 to each pathway with one or more genes mutated and 0 for all
others (**figure 1**).

```{r, results='asis'}
cat("\\newpage")
```
#### Figure 1
```{r child='../code/figure1v2.Rmd'}
```

**Figure 1: Molecular pathway profiles of tumor samples with one or more
mutations.** Each of 377 selected Reactome pathways (columns) is classified as
disrupted if one or more genes is mutated in the tumor sample (rows) where red
and grey represents the pathway is mutated and not, respectively. Tumors are
grouped by tissue of origin using standardized abbreviations from the TCGA
project.

```{r, results='asis'}
cat("\\newpage")
```

We investigated this dataset further using multiple correspondence analysis
(MCA), and visually summarized the analysis with UMAP (**figure 2**). We used
the resulting UMAP graph coordinates to perform density based clustering with
HDBSCAN, which resulted in identification of 10 well-defined clusters capturing
about 80% of the tumor samples. To capture the remaining samples into one of
these 10 clusters we used kNN (see **Supplementary Methods** for details on all
clustering algorithms).

```{r, results='asis'}
cat("\\newpage")
```

```{r, child='../code/figure2.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```

#### Independence from tissue-of-origin.

Having defined tumors in terms of their pathway disruption profile, we sought to
understand whether different cancer types segregated into one or more
predominant classes. To our surprise, most cancer types (with some notable
exceptions) were not heavily biased in one class, and all well-represented
cancer types had tumors in every class (see **figure 3A** and full tumor
profiles in **supplementary figure S1**, see also interactive media from
**supplementary file 1 (html)**), suggesting that, in principle at least, our
pathway-disruptions identify clusters of molecular pathology largely independent
of tissue-of-origin. As an example of one type of cancer that does have a biased
pathway profile, pancreatic adenocarcinoma (PAAD) was predominantly found in
class 8 (**figure 3A** and **supplementary figure S1**). But even PAAD comprises
tumors from the nine remaining classes, meaning that patients suffering these
tumors have potentially different underlying molecular pathologies.

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 3
```{r child = '../code/figure3.Rmd'}
```

**Figure 3: Pathway-based classification is independent of tissue-of-origin features.** 
a) Projection of cancer type identity onto UMAP. Abbrevs: BRCA = Breast cancer,
COAD = Colorectal adenocarcinoma, HNSC = Head and neck squamous cell, PAAD =
Pancreatic adenocarcinoma. b) Projection of breast cancer subtypes onto the
UMAP. c) Projection of BRCA1/2 somatic mutation onto the UMAP. d) Projection of
tumor stage onto the UMAP, regardless of cancer type. e) Projection of
metastatic status onto the UMAP. Abbrevs: M0 = non-metastatic tumors, M1 =
metastatic tumors.

```{r, results='asis'}
cat("\\newpage")
```

#### Independence from molecular and histological subtype.

Many cancers have molecular or histological subtypes defined based on gene
expression or other -omics profiles or pathology lab results. These subtypes
often have different standards of care owing to different overall drug
sensitivity (or other factors). If the histological subtypes represent true
molecular phenotypes, one predicts that histological subtypes should segregate
with our pathway-based classes, therefore providing support for the classes as
proxies for molecular pathology sub-typing. To our surprise, we find a similar
result to the previous analysis of cancer types projected onto the UMAP of
pathway disruptions. To illustrate this, we projected annotations for each of
the breast cancer subtypes, composed of Triple-negative/Basal-like, Her2
positive, normal-like, and luminal A and B subtypes onto the UMAP. These are
among the most heavily studied molecular subtypes in cancer, which each have
different prognoses and standards of care. We did not observe any exclusive
segregation by pathway for these subtype annotations (**Figure 3B** and
**supplemental file 1**). We also projected histological subtype data for the
remaining cancers (see **supplementary figure S2** for the full set of
projections); we find that the subtypes, though often biased towards one or more
classes, are almost never exclusive. We interpret these data in aggregate to
mean that our pathway disruption classes do not correspond to previously
identified molecular subtypes within the parent cancer type.

#### Independence from drivers of genome instability.

There are several well-known familial cancer-causing mutations that have been
interrogated extensively for differences in basic biology, survival and
treatment outcomes. However, the functions of these genes are related to risk
factors such as genome stability generally, proof-reading and DNA damage repair,
and telomere length. _BRCA1/2_ genes for example are key for DNA double-stranded
break repair [@moynihan-1999-brca1;@davies-2001-brca2] and germline mutations in
these genes confer elevated risk for breast, prostate and ovarian cancers. The
mechanism of risk is thought to involve loss of heterozygosity resulting in loss
of the wildtype, functional allele [@brca-loh], so we projected the somatic
mutations for _BRCA1_ and _BRCA2_ genes onto the UMAP, but did not observe
segregation of these mutations into specific clusters (**figure 3C**). We made
similar projections for the mismatch repair (MMR) genes _MSH2_, _MSH6_, _MLH1_,
_MLH3_, _PMS1_ and _PMS2_; _BRIP1_, _RAD51_, _CHEK2_ and _APC_. None of these
genes except for _APC_ exhibited any remarkable specificity with respect to
cluster assignment (**supplementary figure S3**). To look at other risk factors
such as maintenance of DNA methylation levels and telomere length, we projected
somatic mutations of the _TET2_ and _TET3_ genes, plus _TERT_, _TEP1_, and
_DKC1_, and observed similar lack of segregation by cluster (**supplementary
figure S3**).

#### Independence of stage, mutation count and mutation profile.

Tumor staging is based on physico-pathological criteria, including tumor
diameter, which can vary greatly in importance between different tissues. Stage
is used clinically as a proxy for advancement toward a more deadly state and
metastasis. Given these assumptions, it is possible that more advanced tumors
have common pathway disruption profiles, and the UMAP which features a series of
lobe-like structures on a common backbone of tumor samples (**figure 2C**) could
in principle also reflect progression through a series of stages. In support of
this hypothesis, the backbone starts with a cluster of tumors (class 1) that has
the fewest point mutations and culminates in a cluster (class 10) which has
nearly every pathway disrupted (**figure 4**). However, outside of class 10 we
don't observe an obvious trend in the overall mutation burden across the
backbone of the UMAP (**figure 1H**). Nonetheless, to test the hypothesis that
the molecular-pathway disruption clusters represent advancement through stages,
we projected staging data onto the UMAP. Similar to tissue of origin and other
categories of tumor, we did not observe any bias among the stages to specific
pathway disruption classes (**figure 3D**), suggesting that stage is not an
determining factor in pathway disruption classes.

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 4
```{r child = '../code/figure4.Rmd'}
```

**Figure 4: Somatic mutation frequencies for tumors in each class** 
Column uses a logarithmic axis to represent mutation count. Row represents the
tumor samples. Each dot corresponds to the sample mutation count for tumors in
each class, ordered by mutation counts. The median mutation count in each
cluster is indicated by the horizontal line.

```{r, results='asis'}
cat("\\newpage")
```

Finally, as a measure of tumor advancement, metastasis is the condition in which
certain phenotypic criteria are met. These phenotypes include (among others)
loss of differentiation, cell-cell contacts, epithelial to mesenchyme
transition, immune system evasion and tissue invasiveness [@hallmarks_II]. To
determine whether any of our clusters correspond to an especially advanced stage
of cancer across tissue types, we projected the metastasis data onto our UMAP,
and surprisingly we observed an even distribution of metastases across classes
(**figure 3E**). This final observation suggests that our pathway-disruption
classification is dependent on particular combinations of gene mutations
affecting different pathways that can each give rise to advanced stages of
disease and metastasis, regardless and independent of overall mutational burden
(see Discussion for further analysis).

### Tissue specific genes define cluster membership

In order to visualize pathway disruption enrichment across all cancers
(pan-cancer), we created waterfall plots, replacing genes with pathways
(**figure 5A**). A complete list of pathway disruptions with percent mutated
samples and top genes is provided in **supplementary table ST1**. As expected,
these analyses reveal the broad importance of many well known pathways that are
disrupted in cancer, including "PIP3 activates Akt signaling" (77% of samples),
"MAP1K/MAP3K signaling" (70% of samples), "Mitotic G2-G2/M phases" (67% of
samples), "Cellular senescence" (64% of samples), "G2/M Checkpoints" (62% of
samples), _etc._

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 5
```{r child = '../code/figure5.Rmd'}
```

**Figure 5: Pan-cancer enrichment of pathway disruptions.** 
a) Waterfall plot; columns represent tumors. Each row represents a pathway. The
pathways are ranked such that the pathway with the most disruptions across all
7,607 tumors is the first row _etc._ and tumors are ordered to prioritize the
top pathway disruptions successively. The top 100 most mutated pathways are
shown. Red signifies pathway disrupted pathways, grey signifies not disrupted.
b) Heatmap shows the relative enrichment of each pathway (columns) within each
numbered cluster (rows). Effect size is displayed as colors (orange, enrichment;
purple, depletion). c) Proportion of mutated samples in the each significant
pathway (columns; union set of pathways with effect size >= 0.30 in each
cluster) within each numbered cluster (rows).

```{r, results='asis'}
cat("\\newpage")
```

To discover what pathways are most important for clustering, we calculated
percent enrichment for each pathway _within cluster_ relative to all other
clusters combined (see methods) and ranked pathways for each cluster from
highest to lowest enrichment. We visualized the enrichment as a heatmap
(**figure 5B**). Using this approach, we identified 25 pathways highly enriched
(enrichment score $> 0.3$, see methods) in cluster 2, 9 pathways enriched in
cluster 3, 24 pathways in cluster 4, 10 pathways in cluster 5, 10 pathways in
cluster 6, 24 pathways in cluster 7, 24 pathways in cluster 8, 10 pathways in
cluster 9, and 0 pathways for clusters 1 and 10. Clusters 7, 8, and 9 had
several pathways in common. To explore the specific pathways marking each
cluster, we projected disruptions for each of the 377 pathways onto the UMAP
(**supplementary figure S4**). Clusters 3 and 5 were distinguished by metabolic
pathways including RNA and protein biosynthesis (**supplementary figure S4**).
Similarly, cluster 4 was distinguished by mutations affecting regulation of DNA
and histone methylation ("DNA methylation", "PRC2 methylates histones and DNA",
and "Nucleosome assembly"). Clusters 7-9 have in common mutations in
extracellular, intracellular, and immune-related signaling pathways (see
**figure 5C** and **supplementary figure S4**). Cluster 2 had the highest
pathway enrichment levels of the three, having mutations in hedgehog signaling,
"β-catenin degradation", "cellular response to hypoxia", "regulation of cell
cycle" and "apoptosis" among others.

Prior efforts to extract common signatures from pan-cancer datasets met with
difficulty in distinguishing tumor samples from their tissue-specific -omics
data signatures. Given our pathway-disruption based clustering, this raises
the question, are tumor phenotypes driven entirely by common driver genes, or by
“silent” tissue-specific effectors ( _i.e._ too few samples to detect above
statistical significance thresholds), or a combination of both? To answer this
question, we compared top pathway genes for each cluster relative to the TCGA
background to find differentially mutated genes. We ranked odds ratios and
selected the top ten enriched and depleted genes (pvalue < 0.01) for each
cluster (**Figure 6**; odds ratios plot). Clusters 7 and 8, which shared
multiple enrichment in signaling pathways, are largely driven by mutations in
_PI3K_ and its orthologs and _Ras_ genes, respectively (compare _PIK3CA_ and
_KRAS_ panels of **Supplementary figure S5**). Interestingly, cluster 9, which
also shared multiple enrichment in signaling pathways with clusters 7 and 8, is
enriched for both _PIK3CA_ and _KRAS_. This result suggests that tumors in this
cluster could be derivative of either cluster 7 or 8. Clusters 3 and 5, defined
by enrichment in metabolic pathways, had mutations in ribosomal proteins and
nuclear pore complex, respectively. Cluster 4 had mutations in genes responsible
for nucleosome structure. Cluster 2 had mutations in proteasomal subunit genes
involved in protein degradation. We also observed that genes that were enriched
for one cluster are depleted from others ( _e.g. TP53_ is enriched in cluster
6, but depleted in cluster 7; _PIK3CA_ is enriched in cluster 7, but depleted in
clusters 3 and 8). Next, we investigated the proportion of samples per cancer
type for the significant genes within a cluster (**Figure 6**; heatmap).
Surprisingly, cluster-specific tumors were not predominated by one or more
highly mutated genes across all cancers. Instead, when observing the mutation
rate for these genes within samples that belong to a cluster, the mutation rate
is heterogeneous across tumors by tissue origin ( _e.g._ In cluster 4, CESC was
enriched for _H2AFX_, OV was enriched for _HIST1H2BD_, and UCEC was enriched for
_HIST1H2AC_). Even among the top most mutationally enriched genes within
clusters there is no global pattern, indicating that our clusters are not driven
by individual genes, but rather networks as a whole. Taken together, our data
identifies a framework of cancer type-specific mutations associated with
specific clusters.

```{r, results='asis'}
cat("\\newpage")
```

#### Figure 6
```{r child = '../code/figure6.Rmd'}
```

**Figure 6: Gene level analysis reveals tissue-specific class signatures.** 
Odds ratio plot; Column uses a logarithmic axis to represent odds ratio with a
95% confidence interval. Row represents the significant genes from each cluster.
Each cluster was compared against the background (all other clusters) to find
deferentially mutated genes. Significant genes (pvalue < 0.01) were selected and
limited to the top ten results for each clusters. Heatmap; Column represents the
cancer type. Row represents the significant genes as described above. The
heatmap shows the proportion of samples per cluster, per cancer type that is
mutated for each of the significant genes. Depleted significant genes (left).
Enriched significant genes (right).

```{r, results='asis'}
cat("\\newpage")
```

### Enrichment of pathways in metastasis is cluster-specific

Following the same logic we used to investigate cluster-specific enrichment of
pathways (**figure 5B**) we compared metastatic _vs._ non-metastatic
tumors, as we find that metastatic tumors are distributed across all ten
clusters ( _e.g._ **figure 3E**). Using all non-metastatic tumors as background,
we found very low levels of enrichment (≤ 10%) in a handful of pathways. We
reasoned that the individual clusters might be too different to detect global
metastasis enrichment signal given the small sample size (n = 215 metastatic
tumor samples).

Therefore, we calculated cluster-specific enrichment in metastatic tumors and
found a total of 31 enriched pathways across all clusters (**table 1**). A
number of enrichments were found in multiple clusters represented by pathways
that were already shown to be enriched in a neighboring cluster. For example,
“Signaling by _PTK6_” is enriched in non-metastatic samples of cluster 8 (see
**supplementary figure S4**) but not in 7 and 9. This pathway is enriched in
metastatic tumors of clusters 7 and 9 ($p < 10^{-3}$, **table 1**). This is also
true of “Erythropoietin activates _RAS_”, which is enriched in non-metastatic
tumors of cluster 8 (**supplementary figure S4**) and also in metastatic tumors
of clusters 7 and 9. Cluster 4 metastatic samples were enriched for “Fc epsilon
receptor (FCERI) signaling”, a key neutrophil pathway, which is also specific to
clusters 2, 7 and 8 non-metastatic tumors. Thus, pathway lesions specific to
metastases tend toward complementary mutations that are otherwise specific to
non-metastatic tumors of other clusters.

```{r, results='asis'}
cat("\\newpage")
```

#### Table 1
```{r child = '../code/table1.Rmd'}
```

**Table 1: Cluster-specific enriched pathways (effect size ≥ 0.30) in metastasis.**

```{r, results='asis'}
cat("\\newpage")
```

### Pathway disruption clusters vary in short-term prognosis of survival

```{r child='survival.Rmd'}
```

## Discussion
 
```{r child='discussion.Rmd'}
```

## Materials & Methods

All code for producing the analyses and figures herein are included in this
fully reproducible manuscript in R markdown format. R markdown files and all
other models are available from [our
repositories](https://github.com/dennishazelett/TTmanu) on the distributed
version control site, GitHub.

### Selection of pathways

To understand the molecular mechanism of cancer at a pathway level, we used
Reactome (https://reactome.org/), a knowledge-based pathway database. The
mapping files of ENSEMBL genes to pathways, pathway hierarchy relationships, and
complexes to top pathways were downloaded from
https://reactome.org/download-data. Using these data, we imposed pathway
criteria to define basic cellular processes and biochemical pathways: (1)
human-derived pathways ( _“HSA”_) (2) limited to grandchild node for each parent
pathway ( _e.g._ ‘Beta-catenin independent WNT signaling’ in ‘Signal
Transduction’) (3) exclusion of pathways in the parent pathway: "Disease",
"Muscle contraction", and "Reproduction" or pathway names that include any of
the following keywords: "disease", "defect", "cancer", "oncogenic", "disorder",
"mutant", "loss", "infection", "bacteria", or "listeria". While some of the
excluded pathways have been shown to play an important role in cancer, they are
highly specialized ( _e.g._ "PI3K/AKT Signaling in Cancer"). Additionally, for
most of the excluded pathways, a neutral version pathway of the pathway exists (
_e.g._ "PIP3 activates Akt signaling"). Finally, we mapped the 18,577 Ensembl
IDs from the TCGA dataset to the highly selected Reactome pathways. This
operation produced a lookup table that consisted of 377 pathways mapped to 8,940
genes.

### Filtering genes

We filtered likely erroneous mutations due to transcription coupled repair. Our
approach was to determine the status of each gene ( _i.e._ expressed or not
expressed) in each tissue type in order to exclude gene mutations low expressed
enes. To do this, we obtained the TCGA RNA sequencing data adjusted for batch
effect dataset (https://pancanatlas.xenahubs.net). Using the data, we removed
the genes and tumor samples that were not included in our analysis, grouped the
tumor samples by tissue type and computed the mean expression value for each
gene. A minimum threshold of 10 transcripts per million was set for expressed
genes based on an inflection point observed when plotting the mean expression
values of genes ranked by expression in each tissue type. Genes that did not
meet this threshold were considered not expressed. This operation produced a
lookup table for gene expression status in each tumor sample for 18,127 genes.

### Class and stage-specific Enrichment Calculations

We calculated the enrichment of pathways in one set of tumors as the relative
fraction of tumors (with estimated uncertainty) with a mutated gene in the
pathway to all tumors, inclusive of the category of interest. To do this
operation, we use the beta distribution to permute a posterior distribution on
the fraction of tumors with a pathway mutated for each category based on the
observed set, and compared this to the posterior obtained from the full set of
tumors (all tumors) as the distribution of differences between all permuted
samples. We considered a pathway enriched if the 95% range of credible
differences thus obtained excludes 0 and the mean credible difference was
greater than or equal to 30% enrichment, which excludes a large number of small
differences that are not likely to be biologically relevant.

### Models of survival based on population-specific longevity data

We used Bayesian inference to explore the relative impact of cancer diagnosis on
survival at the time of diagnosis. Our model assumes that at birth each person
has a small initial probability per unit time to die, and that this probability
grows exponentially in time at some baseline rate that matches the observed CDC
data. The rate of growth in risk is assumed to jump to a higher level at
diagnosis, which we take as a proxy for onset. This represents an accelerated
aging type model. To fit the model, we constrain the baseline risks using the
available CDC life tables. We then constrain the cancer specific multiplier
using the longevity data within the cancer tumor dataset. Our model estimates
the effect of cancer type and class-specific cancer effects independently,
resulting in a cancer and class specific estimate of the aging rate multiplier.
A detailed description of the CDC longevity model and likelihood is provided in
the **Supplementary methods**.

## Acknowledgements

Wish wish to thank David Van Valen, Simon Knott and Kate Lawrenson for critical
reading of this manuscript, and Ivetth Corona for early discussions and
feedback.

## Declarations

The authors have no conflicts of interest to declare. This work was supported by
a grant from the Cedars-Sinai Precision Health Initiative.

```{r, results='asis'}
cat("\\newpage")
```

```{r child='../text/fig_legend.Rmd'}
```

```{r, results='asis'}
cat("\\newpage")
```

## References
